---
title: "i4_swab_metagenomics.Rmd"
author: "Braden T Tierney"
date: '2022-10-16'
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

### METHOLODOGICAL NOTE

F2 -- HIGH COVERAGE + TOP GENES
F3/4 TOP -- HIGH COVERAGE AND LP SIGS
F3/4 BOTTOM -- HIGH COVERAGE AND NON LP SIGS WITH HIGH L2FC
F6 -- NON LP SIGS HIGH COVERAGE

####### LIBRARIES
```{r}
library(tidyverse)
library(ggplot2)
library(umap)
library(lme4)
library(broom.mixed)
library(gridExtra)
library(broom)
library(ape)
library(cowplot)
library(reshape2)
library(taxonomizr)
library(circlize)
library(ggtree)
library(RColorBrewer)
library(ComplexHeatmap)
library(ggbeeswarm)
library(lmerTest)
library(ComplexUpset)
```
####### DATA LOAD AND ALPHA DIVERSITY
```{r}

setwd('~/Dropbox (Mason Lab)/i4/i4_data_packet/')

sanitize_sample_names <- function(data){
  temp = data%>% rownames_to_column('temp') %>% mutate(namelengths = nchar(temp))
  temp = temp %>% mutate(temp = if_else(namelengths>=3 & str_sub(temp,nchar(temp),nchar(temp))=='D',str_sub(temp,1,nchar(temp)-1),temp))
  return(temp %>% column_to_rownames('temp') %>% select(-namelengths))
}

# load metadata
meta = read.csv('../i4_swab_metadata.csv') %>% mutate(location = if_else(Crew.ID == 'Capsule','Capsule',Body.Location))
meta$SeqID = gsub('ELMB_','',meta$SeqID)
meta$SeqID = gsub('SW_','',meta$SeqID)
meta$Timepoint_Recode = factor(meta$Timepoint)
levels(meta$Timepoint_Recode) = c(NA,'PRE-LAUNCH','POST-LAUNCH','PRE-LAUNCH','MID-FLIGHT','MID-FLIGHT','POST-LAUNCH','POST-LAUNCH','PRE-LAUNCH')

meta = meta %>% distinct %>% mutate(Timepoint_Recode2 = if_else(as.character(Timepoint_Recode) == 'MID-FLIGHT',Timepoint,as.character(Timepoint_Recode)))
meta$Timepoint_Recode2 = factor(meta$Timepoint_Recode2,levels = c('PRE-LAUNCH','Flight 1','Flight 2','POST-LAUNCH'))
meta$Timepoint = factor(meta$Timepoint,levels=c('21-Jun','21-Aug','Sept Pre-Launch','Flight 1','Flight 2','Sept Post-Return','November','21-Dec',NA))
meta$Timepoint_Numeric = as.numeric(meta$Timepoint)
meta$Timepoint_Recode = factor(meta$Timepoint_Recode,levels = c('MID-FLIGHT','PRE-LAUNCH','POST-LAUNCH'))

#meta = meta %>% filter(Body.Location != "Swab Water",location!='Capsule',Body.Location != 'Open Air Control', Body.Location != "Deltoid - Pre-Biospy", !is.na(Body.Location))
meta = meta %>% mutate(isoral = if_else(Body.Location == 'Oral',1,0))
meta = meta %>% mutate(isnasal = if_else(Body.Location == 'Nasal',1,0))
meta = meta %>% mutate(isskin = if_else(Body.Location != 'Oral' & Body.Location != 'Nasal',1,0))
meta = meta %>% mutate(Armpit = if_else(Body.Location == 'Armpit',1,0))
meta = meta %>% mutate(web = if_else(Body.Location == 'Toe Web Space',1,0))
meta = meta %>% mutate(nape = if_else(Body.Location == 'Nape of Neck',1,0))
meta = meta %>% mutate(postauric = if_else(Body.Location == 'Post-Auricular',1,0))
meta = meta %>% mutate(fore = if_else(Body.Location == 'Forearm',1,0))
meta = meta %>% mutate(bb = if_else(Body.Location == 'Belly Button',1,0))
meta = meta %>% mutate(gc = if_else(Body.Location == 'Gluteal Crease',1,0))
meta = meta %>% mutate(nasal = if_else(Body.Location == 'Nasal',1,0))
meta = meta %>% mutate(Tzone = if_else(Body.Location == 'T-Zone',1,0))
meta = meta %>% mutate(Oral = if_else(Body.Location == 'Oral',1,0))

# kraken metag
kraken_metag = read.delim('kraken2_bracken_abundances/kraken2_bracken_metagenomics.tsv',check.names=F)
kraken_tax1 = kraken_metag %>% select(name,taxonomy_id)
kraken_metag_num = kraken_metag %>% select(name,all_of(grep('bracken_num',colnames(.))))%>% melt
counts = kraken_metag_num %>% ungroup %>% select(name,value) %>% group_by(name)  %>% mutate(count = if_else(value>0,1,0))  %>% summarise(s = sum(count))
tokeep = counts%>% select(name) %>% unlist %>% unname
kraken_metag = kraken_metag %>% select(name,all_of(grep('bracken_frac',colnames(.)))) 
colnames(kraken_metag) = map(colnames(kraken_metag),function(x) gsub('.qc.bracken_frac','',x))
kraken_metag = kraken_metag%>%  data.frame(check.names=F)%>% column_to_rownames('name') %>% t %>% data.frame(check.names=F)
kraken_metag = sanitize_sample_names(kraken_metag)

# kraken metat
kraken_metat = read.delim('kraken2_bracken_abundances/kraken2_bracken_metatranscriptomics.tsv',check.names=F)
kraken_tax2 = kraken_metat %>% select(name,taxonomy_id)
kraken_metat_num = kraken_metat %>% select(name,all_of(grep('bracken_num',colnames(.))))%>% melt
counts = kraken_metat_num %>% ungroup %>% select(name,value) %>% group_by(name) %>% mutate(count = if_else(value>0,1,0))  %>% summarise(s = sum(count))
tokeep = counts  %>% select(name) %>% unlist %>% unname
kraken_metat = kraken_metat %>% select(name,all_of(grep('bracken_frac',colnames(.))))%>% data.frame(check.names=F)
colnames(kraken_metat) = map(colnames(kraken_metat),function(x) gsub('.qc.bracken_frac','',x))
kraken_metat = kraken_metat%>% as.data.frame%>% column_to_rownames('name') %>% t%>% data.frame(check.names=F)

# bac GTDB metag/metat
gtdb_metag = read.table("GTDB_r207_xtree_abundances/GTDB_.005_.0025_metagenomics_species_ra.tsv",check.names=F,sep='\t')
gtdb_metat = read.table('GTDB_r207_xtree_abundances/GTDB_.005_.0025_metatranscriptomics_species_ra.tsv',check.names=F,sep='\t') %>% t %>% data.frame(check.names=F)

gtdb_metag = sanitize_sample_names(gtdb_metag  %>% t %>% as.data.frame(check.names=F))

# bac assembled metag/metat
bass_metag = read.table("ASSEMBLED-BACTERIAL-MAGS_.01_.005_metagenomics_species_ra.tsv",check.names=F,sep='\t')
bass_metat = read.table('ASSEMBLED-BACTERIAL-MAGS_.01_.005_metatranscriptomics_species_ra.tsv',check.names=F,sep='\t')

bass_metag = sanitize_sample_names(bass_metag %>% t %>% as.data.frame(check.names=F))

# viral metag/metat
viral_metag = read.csv("~/Dropbox (Mason Lab)/i4/i4_data_packet/GenBank_viral_xtree_abundances/genbank-viral_.02_.0.015_metagenomics_species_ra.tsv",check.names=F,sep='\t')
viral_metat = read.csv('~/Dropbox (Mason Lab)/i4/i4_data_packet/GenBank_viral_xtree_abundances/genbank-viral_.02_.0.015_metatranscriptomics_species_ra.tsv',check.names=F,sep='\t')

viral_metag = sanitize_sample_names(viral_metag  %>% t %>% as.data.frame(check.names=F))

# viral assembled metag/metat
viral_ass_metag = read.table("ASSEMBLED-VIRAL-GENOMES_.01_.0.005_metagenomics_species_ra.tsv",check.names=F,sep='\t')
viral_ass_metat = read.table('ASSEMBLED-VIRAL-GENOMES_.01_.0.005_metatranscriptomics_species_ra.tsv',check.names=F,sep='\t')

viral_ass_metag = sanitize_sample_names(viral_ass_metag %>% t %>% as.data.frame(check.names=F))

# Gene metag/metat

#### NEED TO UPDATE TO LOAD ONLY SIGNIFICANT GENES
gene_metag = read.table('~/Dropbox (Mason Lab)/i4/i4_data_packet/gene_catalog/gene_catalog90_filtered_relab_metagenomics_subset.tsv',sep='\t',check.names=F,header = T,row.names = 1) 
gene_metag = sanitize_sample_names(gene_metag %>% t %>% as.data.frame(check.names=F))

gene_metat = read.table('~/Dropbox (Mason Lab)/i4/i4_data_packet/gene_catalog/gene_catalog90_filtered_relab_metatranscriptomics_subset.tsv',sep='\t',check.names=F,header = T,row.names = 1)
gene_metat = gene_metat %>% t %>% as.data.frame(check.names=F)

compute_diversity <- function(data){
  simpson = map(data, function(x) vegan::diversity(x,'simpson')) %>% data.frame(check.names=F) %>% t %>% data.frame(check.names=F) %>% rownames_to_column('sample_id') %>% mutate(type = 'SIMPSON')

  shannon = map(data, function(x) vegan::diversity(x,'shannon')) %>% data.frame(check.names=F) %>% t %>% data.frame(check.names=F) %>% rownames_to_column('sample_id')  %>% mutate(type = 'SHANNON')
  
  richness = map(data, function(x) sum(x!=0)) %>% data.frame(check.names=F)  %>% t %>% data.frame(check.names=F) %>% rownames_to_column('sample_id')  %>% mutate(type = 'RICHNESS')
  
  divs = bind_rows(shannon,simpson,richness)
  
  colnames(divs)[2] = 'diversity'

  return(divs)
}

kraken2_metag_divs = compute_diversity(kraken_metag %>% t%>% as.data.frame(check.names=F)) %>% mutate(dset = 'KRAKEN2') %>% mutate(seqtype = 'METAGENOMICS')
kraken2_metat_divs = compute_diversity(kraken_metat %>% t%>% as.data.frame(check.names=F)) %>% mutate(dset = 'KRAKEN2') %>% mutate(seqtype = 'METATRANSCRIPTOMICS')

gtdb_metag_divs = compute_diversity(gtdb_metag %>% t %>% as.data.frame(check.names=F)) %>% mutate(dset = 'GTDB')%>% mutate(seqtype = 'METAGENOMICS')
gtdb_metat_divs = compute_diversity(gtdb_metat %>% t %>% as.data.frame(check.names=F)) %>% mutate(dset = 'GTDB')%>% mutate(seqtype = 'METATRANSCRIPTOMICS')

genvir_metag_divs = compute_diversity(viral_metag %>% t %>% as.data.frame(check.names=F)) %>% mutate(dset = 'GENBANK-VIRUSES') %>% mutate(seqtype = 'METAGENOMICS')
genvir_metat_divs = compute_diversity(viral_metat) %>% mutate(dset = 'GENBANK-VIRUSES')%>% mutate(seqtype = 'METATRANSCRIPTOMICS')

bacass_metag_divs = compute_diversity(bass_metag %>%t%>% as.data.frame(check.names=F)) %>% mutate(dset = 'ASSEMBLED-BACTERIA')%>% mutate(seqtype = 'METAGENOMICS')
bacass_metat_divs = compute_diversity(bass_metat) %>% mutate(dset = 'ASSEMBLED-BACTERIA')%>% mutate(seqtype = 'METATRANSCRIPTOMICS')

virass_metag_divs = compute_diversity(viral_ass_metag%>%t %>% as.data.frame(check.names=F)) %>% mutate(dset = 'ASSEMBLED-VIRUSES')%>% mutate(seqtype = 'METAGENOMICS')
virass_metat_divs = compute_diversity(viral_ass_metat) %>% mutate(dset = 'ASSEMBLED-VIRUSES')%>% mutate(seqtype = 'METATRANSCRIPTOMICS')

gene_metag_divs = compute_diversity(gene_metag%>%  t %>% as.data.frame(check.names=F)) %>% mutate(dset = 'GENE-CATALOG')%>% mutate(seqtype = 'METAGENOMICS')
gene_metat_divs = compute_diversity(gene_metat %>% t %>% as.data.frame(check.names=F)) %>% mutate(dset = 'GENE-CATALOG')%>% mutate(seqtype = 'METATRANSCRIPTOMICS')

diversities = bind_rows(kraken2_metag_divs,kraken2_metat_divs,gtdb_metag_divs,gtdb_metat_divs,genvir_metag_divs,genvir_metat_divs,bacass_metag_divs,bacass_metat_divs,virass_metag_divs,virass_metat_divs,gene_metag_divs,gene_metat_divs)

```
####### BETA DIVERSITY COMPUTATION
```{r}
# beta diversity
compute_beta_div <- function(data,seqtype){
  out1 = vegan::vegdist(data) %>% as.matrix %>% reshape2::melt() %>% mutate(seqtype = seqtype)
  colnames(out1) = c('REF','QUERY','DISTANCE','SEQTYPE')
  return(out1)
}

kraken_metag_betadiv = compute_beta_div(kraken_metag,'METAGENOMICS')%>% mutate(dset = 'KRAKEN2')
kraken_metat_betadiv = compute_beta_div(kraken_metat,'METATRANSCRIPTOMICS')%>% mutate(dset = 'KRAKEN2')

viral_metagtemp=viral_metag
viral_metagtemp[is.na(viral_metagtemp)] = 0
viral_metagtemp = viral_metagtemp[rownames(viral_metagtemp)[!rowSums(viral_metagtemp) == 0],]
viral_metag_bdiv =  compute_beta_div(viral_metagtemp,'METAGENOMICS')%>% mutate(dset = 'GENBANK-VIRUSES')
viral_metat_bdiv = compute_beta_div(viral_metat%>% t %>% as.data.frame(check.names=F),'METATRANSCRIPTOMICS')%>% mutate(dset = 'GENBANK-VIRUSES')
viral_ass_metag_bdiv = compute_beta_div(viral_ass_metag,'METAGENOMICS')%>% mutate(dset = 'ASSEMBLED-VIRUSES')
viral_ass_metat_bdiv = compute_beta_div(viral_ass_metat %>% t%>% as.data.frame(check.names=F),'METATRANSCRIPTOMICS')%>% mutate(dset = 'ASSEMBLED-VIRUSES')

bass_metag_bdiv = compute_beta_div(bass_metag,'METAGENOMICS')%>% mutate(dset = 'ASSEMBLED-BACTERIA')
bass_metat_bdiv = compute_beta_div(bass_metat%>% t%>% as.data.frame(check.names=F),'METATRANSCRIPTOMICS')%>% mutate(dset = 'ASSEMBLED-BACTERIA')

gene_metag_bdiv = compute_beta_div(gene_metag,'METAGENOMICS')%>% mutate(dset = 'GENE-CATALOG')
gene_metat_bdiv = compute_beta_div(gene_metat,'METATRANSCRIPTOMICS')%>% mutate(dset = 'GENE-CATALOG')

gtdb_metag_bdiv = compute_beta_div(gtdb_metag,'METAGENOMICS')%>% mutate(dset = 'GTDB')
gtdb_metat_bdiv = compute_beta_div(gtdb_metat,'METATRANSCRIPTOMICS') %>% mutate(dset = 'GTDB')

betadivs = bind_rows(kraken_metag_betadiv,kraken_metat_betadiv,gtdb_metag_bdiv,gtdb_metat_bdiv,gene_metag_bdiv,gene_metat_bdiv,bass_metag_bdiv,bass_metat_bdiv,viral_ass_metag_bdiv,viral_ass_metat_bdiv,viral_metag_bdiv,viral_metat_bdiv)
```
####### LOAD IN TAXONOMIC AND FUNCTIONAL ANNOTATIONS
```{r}
### load in the annotation data
setwd('~/Dropbox (Mason Lab)/i4/i4_data_packet/')

# kraken taxonomies
kraken_tax = bind_rows(kraken_tax1,kraken_tax2) %>% distinct
taxa = getTaxonomy(kraken_tax$taxonomy_id)
taxa = data.frame(taxa) %>% rownames_to_column('taxid')
taxa$taxid = as.numeric(taxa$taxid)
kraken_tax = inner_join(taxa,kraken_tax,by=c('taxid'='taxonomy_id')) %>% rename(kraken_column_name = name)

# genbank viral phyla
genvir_tax = read.delim('../genbank_viral_taxonomies.tsv',header=T) 
xtreemap = read.delim('../xtree_all_db_mapping',sep='\t',header=F)
genvir_tax = left_join(genvir_tax,xtreemap,by= c('contigid'='V1'))
genvir_tax = genvir_tax %>% select(V2,Phy_in_root,Cla_in_root,Ord_in_root,Fam_in_root,Gen_in_root,Spe_in_root) %>% rename(query= V2)
colnames(genvir_tax) = gsub('_in_root','',colnames(genvir_tax))

# gtdb taxonomies (parse from file itself)
gtdb_tax = separate(col=`c(colnames(gtdb_metag), colnames(gtdb_metat))`,tibble(c(colnames(gtdb_metag),colnames(gtdb_metat))),sep=';',into = c('Domain','Phylum','Class','Order','Family','Genus','Species'))
gtdb_tax = gtdb_tax %>% mutate(Domain = str_replace(Domain, "d__", ""))
gtdb_tax = gtdb_tax %>% mutate(Phylum = str_replace(Phylum, "p__", ""))
gtdb_tax = gtdb_tax %>% mutate(Class = str_replace(Class, "c__", ""))
gtdb_tax = gtdb_tax %>% mutate(Order = str_replace(Order, "o__", ""))
gtdb_tax = gtdb_tax %>% mutate(Family = str_replace(Family, "f__", ""))
gtdb_tax = gtdb_tax %>% mutate(Genus = str_replace(Genus, "g__", ""))
gtdb_tax = gtdb_tax %>% mutate(Species = str_replace(Species, "s__", ""))
gtdb_tax = bind_cols(tibble(c(colnames(gtdb_metag),colnames(gtdb_metat))),gtdb_tax)
colnames(gtdb_tax)[1] = 'query'

# assembled bacterial taxonomies
taxdata = read.csv('~/Dropbox (Mason Lab)/i4/i4_data_packet/gtdbtk.bac120.summary.tsv',sep='\t',row.names = 1) %>% mutate(phylum = strsplit(classification,'p__') %>% map_chr(2) %>% strsplit(';') %>% map_chr(1)) %>% rownames_to_column('id') 
taxdata$species = map(taxdata$classification, function(x) gsub("^.*__", "", x)) %>% unlist %>% unname

bar = taxdata$classification
bar = gsub('s__','',bar)
bar = gsub('g__','',bar)
bar = gsub('f__','',bar)
bar = gsub('c__','',bar)
bar = gsub('o__','',bar)
bar = gsub('p__','',bar)
bar = gsub('d__','',bar)

foo = str_split_fixed(bar, ";", 7) %>% data.frame

taxdata = bind_cols(taxdata,foo)
bacass_taxa = taxdata %>% mutate(labelval = if_else(X7 != '',X7,if_else(X6 != '',X6,if_else(X5 != '',X5,if_else(X4 != '',X4,if_else(X3 != '',X3, if_else(X2 != '',X2,X1))))))) %>% select(id,X1,X2,X3,X4,X5,X6,X7) %>% rename(Domain = X1,Phylum = X2, Class = X3, Order = X4,Family = X5,Genus = X6,Species = X7,query = id)

# assembled viral taxonomies
taxdata = read.csv('closest_genbank_taxonomies.csv',sep=',',row.names = 1) %>% mutate(phylum = strsplit(taxonomy,';') %>% map_chr(1)) %>% group_by(query) %>% slice_min(distance_from_reference,n=1) %>% ungroup %>%  select(query,phylum,taxonomy)
taxdata$species = taxdata$taxonomy %>% map(function(x) strsplit(stringi::stri_reverse(x),';') %>% map_chr(1) %>% stringi::stri_reverse()) %>% unlist %>% unname
virass_tax = taxdata

# gene function mapping
gene_annos = read.csv('~/Dropbox (Mason Lab)/i4/i4_data_packet/gene_catalog/i4_90perc_congene_annotations.csv',header=T) 
gene_annos = gene_annos %>% select(Locus.Tag,Gene,Product,DbXrefs)
colnames(gene_annos) = c('yvar','gene_name','product','other_annotations')
```
####### ALPHA DIVERSITY 1
```{r}
### alpha diversity analysis

metasub = meta %>% filter(location != 'Capsule',Body.Location != "Swab Water", Body.Location != "Deltoid - Pre-Biospy", !is.na(Body.Location))

metasub$Timepoint_Recode = factor(metasub$Timepoint_Recode,levels = c('PRE-LAUNCH','MID-FLIGHT','POST-LAUNCH'))
metasub = metasub %>% mutate(Timepoint = if_else(Timepoint == 'Flight 1' | Timepoint == 'Flight 2','Mid-Flight',as.character(Timepoint)))
metasub$Timepoint = factor(metasub$Timepoint,levels=c('Mid-Flight','21-Jun','21-Aug','Sept Pre-Launch','Sept Post-Return','November','21-Dec',NA))
metasub$Timepoint_Recode = factor(metasub$Timepoint_Recode,levels=c('MID-FLIGHT','PRE-LAUNCH','POST-LAUNCH'))

diversity_meta = inner_join(diversities,meta,by=c('sample_id'='SeqID')) %>% mutate(skinoralnasal = if_else(isoral == 1,'ORAL',if_else(isskin==1,'SKIN','NASAL')))

compute_overall_div_lmers <- function(dataset,dset2){
  a = lmer(data = dataset %>% filter(type == 'SHANNON',seqtype == 'METAGENOMICS',dset == dset2), diversity ~ Timepoint_Recode + (1|Crew.ID)) %>% tidy %>% mutate(type = 'SHANNON',seqtype = 'METAGENOMICS',dset = dset2)
  b = lmer(data = dataset %>% filter(type == 'SIMPSON',seqtype == 'METAGENOMICS',dset == dset2), diversity ~ Timepoint_Recode + (1|Crew.ID)) %>% tidy%>% mutate(type = 'SIMPSON',seqtype = 'METAGENOMICS',dset = dset2) 
  c = lmer(data = dataset %>% filter(type == 'RICHNESS',seqtype == 'METAGENOMICS',dset == dset2), diversity ~ Timepoint_Recode + (1|Crew.ID)) %>% tidy %>% mutate(type = 'RICHNESS',seqtype = 'METAGENOMICS',dset = dset2)
  d = lmer(data = dataset %>% filter(type == 'SHANNON',seqtype == 'METATRANSCRIPTOMICS',dset == dset2), diversity ~ Timepoint_Recode + (1|Crew.ID)) %>% tidy %>% mutate(type = 'SHANNON',seqtype = 'METATRANSCRIPTOMICS',dset = dset2)
  e = lmer(data = dataset %>% filter(type == 'SIMPSON',seqtype == 'METATRANSCRIPTOMICS',dset == dset2), diversity ~ Timepoint_Recode + (1|Crew.ID)) %>% tidy%>% mutate(type = 'SIMPSON',seqtype = 'METATRANSCRIPTOMICS',dset = dset2) 
  f = lmer(data = dataset %>% filter(type == 'RICHNESS',seqtype == 'METATRANSCRIPTOMICS',dset == dset2), diversity ~ Timepoint_Recode + (1|Crew.ID)) %>% tidy %>% mutate(type = 'RICHNESS',seqtype = 'METATRANSCRIPTOMICS',dset = dset2)
  g = lmer(data = dataset %>% filter(type == 'RICHNESS',dset == dset2), diversity ~ Timepoint_Recode*seqtype + (1|Crew.ID)) %>% tidy %>% mutate(type = 'RICHNESS',seqtype = 'METATRANSCRIPTOMICS VS METAGENOMICS',dset = dset2)
  h = lmer(data = dataset %>% filter(type == 'SIMPSON',dset == dset2), diversity ~ Timepoint_Recode*seqtype + (1|Crew.ID)) %>% tidy %>% mutate(type = 'RICHNESS',seqtype = 'METATRANSCRIPTOMICS VS METAGENOMICS',dset = dset2)
  i = lmer(data = dataset %>% filter(type == 'SHANNON',dset == dset2), diversity ~ Timepoint_Recode*seqtype + (1|Crew.ID)) %>% tidy %>% mutate(type = 'RICHNESS',seqtype = 'METATRANSCRIPTOMICS VS METAGENOMICS',dset = dset2)
  return(bind_rows(a,b,c,d,e,f,g,h,i))
}


compute_overall_div_lmers_bs <- function(dataset,dset2){
  a = lmer(data = dataset %>% filter(type == 'SHANNON',seqtype == 'METAGENOMICS',dset == dset2), diversity ~ Timepoint_Recode*isoral + (1|Crew.ID)) %>% tidy %>% mutate(type = 'SHANNON',seqtype = 'METAGENOMICS',dset = dset2)%>% mutate(site = 'ORAL')
  b = lmer(data = dataset %>% filter(type == 'SIMPSON',seqtype == 'METAGENOMICS',dset == dset2), diversity ~ Timepoint_Recode*isskin + (1|Crew.ID)) %>% tidy%>% mutate(type = 'SIMPSON',seqtype = 'METAGENOMICS',dset = dset2)  %>% mutate(site = 'SKIN')
  c = lmer(data = dataset %>% filter(type == 'RICHNESS',seqtype == 'METAGENOMICS',dset == dset2), diversity ~ Timepoint_Recode*isnasal + (1|Crew.ID)) %>% tidy %>% mutate(type = 'RICHNESS',seqtype = 'METAGENOMICS',dset = dset2)%>% mutate(site = 'NASAL')
  
  d = lmer(data = dataset %>% filter(type == 'SHANNON',seqtype == 'METAGENOMICS',dset == dset2), diversity ~ Timepoint_Recode*isoral + (1|Crew.ID)) %>% tidy %>% mutate(type = 'SHANNON',seqtype = 'METAGENOMICS',dset = dset2)%>% mutate(site = 'ORAL')
  e = lmer(data = dataset %>% filter(type == 'SIMPSON',seqtype == 'METAGENOMICS',dset == dset2), diversity ~ Timepoint_Recode*isskin + (1|Crew.ID)) %>% tidy%>% mutate(type = 'SIMPSON',seqtype = 'METAGENOMICS',dset = dset2) %>% mutate(site = 'SKIN')
  f = lmer(data = dataset %>% filter(type == 'RICHNESS',seqtype == 'METAGENOMICS',dset == dset2), diversity ~ Timepoint_Recode*isnasal + (1|Crew.ID)) %>% tidy %>% mutate(type = 'RICHNESS',seqtype = 'METAGENOMICS',dset = dset2)%>% mutate(site = 'NASAL')
  
  h = lmer(data = dataset %>% filter(type == 'SHANNON',seqtype == 'METAGENOMICS',dset == dset2), diversity ~ Timepoint_Recode*isoral + (1|Crew.ID)) %>% tidy %>% mutate(type = 'SHANNON',seqtype = 'METAGENOMICS',dset = dset2)%>% mutate(site = 'ORAL')
  i = lmer(data = dataset %>% filter(type == 'SIMPSON',seqtype == 'METAGENOMICS',dset == dset2), diversity ~ Timepoint_Recode*isskin  + (1|Crew.ID)) %>% tidy%>% mutate(type = 'SIMPSON',seqtype = 'METAGENOMICS',dset = dset2)  %>% mutate(site = 'SKIN')
  j = lmer(data = dataset %>% filter(type == 'RICHNESS',seqtype == 'METAGENOMICS',dset == dset2), diversity ~ Timepoint_Recode*isnasal + (1|Crew.ID)) %>% tidy %>% mutate(type = 'RICHNESS',seqtype = 'METAGENOMICS',dset = dset2)%>% mutate(site = 'NASAL')
  
  out1 = bind_rows(a,b,c,d,e,f,h,i,j)
  
   a = lmer(data = dataset %>% filter(type == 'SHANNON',seqtype == 'METATRANSCRIPTOMICS',dset == dset2), diversity ~ Timepoint_Recode*isoral + (1|Crew.ID)) %>% tidy %>% mutate(type = 'SHANNON',seqtype = 'METATRANSCRIPTOMICS',dset = dset2)%>% mutate(site = 'ORAL')
  b = lmer(data = dataset %>% filter(type == 'SIMPSON',seqtype == 'METATRANSCRIPTOMICS',dset == dset2), diversity ~ Timepoint_Recode*isskin  + (1|Crew.ID)) %>% tidy%>% mutate(type = 'SIMPSON',seqtype = 'METATRANSCRIPTOMICS',dset = dset2)  %>% mutate(site = 'SKIN')
  c = lmer(data = dataset %>% filter(type == 'RICHNESS',seqtype == 'METATRANSCRIPTOMICS',dset == dset2), diversity ~ Timepoint_Recode*isnasal + (1|Crew.ID)) %>% tidy %>% mutate(type = 'RICHNESS',seqtype = 'METATRANSCRIPTOMICS',dset = dset2)%>% mutate(site = 'NASAL')
  
  d = lmer(data = dataset %>% filter(type == 'SHANNON',seqtype == 'METATRANSCRIPTOMICS',dset == dset2), diversity ~ Timepoint_Recode*isoral + (1|Crew.ID)) %>% tidy %>% mutate(type = 'SHANNON',seqtype = 'METATRANSCRIPTOMICS',dset = dset2)%>% mutate(site = 'ORAL')
  e = lmer(data = dataset %>% filter(type == 'SIMPSON',seqtype == 'METATRANSCRIPTOMICS',dset == dset2), diversity ~ Timepoint_Recode*isskin + (1|Crew.ID)) %>% tidy%>% mutate(type = 'SIMPSON',seqtype = 'METATRANSCRIPTOMICS',dset = dset2)  %>% mutate(site = 'SKIN')
  f = lmer(data = dataset %>% filter(type == 'RICHNESS',seqtype == 'METATRANSCRIPTOMICS',dset == dset2), diversity ~ Timepoint_Recode*isnasal + (1|Crew.ID)) %>% tidy %>% mutate(type = 'RICHNESS',seqtype = 'METATRANSCRIPTOMICS',dset = dset2)%>% mutate(site = 'NASAL')
  
  h = lmer(data = dataset %>% filter(type == 'SHANNON',seqtype == 'METATRANSCRIPTOMICS',dset == dset2), diversity ~ Timepoint_Recode*isoral + (1|Crew.ID)) %>% tidy %>% mutate(type = 'SHANNON',seqtype = 'METATRANSCRIPTOMICS',dset = dset2)%>% mutate(site = 'ORAL')
  i = lmer(data = dataset %>% filter(type == 'SIMPSON',seqtype == 'METATRANSCRIPTOMICS',dset == dset2), diversity ~ Timepoint_Recode*isskin + (1|Crew.ID)) %>% tidy%>% mutate(type = 'SIMPSON',seqtype = 'METATRANSCRIPTOMICS',dset = dset2) %>% mutate(site = 'SKIN')
  j = lmer(data = dataset %>% filter(type == 'RICHNESS',seqtype == 'METATRANSCRIPTOMICS',dset == dset2), diversity ~ Timepoint_Recode*isnasal + (1|Crew.ID)) %>% tidy %>% mutate(type = 'RICHNESS',seqtype = 'METATRANSCRIPTOMICS',dset = dset2) %>% mutate(site = 'NASAL')
  
  out2 = bind_rows(a,b,c,d,e,f,h,i,j)

  return(bind_rows(out1,out2))
}

# OVERALL DIVERSITY CHANGES
out=list()
for(d in diversity_meta$dset %>% unique){
  out[[d]] = compute_overall_div_lmers(diversity_meta,d)
}
out = bind_rows(out)
write.csv(out,'~/Dropbox (Mason Lab)/i4/random_tables_supp/overall_alpha_div_lmms.csv')
# BODY SITE SPECIFIC DIVERSITY CHANGES
out=list()
for(d in diversity_meta$dset %>% unique){
  out[[d]] = compute_overall_div_lmers_bs(diversity_meta,d)
}
out = bind_rows(out)
write.csv(out,'~/Dropbox (Mason Lab)/i4/random_tables_supp/overall_alpha_div_lmms_bs.csv')
setwd('~/Dropbox (Mason Lab)/i4/diversity_plots')

ggplot(data = diversity_meta %>% filter(type =='SHANNON',!is.na(skinoralnasal)),aes(x = Timepoint_Numeric,y = (diversity))) + geom_smooth(method = 'loess') + theme_bw()+ facet_wrap(~dset + skinoralnasal + seqtype,scales='free')
ggsave('shannon_forsupp.pdf',width=15,height=20)

ggplot(data = diversity_meta %>% filter(type =='SIMPSON',!is.na(skinoralnasal)),aes(x = Timepoint_Numeric,y = (diversity))) + geom_smooth(method = 'loess') + theme_bw()+ facet_wrap(~dset + skinoralnasal + seqtype,scales='free')
ggsave('simpson_forsupp.pdf',width=15,height=20)

ggplot(data = diversity_meta %>% filter(type !='RICHNESS',!is.na(skinoralnasal)),aes(x = Timepoint_Numeric,color=seqtype,y = (diversity))) + geom_smooth(method = 'loess') + theme_bw()+ facet_wrap(~dset + skinoralnasal + type,scales='free')+scale_color_manual(values = c('darkblue','#BE1E2D'))
ggsave('otherdiv_forsupp.pdf',width=15,height=20)


### FOR MAIN TEXT
diversity_meta$dset = factor(diversity_meta$dset, levels = c('GTDB','GENBANK-VIRUSES','GENE-CATALOG','KRAKEN2','ASSEMBLED-VIRUSES','ASSEMBLED-BACTERIA'))
diversity_meta$skinoralnasal = factor(diversity_meta$skinoralnasal, levels = c('ORAL','NASAL','SKIN'))

ggplot(data = diversity_meta %>% filter(!is.na(skinoralnasal),dset == 'GTDB',seqtype == 'METAGENOMICS',type != 'SIMPSON',type != 'SHANNON'),aes(x = Timepoint_Numeric,y = (diversity))) + geom_smooth(method = 'loess',color = 'blue') + theme_bw()+ facet_grid(type + skinoralnasal~ . ,scales='free_y')
ggsave('~/Dropbox (Mason Lab)/i4/diversity_plots/div_gtdb_metag.pdf',width=3,height=6)

ggplot(data = diversity_meta %>% filter(!is.na(skinoralnasal),dset == 'GENBANK-VIRUSES',seqtype == 'METAGENOMICS',type != 'SIMPSON',type != 'SHANNON'),aes(x = Timepoint_Numeric,y = (diversity))) + geom_smooth(method = 'loess',color = 'darkblue') + theme_bw()+ facet_grid(type + skinoralnasal~ . ,scales='free_y')
ggsave('~/Dropbox (Mason Lab)/i4/diversity_plots/div_genvir_metag.pdf',width=3,height=6)

ggplot(data = diversity_meta %>% filter(!is.na(skinoralnasal),dset == 'GENE-CATALOG',seqtype == 'METAGENOMICS',type != 'SIMPSON',type != 'SHANNON'),aes(x = Timepoint_Numeric,y = (diversity))) + geom_smooth(method = 'loess',color = 'darkblue') + theme_bw()+ facet_grid(type + skinoralnasal~ . ,scales='free_y')
ggsave('~/Dropbox (Mason Lab)/i4/diversity_plots/div_genecat_metag.pdf',width=3,height=6)


ggplot(data = diversity_meta %>% filter(!is.na(skinoralnasal),dset == 'GTDB',seqtype == 'METATRANSCRIPTOMICS',type != 'SIMPSON',type != 'SHANNON'),aes(x = Timepoint_Numeric,y = (diversity))) + geom_smooth(method = 'loess',color = '#BE1E2D') + theme_bw()+ facet_grid(type + skinoralnasal~ . ,scales='free_y')
ggsave('~/Dropbox (Mason Lab)/i4/diversity_plots/div_gtdb_metat.pdf',width=3,height=6)

ggplot(data = diversity_meta %>% filter(!is.na(skinoralnasal),dset == 'GENBANK-VIRUSES',seqtype == 'METATRANSCRIPTOMICS',type != 'SIMPSON',type != 'SHANNON'),aes(x = Timepoint_Numeric,y = (diversity))) + geom_smooth(method = 'loess',color = '#BE1E2D') + theme_bw()+ facet_grid(type + skinoralnasal~ . ,scales='free_y')
ggsave('~/Dropbox (Mason Lab)/i4/diversity_plots/div_genvir_metat.pdf',width=3,height=6)

ggplot(data = diversity_meta %>% filter(!is.na(skinoralnasal),dset == 'GENE-CATALOG',seqtype == 'METATRANSCRIPTOMICS',type != 'SIMPSON',type != 'SHANNON'),aes(x = Timepoint_Numeric,y = (diversity))) + geom_smooth(method = 'loess',color = '#BE1E2D') + theme_bw()+ facet_grid(type + skinoralnasal~ . ,scales='free_y')
ggsave('~/Dropbox (Mason Lab)/i4/diversity_plots/div_genecat_metat.pdf',width=3,height=6)

```
####### BETA DIVERSITY 1
```{r}
### beta diversity analysis (similarity with each other and capsule over time, change in different bodysites over time)

metasub = meta %>% filter(Body.Location != "Swab Water", Body.Location != "Deltoid - Pre-Biospy", !is.na(Body.Location))

metasub$Timepoint_Recode = factor(metasub$Timepoint_Recode,levels = c('PRE-LAUNCH','MID-FLIGHT','POST-LAUNCH'))
metasub = metasub %>% mutate(Timepoint = if_else(Timepoint == 'Flight 1' | Timepoint == 'Flight 2','Mid-Flight',as.character(Timepoint)))
#metasub$Timepoint = factor(metasub$Timepoint,levels=c('Mid-Flight','21-Jun','21-Aug','Sept Pre-Launch','Sept Post-Return','November','21-Dec',NA))
metasub$Timepoint_Recode = factor(metasub$Timepoint_Recode,levels=c('MID-FLIGHT','PRE-LAUNCH','POST-LAUNCH'))

m_ref = metasub
m_query = metasub

colnames(m_ref) = paste('REF_',colnames(metasub),sep='') 
m_ref = m_ref %>% dplyr::rename(REF = REF_SeqID)
colnames(m_query) = paste('QUERY_',colnames(metasub),sep='')
m_query = m_query %>% dplyr::rename(QUERY = QUERY_SeqID)

betadivs_meta = inner_join(betadivs,m_query %>% select(QUERY,QUERY_location,QUERY_Timepoint_Recode,QUERY_Timepoint_Numeric,QUERY_Timepoint,QUERY_Timepoint_Recode2,QUERY_Body.Location,QUERY_Crew.ID),by=c('QUERY'='QUERY')) 
betadivs_meta = inner_join(betadivs_meta,m_ref %>% select(REF,REF_location,REF_Timepoint_Recode,REF_Timepoint_Numeric,REF_Timepoint,REF_Timepoint_Recode2,REF_Body.Location,REF_Crew.ID),by=c('REF'='REF')) 
```
####### BETA DIVERSITY VS CAPSULE
```{r}
# visualizing similarity to each other vs capsule
col_fun = colorRamp2(c(.1,.4,.8,1), rev(c("#2ED8F0", "#1768BD","#000000","#88979E")))

plot_betadiv_info <- function(betadivs_meta,refdb,dtype){
  betadivs_meta_sub = betadivs_meta %>% filter(dset ==refdb,SEQTYPE == dtype) %>% filter(QUERY_location == REF_location | REF_location == 'Capsule',REF_Crew.ID != QUERY_Crew.ID,QUERY_Timepoint == REF_Timepoint,REF_Timepoint_Recode == QUERY_Timepoint_Recode,QUERY_location != 'Open Air Control',REF_location != 'Open Air Control')%>% select(DISTANCE,QUERY_location,REF_location,QUERY_Timepoint_Recode,REF_Timepoint_Recode,REF_Crew.ID) 
  
  wgs_bacterial_dist_meta_sub = betadivs_meta_sub %>% group_by(REF_Crew.ID,QUERY_Timepoint_Recode,QUERY_location) %>% summarise(DISTANCE_AVG = mean(DISTANCE)) %>% dcast(QUERY_Timepoint_Recode + REF_Crew.ID ~ QUERY_location,value.var = 'DISTANCE_AVG') 
  
  a = filter(wgs_bacterial_dist_meta_sub %>% filter(QUERY_Timepoint_Recode == 'PRE-LAUNCH')) %>% select(-QUERY_Timepoint_Recode) %>% filter(REF_Crew.ID != 'Capsule')%>% column_to_rownames('REF_Crew.ID') 
  b = filter(wgs_bacterial_dist_meta_sub %>% filter(QUERY_Timepoint_Recode == 'MID-FLIGHT'))%>% select(-QUERY_Timepoint_Recode) %>% column_to_rownames('REF_Crew.ID')
  c = filter(wgs_bacterial_dist_meta_sub %>% filter(QUERY_Timepoint_Recode == 'POST-LAUNCH'))%>% select(-QUERY_Timepoint_Recode) %>% column_to_rownames('REF_Crew.ID')
  
  #roworder = c("Oral", "Gluteal Crease","Nasal","T-Zone","Post-Auricular","Belly Button","Toe Web Space","Nape of Neck", "Forearm","Armpit")
  roworder = b['Capsule',] %>% melt %>% arrange(desc(value)) %>% mutate(variable = as.character(variable))%>% select(variable) %>% unlist %>% unname
  
  betadivs_meta_sub$REF_Timepoint_Recode = factor(betadivs_meta_sub$REF_Timepoint_Recode,levels=c('PRE-LAUNCH','MID-FLIGHT','POST-LAUNCH'))
  betadivs_meta_sub$QUERY_Timepoint_Recode = factor(betadivs_meta_sub$QUERY_Timepoint_Recode,levels=c('PRE-LAUNCH','MID-FLIGHT','POST-LAUNCH'))
  betadivs_meta_sub$QUERY_location = factor(betadivs_meta_sub$QUERY_location,levels=roworder)
  overallplot = ggplot(betadivs_meta_sub %>% filter(REF_location != 'Capsule'),aes(x = QUERY_location,y=DISTANCE,color = as.factor(QUERY_Timepoint_Recode)))+ geom_quasirandom(dodge.width =.8,alpha=.3) +geom_boxplot(alpha=.1)  + theme_bw() +scale_color_brewer(palette='Set1') + xlab('') + theme(axis.text.x = element_blank()) + ylim(0,1)
  
  ### COMPUTE STATS 
  
  temp = betadivs_meta_sub %>% mutate(REF_Timepoint_Recode = factor(REF_Timepoint_Recode,levels= c('PRE-LAUNCH','MID-FLIGHT','POST-LAUNCH'))) %>% filter(REF_location != 'Capsule')
  
  regout = lmer(data = temp,DISTANCE ~ REF_Timepoint_Recode*REF_location + (1|REF_Crew.ID)) %>% tidy
  
  write.csv(regout,'~/Dropbox (Mason Lab)/i4/random_tables_supp/betadiversity_regressions.csv')
  
  ap = Heatmap(a %>% select(roworder),cluster_columns =F,name = 'Bacterial WGS Dissimilarity',column_title_side = 'top',column_dend_side = 'bottom',row_split  = c('Crew','Crew','Crew','Crew'),col=col_fun,show_row_names = T,cluster_rows=F)
  bp =  Heatmap(b %>% select(roworder),cluster_columns =F,name = 'Bacterial WGS Dissimilarity',column_title_side = 'top',column_dend_side = 'bottom',row_split  = c('Crew','Crew','Crew','Crew','Capsule'),col=col_fun,show_row_names = T,cluster_rows=F)
  cp = Heatmap(c %>% select(roworder),cluster_columns =F,name = 'Bacterial WGS Dissimilarity',column_title_side = 'top',column_dend_side = 'bottom',row_split  = c('Crew','Crew','Crew','Crew'),col=col_fun,show_row_names = T,cluster_rows=F)
  
  heatmapplot = ap %v% bp %v% cp
  
  ggsave(plot = overallplot,filename = paste('beta_diversity/overview_',refdb,'_',dtype,'.pdf',sep=''),width=8,height=2)
  pdf(paste('beta_diversity/',refdb,'_',dtype,'.pdf',sep=''),width=6,height=4)
  draw(heatmapplot)
  dev.off()
  
  return(list(overallplot,heatmapplot,regout))  
}

setwd('~/Dropbox (Mason Lab)/i4')

gtdb_metag_bdivplots = plot_betadiv_info(betadivs_meta,'GTDB','METAGENOMICS')
gtdb_metat_bdivplots = plot_betadiv_info(betadivs_meta,'GTDB','METATRANSCRIPTOMICS')

kraken_metag_bdivplots = plot_betadiv_info(betadivs_meta,'KRAKEN2','METAGENOMICS')
kraken_metat_bdivplots = plot_betadiv_info(betadivs_meta,'KRAKEN2','METATRANSCRIPTOMICS')

bacass_metag_bdivplots = plot_betadiv_info(betadivs_meta,'ASSEMBLED-BACTERIA','METAGENOMICS')
bacass_metat_bdivplots = plot_betadiv_info(betadivs_meta,'ASSEMBLED-BACTERIA','METATRANSCRIPTOMICS')

virass_metag_bdivplots = plot_betadiv_info(betadivs_meta,'ASSEMBLED-VIRUSES','METAGENOMICS')
virass_metat_bdivplots = plot_betadiv_info(betadivs_meta,'ASSEMBLED-VIRUSES','METATRANSCRIPTOMICS')

virgen_metag_bdivplots = plot_betadiv_info(betadivs_meta,'GENBANK-VIRUSES','METAGENOMICS')
virgen_metat_bdivplots = plot_betadiv_info(betadivs_meta,'GENBANK-VIRUSES','METATRANSCRIPTOMICS')

gene_metag_bdivplots = plot_betadiv_info(betadivs_meta,'GENE-CATALOG','METAGENOMICS')
gene_metat_bdivplots = plot_betadiv_info(betadivs_meta,'GENE-CATALOG','METATRANSCRIPTOMICS')

betadivreg = bind_rows(bind_rows(gtdb_metag_bdivplots[[3]] %>% mutate(seqtype = 'METAGENOMICS'),gtdb_metat_bdivplots[[3]]%>% mutate(seqtype = 'METATRANSCRIPTOMICS')) %>% mutate(dataset = 'GTDB'),bind_rows(kraken_metag_bdivplots[[3]] %>% mutate(seqtype = 'METAGENOMICS'),kraken_metat_bdivplots[[3]]%>% mutate(seqtype = 'METATRANSCRIPTOMICS')) %>% mutate(dataset = 'KRAKEN2'),bind_rows(bacass_metag_bdivplots[[3]] %>% mutate(seqtype = 'METAGENOMICS'),bacass_metat_bdivplots[[3]]%>% mutate(seqtype = 'METATRANSCRIPTOMICS')) %>% mutate(dataset = 'ASSEMBLED BACTERIA'),bind_rows(virass_metag_bdivplots[[3]] %>% mutate(seqtype = 'METAGENOMICS'),virass_metat_bdivplots[[3]]%>% mutate(seqtype = 'METATRANSCRIPTOMICS')) %>% mutate(dataset = 'ASSEMBLED VIRUSES'),bind_rows(virgen_metag_bdivplots[[3]] %>% mutate(seqtype = 'METAGENOMICS'),virgen_metat_bdivplots[[3]]%>% mutate(seqtype = 'METATRANSCRIPTOMICS')) %>% mutate(dataset = 'GENBANK VIRUSES'),bind_rows(gene_metag_bdivplots[[3]] %>% mutate(seqtype = 'METAGENOMICS'),gene_metat_bdivplots[[3]]%>% mutate(seqtype = 'METATRANSCRIPTOMICS')) %>% mutate(dataset = 'GENE CATALOG'))
write.csv(betadivreg,'~/Dropbox (Mason Lab)/i4/random_tables_supp/betadivreg.csv')


```
######## LOAD IN REGRESSION OUTPUT
```{r}

setwd('~/Dropbox (Mason Lab)/i4/')

### LOAD IN REGRESSION OUTPUT AND MERGE

regdata = list()
### GTDB
regdata[[1]] = read.table("i4_data_packet/differential_abundance_or_expression_analysis/regression_output_oral_bacterial_dna_species_xtree_.05_.0025_GTDB_r207.tsv",sep='\t') %>% mutate(seqtype = 'METAGENOMICS') %>% mutate(regressionclass = 'Oral') %>% filter(grepl('Timepoint',term))   %>% mutate(dset = 'GTDB')
regdata[[2]] = read.table("i4_data_packet/differential_abundance_or_expression_analysis/regression_output_oral_bacterial_rna_species_xtree_.05_.0025_GTDB_r207.tsv",sep='\t') %>% mutate(seqtype = 'METATRANSCRIPTOMICS') %>% mutate(regressionclass = 'Oral') %>% filter(grepl('Timepoint',term)) %>% mutate(dset = 'GTDB')
regdata[[3]] = read.table("i4_data_packet/differential_abundance_or_expression_analysis/regression_output_nasal_bacterial_dna_species_xtree_.05_.0025_GTDB_r207.tsv",sep='\t') %>% mutate(seqtype = 'METAGENOMICS') %>% mutate(regressionclass = 'Nasal') %>% filter(grepl('Timepoint',term))   %>% mutate(dset = 'GTDB')
regdata[[4]] = read.table("i4_data_packet/differential_abundance_or_expression_analysis/regression_output_nasal_bacterial_rna_species_xtree_.05_.0025_GTDB_r207.tsv",sep='\t') %>% mutate(seqtype = 'METATRANSCRIPTOMICS') %>% mutate(regressionclass = 'Nasal') %>% filter(grepl('Timepoint',term)) %>% mutate(dset = 'GTDB')
regdata[[5]] = read.table("i4_data_packet/differential_abundance_or_expression_analysis/regression_output_skin_bacterial_dna_species_xtree_.05_.0025_GTDB_r207.tsv",sep='\t') %>% mutate(seqtype = 'METAGENOMICS') %>% mutate(regressionclass = 'Skin') %>% filter(grepl('Timepoint',term)) %>% mutate(dset = 'GTDB')
regdata[[6]] = read.table("i4_data_packet/differential_abundance_or_expression_analysis/regression_output_skin_bacterial_rna_species_xtree_.05_.0025_GTDB_r207.tsv",sep='\t') %>% mutate(seqtype = 'METATRANSCRIPTOMICS') %>% mutate(regressionclass = 'Skin') %>% filter(grepl('Timepoint',term)) %>% mutate(dset = 'GTDB')
regdata[[7]] = read.table("i4_data_packet/differential_abundance_or_expression_analysis/regression_output_skin_site_by_site_bacterial_dna_species_xtree_.05_.0025_GTDB_r207.tsv",sep='\t') %>% mutate(seqtype = 'METAGENOMICS') %>% mutate(regressionclass = 'Skin_Components') %>% filter(grepl('Timepoint',term)) %>% mutate(dset = 'GTDB')
regdata[[8]] = read.table("i4_data_packet/differential_abundance_or_expression_analysis/regression_output_skin_site_by_site_bacterial_rna_species_xtree_.05_.0025_GTDB_r207.tsv",sep='\t') %>% mutate(seqtype = 'METATRANSCRIPTOMICS') %>% mutate(regressionclass = 'Skin_Components') %>% filter(grepl('Timepoint',term)) %>% mutate(dset = 'GTDB')

### BACTERIAL ASSEMBLY
regdata[[9]] = read.table("i4_data_packet/differential_abundance_or_expression_analysis/regression_output_oral_bacterial_dna_species_xtree_.05_.0025_ASSEMBLIES.tsv",sep='\t') %>% mutate(seqtype = 'METAGENOMICS') %>% mutate(regressionclass = 'Oral') %>% filter(grepl('Timepoint',term)) %>% mutate(dset = 'ASSEMBLED-BACTERIA')
regdata[[10]] = read.table("i4_data_packet/differential_abundance_or_expression_analysis/regression_output_oral_bacterial_rna_species_xtree_.05_.0025_ASSEMBLIES.tsv",sep='\t') %>% mutate(seqtype = 'METATRANSCRIPTOMICS') %>% mutate(regressionclass = 'Oral') %>% filter(grepl('Timepoint',term)) %>% mutate(dset = 'ASSEMBLED-BACTERIA')
regdata[[11]] = read.table("i4_data_packet/differential_abundance_or_expression_analysis/regression_output_nasal_bacterial_dna_species_xtree_.05_.0025_ASSEMBLIES.tsv",sep='\t') %>% mutate(seqtype = 'METAGENOMICS') %>% mutate(regressionclass = 'Nasal') %>% filter(grepl('Timepoint',term))   %>% mutate(dset = 'ASSEMBLED-BACTERIA')
regdata[[12]] = read.table("i4_data_packet/differential_abundance_or_expression_analysis/regression_output_nasal_bacterial_rna_species_xtree_.05_.0025_ASSEMBLIES.tsv",sep='\t') %>% mutate(seqtype = 'METATRANSCRIPTOMICS') %>% mutate(regressionclass = 'Nasal') %>% filter(grepl('Timepoint',term)) %>% mutate(dset = 'ASSEMBLED-BACTERIA')
regdata[[13]] = read.table("i4_data_packet/differential_abundance_or_expression_analysis/regression_output_skin_bacterial_dna_species_xtree_.05_.0025_ASSEMBLIES.tsv",sep='\t') %>% mutate(seqtype = 'METAGENOMICS') %>% mutate(regressionclass = 'Skin') %>% filter(grepl('Timepoint',term)) %>% mutate(dset = 'ASSEMBLED-BACTERIA')
regdata[[14]] = read.table("i4_data_packet/differential_abundance_or_expression_analysis/regression_output_skin_bacterial_rna_species_xtree_.05_.0025_ASSEMBLIES.tsv",sep='\t') %>% mutate(seqtype = 'METATRANSCRIPTOMICS') %>% mutate(regressionclass = 'Skin') %>% filter(grepl('Timepoint',term))  %>% mutate(dset = 'ASSEMBLED-BACTERIA')
regdata[[15]] = read.table("i4_data_packet/differential_abundance_or_expression_analysis/regression_output_skin_site_by_site_bacterial_dna_species_xtree_.05_.0025_ASSEMBLIES.tsv",sep='\t') %>% mutate(seqtype = 'METAGENOMICS') %>% mutate(regressionclass = 'Skin_Components') %>% filter(grepl('Timepoint',term))  %>% mutate(dset = 'ASSEMBLED-BACTERIA')
regdata[[16]] = read.table("i4_data_packet/differential_abundance_or_expression_analysis/regression_output_skin_site_by_site_bacterial_rna_species_xtree_.05_.0025_ASSEMBLIES.tsv",sep='\t') %>% mutate(seqtype = 'METATRANSCRIPTOMICS') %>% mutate(regressionclass = 'Skin_Components') %>% filter(grepl('Timepoint',term)) %>% mutate(dset = 'ASSEMBLED-BACTERIA')

### VIRAL GENBANK

regdata[[17]] = read.csv("i4_data_packet/differential_abundance_or_expression_analysis/regression_output_oral_viral_dna_species_xtree_.02_.0.015_VIRAL_GENBANK.tsv",sep='\t') %>% mutate(seqtype = 'METAGENOMICS') %>% mutate(regressionclass = 'Oral') %>% filter(grepl('Timepoint',term)) %>% mutate(dset = 'GENBANK-VIRUSES')
regdata[[18]] = read.csv("i4_data_packet/differential_abundance_or_expression_analysis/regression_output_oral_viral_rna_species_xtree_.02_.0.015_VIRAL_GENBANK.tsv",sep='\t') %>% mutate(seqtype = 'METATRANSCRIPTOMICS') %>% mutate(regressionclass = 'Oral') %>% filter(grepl('Timepoint',term)) %>% mutate(dset = 'GENBANK-VIRUSES')
regdata[[19]] = read.csv("i4_data_packet/differential_abundance_or_expression_analysis/regression_output_nasal_viral_dna_species_xtree_.02_.0.015_VIRAL_GENBANK.tsv",sep='\t') %>% mutate(seqtype = 'METAGENOMICS') %>% mutate(regressionclass = 'Nasal') %>% filter(grepl('Timepoint',term))   %>% mutate(dset = 'GENBANK-VIRUSES')
regdata[[20]] = read.csv("i4_data_packet/differential_abundance_or_expression_analysis/regression_output_nasal_viral_rna_species_xtree_.02_.0.015_VIRAL_GENBANK.tsv",sep='\t') %>% mutate(seqtype = 'METATRANSCRIPTOMICS') %>% mutate(regressionclass = 'Nasal') %>% filter(grepl('Timepoint',term)) %>% mutate(dset = 'GENBANK-VIRUSES')
regdata[[21]] = read.csv("i4_data_packet/differential_abundance_or_expression_analysis/regression_output_skin_viral_dna_species_xtree_.02_.0.015_VIRAL_GENBANK.tsv",sep='\t') %>% mutate(seqtype = 'METAGENOMICS') %>% mutate(regressionclass = 'Skin') %>% filter(grepl('Timepoint',term)) %>% mutate(dset = 'GENBANK-VIRUSES')
regdata[[22]] = read.csv("i4_data_packet/differential_abundance_or_expression_analysis/regression_output_skin_viral_rna_species_xtree_.02_.0.015_VIRAL_GENBANK.tsv",sep='\t') %>% mutate(seqtype = 'METATRANSCRIPTOMICS') %>% mutate(regressionclass = 'Skin') %>% filter(grepl('Timepoint',term)) %>% mutate(dset = 'GENBANK-VIRUSES')
regdata[[23]] = read.csv("i4_data_packet/differential_abundance_or_expression_analysis/regression_output_skin_site_by_site_viral_dna_species_xtree_.02_.0.015_VIRAL_GENBANK.tsv",sep='\t') %>% mutate(seqtype = 'METAGENOMICS') %>% mutate(regressionclass = 'Skin_Components') %>% filter(grepl('Timepoint',term)) %>% mutate(dset = 'GENBANK-VIRUSES')
regdata[[24]] = read.csv("i4_data_packet/differential_abundance_or_expression_analysis/regression_output_skin_site_by_site_viral_rna_species_xtree_.02_.0.015_VIRAL_GENBANK.tsv",sep='\t') %>% mutate(seqtype = 'METATRANSCRIPTOMICS') %>% mutate(regressionclass = 'Skin_Components') %>% filter(grepl('Timepoint',term)) %>% mutate(dset = 'GENBANK-VIRUSES')

### VIRAL ASSEMBLED

regdata[[25]] = read.csv("i4_data_packet/differential_abundance_or_expression_analysis/regression_output_oral_viral_dna_species_xtree_.01_.0.005_ASSEMBLIES.tsv",sep='\t') %>% mutate(seqtype = 'METAGENOMICS') %>% mutate(regressionclass = 'Oral') %>% filter(grepl('Timepoint',term)) %>% mutate(dset = 'ASSEMBLED-VIRUSES')
regdata[[26]] = read.csv("i4_data_packet/differential_abundance_or_expression_analysis/regression_output_oral_viral_rna_species_xtree_.01_.0.005_ASSEMBLIES.tsv",sep='\t') %>% mutate(seqtype = 'METATRANSCRIPTOMICS') %>% mutate(regressionclass = 'Oral') %>% filter(grepl('Timepoint',term)) %>% mutate(dset = 'ASSEMBLED-VIRUSES')
regdata[[27]] = read.csv("i4_data_packet/differential_abundance_or_expression_analysis/regression_output_nasal_viral_dna_species_xtree_.01_.0.005_ASSEMBLIES.tsv",sep='\t') %>% mutate(seqtype = 'METAGENOMICS') %>% mutate(regressionclass = 'Nasal') %>% filter(grepl('Timepoint',term))   %>% mutate(dset = 'ASSEMBLED-VIRUSES')
regdata[[28]] = read.csv("i4_data_packet/differential_abundance_or_expression_analysis/regression_output_nasal_viral_rna_species_xtree_.01_.0.005_ASSEMBLIES.tsv",sep='\t') %>% mutate(seqtype = 'METATRANSCRIPTOMICS') %>% mutate(regressionclass = 'Nasal') %>% filter(grepl('Timepoint',term)) %>% mutate(dset = 'ASSEMBLED-VIRUSES')
regdata[[29]] = read.csv("i4_data_packet/differential_abundance_or_expression_analysis/regression_output_skin_viral_dna_species_xtree_.01_.0.005_ASSEMBLIES.tsv",sep='\t') %>% mutate(seqtype = 'METAGENOMICS') %>% mutate(regressionclass = 'Skin') %>% filter(grepl('Timepoint',term)) %>% mutate(dset = 'ASSEMBLED-VIRUSES')
regdata[[30]] = read.csv("i4_data_packet/differential_abundance_or_expression_analysis/regression_output_skin_viral_rna_species_xtree_.01_.0.005_ASSEMBLIES.tsv",sep='\t') %>% mutate(seqtype = 'METATRANSCRIPTOMICS') %>% mutate(regressionclass = 'Skin') %>% filter(grepl('Timepoint',term)) %>% mutate(dset = 'ASSEMBLED-VIRUSES')
regdata[[31]] = read.csv("i4_data_packet/differential_abundance_or_expression_analysis/regression_output_skin_site_by_site_viral_dna_species_xtree_.01_.0.005_ASSEMBLIES.tsv",sep='\t') %>% mutate(seqtype = 'METAGENOMICS') %>% mutate(regressionclass = 'Skin_Components') %>% filter(grepl('Timepoint',term)) %>% mutate(dset = 'ASSEMBLED-VIRUSES')
regdata[[32]] = read.csv("i4_data_packet/differential_abundance_or_expression_analysis/regression_output_skin_site_by_site_viral_rna_species_xtree_.01_.0.005_ASSEMBLIES.tsv",sep='\t') %>% mutate(seqtype = 'METATRANSCRIPTOMICS') %>% mutate(regressionclass = 'Skin_Components') %>% filter(grepl('Timepoint',term)) %>% mutate(dset = 'ASSEMBLED-VIRUSES')

### KRAKEN2
regdata[[33]] = read.csv("i4_data_packet/differential_abundance_or_expression_analysis/regression_output_oral_bacterial-viral_dna_species_kraken2_none_KRAKEN2_REFSEQDB.tsv",sep='\t') %>% mutate(seqtype = 'METAGENOMICS') %>% mutate(regressionclass = 'Oral') %>% filter(grepl('Timepoint',term)) %>% mutate(dset = 'KRAKEN2')
regdata[[34]] = read.csv("i4_data_packet/differential_abundance_or_expression_analysis/regression_output_oral_bacterial-viral_rna_species_kraken2_none_KRAKEN2_REFSEQDB.tsv",sep='\t') %>% mutate(seqtype = 'METATRANSCRIPTOMICS') %>% mutate(regressionclass = 'Oral') %>% filter(grepl('Timepoint',term)) %>% mutate(dset = 'KRAKEN2')
regdata[[35]] = read.csv("i4_data_packet/differential_abundance_or_expression_analysis/regression_output_nasal_bacterial-viral_dna_species_kraken2_none_KRAKEN2_REFSEQDB.tsv",sep='\t') %>% mutate(seqtype = 'METAGENOMICS') %>% mutate(regressionclass = 'Nasal') %>% filter(grepl('Timepoint',term))   %>% mutate(dset = 'KRAKEN2')
regdata[[36]] = read.csv("i4_data_packet/differential_abundance_or_expression_analysis/regression_output_nasal_bacterial-viral_rna_species_kraken2_none_KRAKEN2_REFSEQDB.tsv",sep='\t') %>% mutate(seqtype = 'METATRANSCRIPTOMICS') %>% mutate(regressionclass = 'Nasal') %>% filter(grepl('Timepoint',term)) %>% mutate(dset = 'KRAKEN2')
regdata[[37]] = read.csv("i4_data_packet/differential_abundance_or_expression_analysis/regression_output_skin_bacterial-viral_dna_species_kraken2_none_KRAKEN2_REFSEQDB.tsv",sep='\t') %>% mutate(seqtype = 'METAGENOMICS') %>% mutate(regressionclass = 'Skin') %>% filter(grepl('Timepoint',term)) %>% mutate(dset = 'KRAKEN2')
regdata[[38]] = read.csv("i4_data_packet/differential_abundance_or_expression_analysis/regression_output_skin_bacterial-viral_rna_species_kraken2_none_KRAKEN2_REFSEQDB.tsv",sep='\t') %>% mutate(seqtype = 'METATRANSCRIPTOMICS') %>% mutate(regressionclass = 'Skin') %>% filter(grepl('Timepoint',term)) %>% mutate(dset = 'KRAKEN2')
regdata[[39]] = read.csv("i4_data_packet/differential_abundance_or_expression_analysis/regression_output_skin_site_by_site_bacterial-viral_dna_species_kraken2_none_KRAKEN2_REFSEQDB.tsv",sep='\t') %>% mutate(seqtype = 'METAGENOMICS') %>% mutate(regressionclass = 'Skin_Components') %>% filter(grepl('Timepoint',term)) %>% mutate(dset = 'KRAKEN2')
regdata[[40]] = read.csv("i4_data_packet/differential_abundance_or_expression_analysis/regression_output_skin_site_by_site_bacterial-viral_rna_species_kraken2_none_KRAKEN2_REFSEQDB.tsv",sep='\t') %>% mutate(seqtype = 'METATRANSCRIPTOMICS') %>% mutate(regressionclass = 'Skin_Components') %>% filter(grepl('Timepoint',term)) %>% mutate(dset = 'KRAKEN2')

### GENE
regdata[[41]] = read.csv("i4_data_packet/differential_abundance_or_expression_analysis/regression_output_oral_merged_gene_metag.tsv",sep=' ') %>% mutate(seqtype = 'METAGENOMICS') %>% mutate(regressionclass = 'Oral') %>% filter(grepl('Timepoint',term)) %>% mutate(dset = 'GENE-CATALOG')
regdata[[42]] = read.csv("i4_data_packet/differential_abundance_or_expression_analysis/regression_output_oral_merged_gene_metat.tsv",sep=' ') %>% mutate(seqtype = 'METATRANSCRIPTOMICS') %>% mutate(regressionclass = 'Oral') %>% filter(grepl('Timepoint',term)) %>% mutate(dset = 'GENE-CATALOG')
regdata[[43]] = read.csv("i4_data_packet/differential_abundance_or_expression_analysis/regression_output_nasal_merged_gene_metag.tsv",sep=' ') %>% mutate(seqtype = 'METAGENOMICS') %>% mutate(regressionclass = 'Nasal') %>% filter(grepl('Timepoint',term))   %>% mutate(dset = 'GENE-CATALOG')
regdata[[44]] = read.csv("i4_data_packet/differential_abundance_or_expression_analysis/regression_output_nasal_merged_gene_metat.tsv",sep=' ') %>% mutate(seqtype = 'METATRANSCRIPTOMICS') %>% mutate(regressionclass = 'Nasal') %>% filter(grepl('Timepoint',term)) %>% mutate(dset = 'GENE-CATALOG')
regdata[[45]] = read.csv("i4_data_packet/differential_abundance_or_expression_analysis/regression_output_skin_merged_gene_metag.tsv",sep=' ') %>% mutate(seqtype = 'METAGENOMICS') %>% mutate(regressionclass = 'Skin') %>% filter(grepl('Timepoint',term)) %>% mutate(dset = 'GENE-CATALOG')
regdata[[46]] = read.csv("i4_data_packet/differential_abundance_or_expression_analysis/regression_output_skin_merged_gene_metat.tsv",sep=' ') %>% mutate(seqtype = 'METATRANSCRIPTOMICS') %>% mutate(regressionclass = 'Skin') %>% filter(grepl('Timepoint',term)) %>% mutate(dset = 'GENE-CATALOG')
regdata[[47]] = read.csv("i4_data_packet/differential_abundance_or_expression_analysis/regression_output_skinseparates_merged_gene_metag.tsv",sep=' ') %>% mutate(seqtype = 'METAGENOMICS') %>% mutate(regressionclass = 'Skin_Components') %>% filter(grepl('Timepoint',term)) %>% mutate(dset = 'GENE-CATALOG')
regdata[[48]] = read.csv("i4_data_packet/differential_abundance_or_expression_analysis/regression_output_skinseparates_merged_gene_metat.tsv",sep=' ') %>% mutate(seqtype = 'METATRANSCRIPTOMICS') %>% mutate(regressionclass = 'Skin_Components') %>% filter(grepl('Timepoint',term)) %>% mutate(dset = 'GENE-CATALOG')

### OVERALL
regdata[[49]] = read.table("i4_data_packet/differential_abundance_or_expression_analysis/regression_output_overall_bacterial_dna_species_xtree_.05_.0025_GTDB_r207.tsv",sep='\t') %>% mutate(seqtype = 'METAGENOMICS') %>% mutate(regressionclass = 'Overall') %>% filter(grepl('Timepoint',term))   %>% mutate(dset = 'GTDB')%>% mutate(term = if_else(term == 'Timepoint_RecodePRE-LAUNCH','PRE-LAUNCH --- OVERALL',if_else(term == 'Timepoint_RecodePOST-LAUNCH','POST-LAUNCH --- OVERALL',term)))
regdata[[50]] = read.table("i4_data_packet/differential_abundance_or_expression_analysis/regression_output_overall_bacterial_rna_species_xtree_.05_.0025_GTDB_r207.tsv",sep='\t') %>% mutate(seqtype = 'METATRANSCRIPTOMICS') %>% mutate(regressionclass = 'Overall') %>% filter(grepl('Timepoint',term)) %>% mutate(dset = 'GTDB')%>% mutate(term = if_else(term == 'Timepoint_RecodePRE-LAUNCH','PRE-LAUNCH --- OVERALL',if_else(term == 'Timepoint_RecodePOST-LAUNCH','POST-LAUNCH --- OVERALL',term)))
regdata[[51]] = read.table("i4_data_packet/differential_abundance_or_expression_analysis/regression_output_overall_bacterial_dna_species_xtree_.05_.0025_ASSEMBLIES.tsv",sep='\t') %>% mutate(seqtype = 'METAGENOMICS') %>% mutate(regressionclass = 'Overall') %>% filter(grepl('Timepoint',term)) %>% mutate(dset = 'ASSEMBLED-BACTERIA')%>% mutate(term = if_else(term == 'Timepoint_RecodePRE-LAUNCH','PRE-LAUNCH --- OVERALL',if_else(term == 'Timepoint_RecodePOST-LAUNCH','POST-LAUNCH --- OVERALL',term)))
regdata[[52]] = read.table("i4_data_packet/differential_abundance_or_expression_analysis/regression_output_overall_bacterial_rna_species_xtree_.05_.0025_ASSEMBLIES.tsv",sep='\t') %>% mutate(seqtype = 'METATRANSCRIPTOMICS') %>% mutate(regressionclass = 'Overall') %>% filter(grepl('Timepoint',term)) %>% mutate(dset = 'ASSEMBLED-BACTERIA')%>% mutate(term = if_else(term == 'Timepoint_RecodePRE-LAUNCH','PRE-LAUNCH --- OVERALL',if_else(term == 'Timepoint_RecodePOST-LAUNCH','POST-LAUNCH --- OVERALL',term)))
regdata[[53]] = read.csv("i4_data_packet/differential_abundance_or_expression_analysis/regression_output_overall_viral_dna_species_xtree_.01_.0.005_VIRAL_GENBANK.tsv",sep='\t') %>% mutate(seqtype = 'METAGENOMICS') %>% mutate(regressionclass = 'Overall') %>% filter(grepl('Timepoint',term)) %>% mutate(dset = 'GENBANK-VIRUSES')%>% mutate(term = if_else(term == 'Timepoint_RecodePRE-LAUNCH','PRE-LAUNCH --- OVERALL',if_else(term == 'Timepoint_RecodePOST-LAUNCH','POST-LAUNCH --- OVERALL',term)))
regdata[[54]] = read.csv("i4_data_packet/differential_abundance_or_expression_analysis/regression_output_overall_viral_rna_species_xtree_.01_.0.005_VIRAL_GENBANK.tsv",sep='\t') %>% mutate(seqtype = 'METATRANSCRIPTOMICS') %>% mutate(regressionclass = 'Overall') %>% filter(grepl('Timepoint',term)) %>% mutate(dset = 'GENBANK-VIRUSES')%>% mutate(term = if_else(term == 'Timepoint_RecodePRE-LAUNCH','PRE-LAUNCH --- OVERALL',if_else(term == 'Timepoint_RecodePOST-LAUNCH','POST-LAUNCH --- OVERALL',term)))
regdata[[55]] = read.csv("i4_data_packet/differential_abundance_or_expression_analysis/regression_output_overall_viral_dna_species_xtree_.01_.0.005_ASSEMBLIES.tsv",sep='\t') %>% mutate(seqtype = 'METAGENOMICS') %>% mutate(regressionclass = 'Overall') %>% filter(grepl('Timepoint',term)) %>% mutate(dset = 'ASSEMBLED-VIRUSES')%>% mutate(term = if_else(term == 'Timepoint_RecodePRE-LAUNCH','PRE-LAUNCH --- OVERALL',if_else(term == 'Timepoint_RecodePOST-LAUNCH','POST-LAUNCH --- OVERALL',term)))
regdata[[56]] = read.csv("i4_data_packet/differential_abundance_or_expression_analysis/regression_output_overall_viral_rna_species_xtree_.01_.0.005_ASSEMBLIES.tsv",sep='\t') %>% mutate(seqtype = 'METATRANSCRIPTOMICS') %>% mutate(regressionclass = 'Overall') %>% filter(grepl('Timepoint',term)) %>% mutate(dset = 'ASSEMBLED-VIRUSES')%>% mutate(term = if_else(term == 'Timepoint_RecodePRE-LAUNCH','PRE-LAUNCH --- OVERALL',if_else(term == 'Timepoint_RecodePOST-LAUNCH','POST-LAUNCH --- OVERALL',term)))
regdata[[57]] = read.csv("i4_data_packet/differential_abundance_or_expression_analysis/regression_output_overall_bacterial-viral_dna_species_kraken2_none_KRAKEN2_REFSEQDB.tsv",sep='\t') %>% mutate(seqtype = 'METAGENOMICS') %>% mutate(regressionclass = 'Overall') %>% filter(grepl('Timepoint',term)) %>% mutate(dset = 'KRAKEN2')%>% mutate(term = if_else(term == 'Timepoint_RecodePRE-LAUNCH','PRE-LAUNCH --- OVERALL',if_else(term == 'Timepoint_RecodePOST-LAUNCH','POST-LAUNCH --- OVERALL',term)))
regdata[[58]] = read.csv("i4_data_packet/differential_abundance_or_expression_analysis/regression_output_overall_bacterial-viral_rna_species_kraken2_none_KRAKEN2_REFSEQDB.tsv",sep='\t') %>% mutate(seqtype = 'METATRANSCRIPTOMICS') %>% mutate(regressionclass = 'Overall') %>% filter(grepl('Timepoint',term)) %>% mutate(dset = 'KRAKEN2') %>% mutate(term = if_else(term == 'Timepoint_RecodePRE-LAUNCH','PRE-LAUNCH --- OVERALL',if_else(term == 'Timepoint_RecodePOST-LAUNCH','POST-LAUNCH --- OVERALL',term)))
regdata[[59]] = read.csv("i4_data_packet/differential_abundance_or_expression_analysis/regression_output_overall_merged_gene_metag.tsv",sep=' ') %>% mutate(seqtype = 'METAGENOMICS') %>% mutate(regressionclass = 'Overall') %>% filter(grepl('Timepoint',term)) %>% mutate(dset = 'GENE-CATALOG')
regdata[[60]] = read.csv("i4_data_packet/differential_abundance_or_expression_analysis/regression_output_overall_merged_gene_metat.tsv",sep=' ') %>% mutate(seqtype = 'METATRANSCRIPTOMICS') %>% mutate(regressionclass = 'Overall') %>% filter(grepl('Timepoint',term)) %>% mutate(dset = 'GENE-CATALOG')

regression_output = bind_rows(regdata)

regression_output$term = gsub('Timepoint_Recode','',regression_output$term)
regression_output$term = gsub(':','',regression_output$term)
regression_output$term = gsub(':Body.Location','',regression_output$term)
regression_output$term = gsub('PRE-LAUNCH','PRE-LAUNCH --- ',regression_output$term)
regression_output$term = gsub('POST-LAUNCH','POST-LAUNCH --- ',regression_output$term)
regression_output$term = gsub('isskin','Skin',regression_output$term)
regression_output$term = gsub('isoral','Oral',regression_output$term)
regression_output$term = gsub('nape','Nape of Neck',regression_output$term)
regression_output$term = gsub('postauric','Post-Auricular',regression_output$term)
regression_output$term = gsub('bb','Belly Button',regression_output$term)
regression_output$term = gsub('gc','Gluteal Crease',regression_output$term)
regression_output$term = gsub('Tzone','T-Zone',regression_output$term)
regression_output$term = gsub('web','Toe Web Space',regression_output$term)
regression_output$term = gsub('fore','Forearm',regression_output$term)
regression_output$term = gsub(' ---  --- ',' --- ',regression_output$term)
regression_output$term = gsub('isnasal','Nasal',regression_output$term)
regression_output = regression_output %>% mutate(term = if_else(term == 'PRE-LAUNCH --- ','PRE-LAUNCH',term,))
regression_output = regression_output %>% mutate(term = if_else(term == 'POST-LAUNCH --- ','POST-LAUNCH',term,))
regression_output$term = factor(regression_output$term,levels = c("PRE-LAUNCH --- OVERALL","POST-LAUNCH --- OVERALL","PRE-LAUNCH","POST-LAUNCH","PRE-LAUNCH --- Armpit","PRE-LAUNCH --- Belly Button","PRE-LAUNCH --- Forearm", "PRE-LAUNCH --- Gluteal Crease","PRE-LAUNCH --- Nape of Neck","PRE-LAUNCH --- Nasal","PRE-LAUNCH --- Post-Auricular","PRE-LAUNCH --- T-Zone","PRE-LAUNCH --- Toe Web Space","PRE-LAUNCH --- Skin","PRE-LAUNCH --- Oral","POST-LAUNCH --- Armpit","POST-LAUNCH --- Belly Button", "POST-LAUNCH --- Forearm","POST-LAUNCH --- Gluteal Crease" ,"POST-LAUNCH --- Nape of Neck", "POST-LAUNCH --- Nasal", "POST-LAUNCH --- Post-Auricular","POST-LAUNCH --- T-Zone", "POST-LAUNCH --- Toe Web Space","POST-LAUNCH --- Skin","POST-LAUNCH --- Oral"))

regression_output = regression_output %>% mutate(launch = if_else(grepl('PRE',term),'PRE-LAUNCH','POST-LAUNCH')) %>% mutate(direction = if_else(estimate>0,'positive','negative')) %>% filter(term != 'POST-LAUNCH --- OVERALL')%>% filter(term != 'PRE-LAUNCH --- OVERALL')

a =regression_output %>% filter(launch == 'PRE-LAUNCH') %>% filter(grepl('---',term)) %>% mutate(regclass2 = strsplit(as.character(term),' --- ') %>% map_chr(2))  %>% dplyr::rename(PRE=launch,adj_pre = BH_adjusted,est_pre = estimate) %>% select(yvar,dset,seqtype,regclass2,PRE,adj_pre,est_pre)
#a2 =regression_output %>% filter(launch == 'PRE-LAUNCH')%>% filter(term == 'PRE-LAUNCH --- OVERALL') %>% mutate(regclass2 = 'Overall') %>% mutate(mergeterm = paste(yvar,dset,seqtype,regclass2)) %>% dplyr::rename(PRE=launch,adj_pre = BH_adjusted,est_pre = estimate) %>% select(yvar,dset,seqtype,regclass2,PRE,adj_pre,est_pre)
a3 =regression_output %>% filter(regressionclass == 'Overall') %>% filter(launch == 'PRE-LAUNCH')%>% filter(term == 'PRE-LAUNCH') %>% mutate(regclass2 = 'Overall') %>% mutate(mergeterm = paste(yvar,dset,seqtype,regclass2)) %>% dplyr::rename(PRE=launch,adj_pre = BH_adjusted,est_pre = estimate) %>% select(yvar,dset,seqtype,regclass2,PRE,adj_pre,est_pre)


a = bind_rows(a,a3)

b =regression_output %>% filter(launch == 'POST-LAUNCH') %>% filter(grepl('---',term)) %>% mutate(regclass2 = strsplit(as.character(term),' --- ') %>% map_chr(2))%>% dplyr::rename(POST=launch,adj_post = BH_adjusted,est_post = estimate) %>% select(yvar,dset,seqtype,regclass2,POST,adj_post,est_post)
#b2 =regression_output %>% filter(launch == 'POST-LAUNCH') %>% filter(term == 'POST-LAUNCH --- OVERALL') %>% mutate(regclass2 = 'Overall') %>% mutate(mergeterm = paste(yvar,dset,seqtype,regclass2)) %>% dplyr::rename(POST=launch,adj_post = BH_adjusted,est_post = estimate) %>% select(yvar,dset,seqtype,regclass2,POST,adj_post,est_post)
b3 =regression_output %>% filter(regressionclass == 'Overall')%>% filter(launch == 'POST-LAUNCH') %>% filter(term == 'POST-LAUNCH') %>% mutate(regclass2 = 'Overall') %>% mutate(mergeterm = paste(yvar,dset,seqtype,regclass2)) %>% dplyr::rename(POST=launch,adj_post = BH_adjusted,est_post = estimate) %>% select(yvar,dset,seqtype,regclass2,POST,adj_post,est_post)
b = bind_rows(b,b3)

c = inner_join(a,b)

c = c %>% filter(adj_pre<0.05 |  adj_post<0.05)

### MODERATE CONSERVATISM
#c = c %>% mutate(timetrend = 'NONE')%>% mutate(timetrend = if_else(est_pre<0 & est_post>0 & adj_pre <0.05 |est_pre<0 & est_post>0 & adj_post <0.05,'Increased in/after flight',timetrend)) %>% mutate(timetrend = if_else(est_pre>0 & est_post<0 & adj_pre <0.05 | est_pre>0 & est_post<0 & adj_post <0.05,'Decreased in/after flight',timetrend)) %>% mutate(timetrend = if_else(est_pre<0 & est_post<0 & adj_pre<0.05 | est_pre<0 & est_post<0 & adj_post <0.05,'Increased in flight',timetrend)) %>% mutate(timetrend = if_else(est_pre>0 & est_post>0 & adj_pre<0.05 | est_pre>0 & est_post>0  & adj_post <0.05,'Decreased in flight',timetrend)) 

### HIGH CONSERVATISM
#c = c %>% mutate(timetrend = 'NONE')%>% mutate(timetrend = if_else(est_pre<0 & est_post>0 & adj_pre <0.05 |est_pre<0 & est_post>0 & adj_post <0.05,'Increased in/after flight',timetrend)) %>% mutate(timetrend = if_else(est_pre>0 & est_post<0 & adj_pre <0.05 | est_pre>0 & est_post<0 & adj_post <0.05,'Decreased in/after flight',timetrend)) %>% mutate(timetrend = if_else(est_pre<0 & est_post<0 & adj_pre<0.05 & adj_post <0.05,'Increased in flight',timetrend)) %>% mutate(timetrend = if_else(est_pre>0 & est_post>0 & adj_pre<0.05 & adj_post <0.05,'Decreased in flight',timetrend)) 

### BOTH
c = c %>% mutate(timetrendrank = 'NONE')%>% mutate(timetrendrank = if_else(est_pre<0 & est_post>0 & adj_pre <0.05 |est_pre<0 & est_post>0 & adj_post <0.05,'Increased in/after flight',timetrendrank)) %>% mutate(timetrendrank = if_else(est_pre>0 & est_post<0 & adj_pre <0.05 | est_pre>0 & est_post<0 & adj_post <0.05,'Decreased in/after flight',timetrendrank)) %>% mutate(timetrendrank = if_else(est_pre<0 & est_post<0 & adj_pre<0.05 | est_pre<0 & est_post<0 & adj_post <0.05,'Increased in flight -- LP',timetrendrank)) %>% mutate(timetrendrank = if_else(est_pre>0 & est_post>0 & adj_pre<0.05 | est_pre>0 & est_post>0  & adj_post <0.05,'Decreased in flight -- LP',timetrendrank)) %>% mutate(timetrendrank = if_else(est_pre<0 & est_post<0 & adj_pre<0.05 & adj_post <0.05,'Increased in flight',timetrendrank)) %>% mutate(timetrendrank = if_else(est_pre>0 & est_post>0 & adj_pre<0.05 & adj_post <0.05,'Decreased in flight',timetrendrank)) 
c = c %>% mutate(timetrend = gsub(' -- LP','',timetrendrank))

fortrendline = c %>% filter(regclass2 != 'OVERALL') %>% mutate(mergeid = paste(yvar,dset,seqtype,regclass2)) 
```
######## SPLINES AND SUMMARY PLOTS
```{r}
### GENERATE SUMMARY PLOTS

setwd('~/Dropbox (Mason Lab)/i4/')

simple_splines <- function(dataset,metagdata,metatdata,fortrendline){
	temp_metag = fortrendline %>% filter(dset == dataset,seqtype == 'METAGENOMICS',regclass2 != 'Skin')
	temp_metat = fortrendline %>% filter(dset == dataset,seqtype == 'METATRANSCRIPTOMICS',regclass2 != 'Skin')

	temp2_metag_sub = metagdata %>% select(any_of(temp_metag$yvar)) %>% t %>% as.data.frame(check.names = F) %>% rownames_to_column('yvar') %>% melt
	temp_metag = inner_join(temp_metag,temp2_metag_sub)
	temp_metag = left_join(temp_metag,meta %>% rename(variable = SeqID))

	temp2_metat_sub = metatdata %>% select(any_of(temp_metat$yvar)) %>% t %>% as.data.frame(check.names = F) %>% rownames_to_column('yvar') %>% melt
	temp_metat = inner_join(temp_metat,temp2_metat_sub)
	temp_metat = left_join(temp_metat,meta %>% rename(variable = SeqID))

	temp = bind_rows(temp_metag,temp_metat)
	temp = temp %>% mutate(mergeid = paste(yvar,dset,seqtype,location))
	temp = temp %>% filter(mergeid %in% fortrendline$mergeid)

	## compute log2fc -- BEFORE/AFTER, BEFORE MID, AFTER
	l2fc = temp %>% select(yvar,Timepoint_Recode,location,seqtype,dset,value) %>% group_by(yvar,Timepoint_Recode,location,seqtype,dset) %>% filter(!is.na(Timepoint_Recode))%>% summarise(mean = mean(value)) %>% dcast(yvar+location+seqtype+dset~Timepoint_Recode,value.var='mean')
	l2fc[is.na(l2fc)]=0
	l2fc$PRE_MID = log2((l2fc$`PRE-LAUNCH`+0.0000001)/(l2fc$`MID-FLIGHT`+0.0000001))
	l2fc$POST_MID = log2((l2fc$`POST-LAUNCH`+0.0000001)/(l2fc$`MID-FLIGHT`+0.0000001))
	l2fc$POST_PRE = log2((l2fc$`POST-LAUNCH`+0.0000001)/(l2fc$`PRE-LAUNCH`+0.0000001))

	q1=quantile(abs(l2fc$PRE_MID),na.rm=T,.25)[[1]]
	q2=quantile(abs(l2fc$POST_MID),na.rm=T,.25)[[1]]
	q3=quantile(abs(l2fc$POST_PRE),na.rm=T,.25)[[1]]

	#l2fc = l2fc %>% filter(PRE_MID > q1 | PRE_MID > q2 |PRE_MID > q3)

	l2fc = l2fc %>% select(-all_of(c("MID-FLIGHT","PRE-LAUNCH","POST-LAUNCH"))) # %>% melt(id.vars = c('yvar','location','seqtype','dset'))
	colnames(l2fc) = c('yvar','location','seqtype','dset','PRE_MID','POST_MID','POST_PRE')
	temp = inner_join(temp,l2fc)

	plot = ggplot(temp,aes(x = Timepoint_Numeric,color = seqtype,y=log(value+0.000000001))) + geom_smooth(method = 'loess',alpha=.7)  + facet_grid( timetrend~regclass2 ,scales = 'free_y') + theme_bw() + scale_color_brewer(palette= 2)
	return(list(temp %>% mutate(PLOTTYPE = dataset) %>% filter(regclass2 == location),plot))
}

fullcounts = fortrendline %>% filter(dset == 'GTDB' | dset == 'GENBANK-VIRUSES' | dset == 'GENE-CATALOG', regclass2 != 'Skin') %>% select(dset,seqtype,timetrend) %>% group_by(dset,seqtype,timetrend) %>% count
fullcounts$dset = factor(fullcounts$dset,levels = c('ASSEMBLED-BACTERIA','GTDB','KRAKEN2','ASSEMBLED-VIRUSES','GENBANK-VIRUSES','GENE-CATALOG'))

ggplot(fullcounts,aes(x = timetrend, y = n, fill =seqtype)) + theme_bw() + geom_bar(stat='identity',position='dodge') + facet_grid(rows = vars(dset),scales='free_y') + ylab('Number of significant features') +scale_fill_manual(values = c('darkblue','#BE1E2D'))+ theme(legend.position = 'None') + theme(axis.text.x  = element_text(angle=60,hjust=1))
ggsave('~/Dropbox (Mason Lab)/i4/regressionsummary/overall_summarycounts.pdf',width=2,height=5)

summaryplotfodder = fortrendline %>% select(dset,seqtype,regclass2,timetrend) %>% group_by(dset,seqtype,regclass2,timetrend) %>% count

summaryplotfodder$dset = factor(summaryplotfodder$dset,levels = c('ASSEMBLED-BACTERIA','GTDB','KRAKEN2','ASSEMBLED-VIRUSES','GENBANK-VIRUSES','GENE-CATALOG'))
summaryplotfodder$regclass2 = factor(summaryplotfodder$regclass2,levels = c('Overall','Oral','Nasal','Skin','Armpit','Belly Button','Forearm','Gluteal Crease','Nape of Neck','Post-Auricular','T-Zone','Toe Web Space'))

## SUPPLEMENT -- ALL RESULTS 
ggplot(summaryplotfodder,aes(y = timetrend, x = n, fill =seqtype)) + theme_bw() + geom_bar(stat='identity',position='dodge') + scale_x_log10()+ facet_grid(rows = vars(dset), cols = vars(regclass2)) + ylab('') + xlab('')+scale_fill_manual(values = c('darkblue','#BE1E2D'))
ggsave('~/Dropbox (Mason Lab)/i4/regressionsummary/all_counts_by_type_supp.pdf',width=16,height=9)

## MAIN -- PRIMARY RESULTS AND TOTAL INCREASE

ggplot(summaryplotfodder %>% filter(dset == 'GTDB' | dset == 'GENBANK-VIRUSES' | dset == 'GENE-CATALOG', regclass2 != 'Skin' ,regclass2 != 'Overall'),aes(y = timetrend, x = n, fill =seqtype)) + theme_bw() + geom_bar(stat='identity',position='dodge') +scale_x_log10() + facet_grid(rows = vars(dset), cols = vars(regclass2)) + ylab('') + xlab('log10(Number of significant hits)')+scale_fill_manual(values = c('darkblue','#BE1E2D'))+ theme(legend.position = 'None')
ggsave('~/Dropbox (Mason Lab)/i4/regressionsummary/sig_counts_by_type.pdf',width=12,height=4)

### find trajectory types
####### TIGHTEN UP P VALUE RESTRICTIONS TO MAKE SPLINES TRACTABLE

fortrendline2 =  fortrendline %>%filter(regclass2 != 'OVERALL',!grepl('LP',timetrendrank)) %>% mutate(mergeid = paste(yvar,dset,seqtype,regclass2)) 

gtdb_traj = simple_splines('GTDB',gtdb_metag,gtdb_metat,fortrendline2)
#kraken_traj = simple_splines('KRAKEN2',kraken_metag,kraken_metat,fortrendline2)
#bacass_traj = simple_splines('ASSEMBLED-BACTERIA',bass_metag,bass_metat %>% t %>% as.data.frame(check.names=F),fortrendline2)
#virass_traj = simple_splines('ASSEMBLED-VIRUSES',viral_ass_metag,viral_ass_metat %>% t %>% as.data.frame(check.names=F),fortrendline2)
genvir_traj = simple_splines('GENBANK-VIRUSES',viral_metag,viral_metat %>% t %>% as.data.frame(check.names=F),fortrendline2)
gene_traj = simple_splines('GENE-CATALOG',gene_metag,gene_metat,fortrendline2)

gtdb_genvir = bind_rows(gtdb_traj[[1]],genvir_traj[[1]]) #%>% filter(timetrend == 'Increased in flight' | timetrend == 'Decreased in flight')
#gtdb_genvir = gtdb_genvir %>% filter(yvar %in% (gtdb_genvir %>% filter(value!=0)%>% select(Crew.ID,yvar) %>% unique %>% group_by(yvar) %>% count %>% filter(n>1) %>% select(yvar) %>% unlist %>% unname))
temp = gene_traj[[1]] #%>% filter(timetrend == 'Increased in flight' | timetrend == 'Decreased in flight')
#temp = temp %>% filter(yvar %in% (temp %>% filter(value!=0)%>% select(Crew.ID,yvar) %>% unique %>% group_by(yvar) %>% csount %>% filter(n>1) %>% select(yvar) %>% unlist %>% unname))
temp1 =temp %>% group_by(timetrend,regclass2) %>% slice_max(abs(est_pre),n=100)
temp2 =temp %>% group_by(timetrend,regclass2) %>% slice_max(abs(est_post),n=100)
gene_traj[[1]] = bind_rows(temp1,temp2)
gtdb_genvir_gene = bind_rows(gene_traj[[1]],gtdb_genvir)

gtdb_genvir_gene$regclass2 = factor(gtdb_genvir_gene$regclass2,levels = c('Overall','Oral','Nasal','Skin','Armpit','Belly Button','Forearm','Gluteal Crease','Nape of Neck','Post-Auricular','T-Zone','Toe Web Space'))

gtdb_genvir_gene$PLOTTYPE[gtdb_genvir_gene$PLOTTYPE == 'GENE-CATALOG'] = 'Genes'
gtdb_genvir_gene$PLOTTYPE[gtdb_genvir_gene$PLOTTYPE == 'GTDB'] = 'Bacteria/Archaea'
gtdb_genvir_gene$PLOTTYPE[gtdb_genvir_gene$PLOTTYPE == 'GENBANK-VIRUSES'] = 'Viruses'

gtdb_genvir_gene$PLOTTYPE= factor(gtdb_genvir_gene$PLOTTYPE,levels = c('Bacteria/Archaea','Viruses','Genes'))

gtdb_genvir_gene =  gtdb_genvir_gene %>% mutate(regressionclass = if_else(regclass2 != 'Oral' & regclass2 != 'Nasal','Skin',as.character(regclass2)))

plot1 = ggplot(gtdb_genvir_gene %>% filter(timetrend == 'Increased in flight' | timetrend == 'Decreased in flight'),aes(x = Timepoint_Numeric,linetype = timetrend,color = seqtype,y=log(value+0.000001))) + geom_smooth(method = 'loess',alpha=.4,level=.95,position='jitter')  + facet_grid(regclass2~PLOTTYPE ,scales = 'free_y') + theme_bw() +scale_color_manual(values = c('darkblue','#BE1E2D')) + theme(legend.position = 'None')
ggsave(plot=plot1,'regressionsummary/inc_dec_spline.pdf',width=4,height=13)

plot2 = ggplot(gtdb_genvir_gene%>% filter(timetrend != 'Increased in flight' & timetrend != 'Decreased in flight',timetrend !='NONE'),aes(x = Timepoint_Numeric,linetype = timetrend,color = seqtype,y=log(value+0.000001))) + geom_smooth(method = 'loess',alpha=.4,level=.95,position='jitter')  + facet_grid(regclass2~PLOTTYPE ,scales = 'free_y') + theme_bw() +scale_color_manual(values = c('darkblue','#BE1E2D')) + theme(legend.position = 'None')
ggsave(plot=plot2,'regressionsummary/other_spline.pdf',width=4,height=13)

```
######## UPSETR PLOTS
```{r}
### SUMMARY UPSET R PLOTS -- ALSO GENERATE BETWEEN DATA TYPES FOR SUPPLEMENT

# overall

check_argument = function(
    value,
    allowed,
    description
) {
    if (!(value %in% allowed)) {
        stop(
            paste0(
                description,
                ' has to be one of: ',
                paste(allowed, collapse=' or '),
                ', not "',
                value,
                '"'

            )
        )
    }
}


solve_mode = function (mode) {
  check_argument(
      mode,
      allowed = c(
          'exclusive_intersection', 'distinct',
          'inclusive_intersection', 'intersect',
          'exclusive_union', # no alias
          'inclusive_union', 'union'
      ),
      'mode'
  )

    # resolve aliases
    mode = switch(
        mode,
        distinct='exclusive_intersection',
        intersect='inclusive_intersection',
        union='inclusive_union',
        mode
    )
}


get_mode_presence = function(mode, prefix='in_', symbol=TRUE) {
    column = paste0(prefix, solve_mode(mode))
    if (symbol) {
        sym(column)
    } else {
        column
    }
}

presence = get_mode_presence('exclusive_intersection')

summarise_values = function(df) {
    aggregate(
        as.formula(paste0(presence, '~ intersection')),
        df,
        FUN=sum
    )
}

overall_intersections = regression_output %>% filter(BH_adjusted<0.05) %>% filter(regressionclass != 'Skin')  %>% filter(grepl('---',term)) %>% mutate(regclass2 = strsplit(as.character(term),' --- ') %>% map_chr(2)) %>% mutate(seqval =1 ) %>% dcast(yvar + direction+ launch  + dset + seqtype ~ regclass2,value.var ='seqval') 

overall_intersections = overall_intersections %>% mutate_if(is.numeric, ~1 * (. != 0))

overall_intersections[is.na(overall_intersections)]=0

overall_intersections2 = overall_intersections %>% filter(dset == 'GTDB') %>% distinct
pdf('~/Dropbox (Mason Lab)/i4/regressionsummary/gtdb_upset.pdf',width=5,height=4)
upset(overall_intersections2,set_sizes = FALSE,min_size=2,intersect = colnames(overall_intersections2)[6:ncol(overall_intersections2)],base_annotations=list('Intersection size'=intersection_size(counts=FALSE,mapping=aes(fill=seqtype))+theme(legend.position = 'None')+scale_fill_manual(values = c('darkblue','#BE1E2D'))),width_ratio=0.1)
dev.off()

overall_intersections2 = overall_intersections %>% filter(dset == 'GENBANK-VIRUSES') %>% distinct

pdf('~/Dropbox (Mason Lab)/i4/regressionsummary/genvir_upset.pdf',width=5,height=4)
upset(overall_intersections2,set_sizes = FALSE,min_size=2,intersect = colnames(overall_intersections2)[6:ncol(overall_intersections2)],base_annotations=list('Intersection size'=intersection_size(counts=FALSE,mapping=aes(fill=seqtype))+theme(legend.position = 'None')+scale_fill_manual(values = c('darkblue','#BE1E2D'))),width_ratio=0.1)
dev.off()

overall_intersections2 = overall_intersections %>% filter(dset == 'ASSEMBLED-BACTERIA') %>% distinct

pdf('~/Dropbox (Mason Lab)/i4/regressionsummary/bacass_upset.pdf',width=5,height=4)
upset(overall_intersections2,set_sizes = FALSE,min_size=2,intersect = colnames(overall_intersections2)[6:ncol(overall_intersections2)],base_annotations=list('Intersection size'=intersection_size(counts=FALSE,mapping=aes(fill=seqtype))+theme(legend.position = 'None')+scale_fill_manual(values = c('darkblue','#BE1E2D'))),width_ratio=0.1)
dev.off()

overall_intersections2 = overall_intersections %>% filter(dset == 'ASSEMBLED-VIRUSES') %>% distinct

pdf('~/Dropbox (Mason Lab)/i4/regressionsummary/virass_upset.pdf',width=5,height=4)
upset(overall_intersections2,set_sizes = FALSE,min_size=2,intersect = colnames(overall_intersections2)[6:ncol(overall_intersections2)],base_annotations=list('Intersection size'=intersection_size(counts=FALSE,mapping=aes(fill=seqtype))+theme(legend.position = 'None')+scale_fill_manual(values = c('darkblue','#BE1E2D'))),width_ratio=0.1)
dev.off()


overall_intersections2 = overall_intersections %>% filter(dset == 'GENE-CATALOG') %>% distinct
### NOTE IN METHODS VARIATION IN INTERSECTION SIZE

pdf('~/Dropbox (Mason Lab)/i4/regressionsummary/genecat_upset.pdf',width=5,height=4)
upset(overall_intersections2,set_sizes = FALSE,min_size=50,intersect = colnames(overall_intersections2)[6:ncol(overall_intersections2)],base_annotations=list('Intersection size'=intersection_size(counts=FALSE,mapping=aes(fill=seqtype))+theme(legend.position = 'None')+scale_fill_manual(values = c('darkblue','#BE1E2D'))),width_ratio=0.1)
dev.off()

overall_intersections2 = overall_intersections %>% filter(dset == 'KRAKEN2') %>% distinct

pdf('~/Dropbox (Mason Lab)/i4/regressionsummary/kraken2_upset.pdf',width=5,height=4)
upset(overall_intersections2,set_sizes = FALSE,min_size=2,intersect = colnames(overall_intersections2)[6:ncol(overall_intersections2)],base_annotations=list('Intersection size'=intersection_size(counts=FALSE,mapping=aes(fill=seqtype))+theme(legend.position = 'None')+scale_fill_manual(values = c('darkblue','#BE1E2D'))),width_ratio=0.1)
dev.off()


```
###SPECIES-LEVEL SHIFTS BY REGRESSION
```{r}

categories = c %>% filter(regclass2 != 'OVERALL' &regclass2 != 'Overall') %>% select(yvar,seqtype,dset,timetrend,timetrendrank,regclass2,adj_pre,est_pre,adj_post,est_post)%>% filter(regclass2 != 'Skin') %>% mutate(regressionclass = if_else(regclass2 == 'Nasal' | regclass2 == 'Oral',regclass2,'Skin_Components'))

generate_fc_heatmap <- function(regression_output,dataset,metagdata,metatdata,regclass,categories,taxdata){
  output=list()
  minval = 0.0000000001
  	temp_metat = regression_output %>% filter(BH_adjusted<0.05,dset == dataset,seqtype == 'METATRANSCRIPTOMICS',regressionclass == regclass)%>% filter(grepl('---',term)) %>% mutate(regclass2 = strsplit(as.character(term),' --- ') %>% map_chr(2))
  	temp_metat = inner_join(temp_metat,categories) %>% select(timetrend,regressionclass,yvar,dset,regclass2) %>% distinct
  	colormapping = setNames(as.list(c(brewer.pal(n=11,'Set3'),brewer.pal(n=11,'Set3'),'gray',brewer.pal(n=8,'Set1'),brewer.pal(n=3,'Paired')[[1]],brewer.pal(n=3,'Paired')[[2]])), c('Actinobacteriota','Bacteroidota','Firmicutes','Firmicutes_A','Firmicutes_C','Fusobacteriota','Patescibacteria','Cyanobacteria','Deinococcota','Proteobacteria','Verrucomicrobiota','Cossaviricota','Cressdnaviricota','Duplornaviricota','Kitrinoviricota','Negarnaviricota','Phixviricota','Pisuviricota','Uroviricota','Lenarviricota','Nucleocytoviricota','Artverviricota','Other/Unclassified','Actinobacteria','Euryarchaeota','Actinobacteria','Crenarchaeota','Peploviricota','Chlamydiae','Candidatus Micrarchaeota','Hofneiviricota','Phy_of_Pospiviroidae','Phy_of_Tolecusatellitidae'))

      	 	if(nrow(temp_metat)>0){
      print('Generating metatranscriptomic heatmap')

    	temp2_metat_sub = metatdata %>% select(any_of(temp_metat$yvar)) %>% t %>% as.data.frame(check.names = F) %>% rownames_to_column('yvar') %>% melt
    	temp_metat = inner_join(temp_metat,temp2_metat_sub)
    	temp_metat = left_join(temp_metat,meta %>% rename(variable = SeqID)) %>% filter(!is.na(Timepoint))
    
    	# generate l2fc heatmap
  	l2fc_hm = temp_metat %>% filter(regclass2 == location)%>% group_by(yvar,Timepoint) %>% summarise(mean = mean(value)) %>% dcast(yvar ~ Timepoint,value.var ='mean') %>% mutate(mean_flight_abundance = (`Flight 1` + `Flight 2`)/2) %>% mutate(l2fc_PRE_june = log2((mean_flight_abundance+minval)) - log2(`21-Jun`+minval)) %>% mutate(l2fc_PRE_aug = log2((mean_flight_abundance+minval)) - log2(`21-Aug`+minval)) %>% mutate(l2fc_PRE_sept = log2((mean_flight_abundance+minval)) - log2(`Sept Pre-Launch`+minval)) %>% mutate(l2fcMID = 0) %>% mutate(l2fc_POST_sept = -1*(log2((mean_flight_abundance+minval)) - log2(`Sept Post-Return`+minval))) %>% mutate(l2fc_POST_nov = -1*(log2((mean_flight_abundance+minval)) - log2(`November`+minval))) %>% mutate(l2fc_POST_dec = -1*(log2((mean_flight_abundance+minval)) - log2(`21-Dec`+minval))) %>% select(-`Flight 1`,-`Flight 2`) %>% column_to_rownames("yvar") %>% t %>% as.data.frame(check.names=F)
  	l2fc_hm = l2fc_hm[grepl('l2fc',rownames(l2fc_hm)),,drop=F]
    l2fc_hm = l2fc_hm[nrow(l2fc_hm):1,,drop=F]
  	l2fc_hm1 = l2fc_hm
  	# sort tax dat
  	colannotation =  temp_metat%>% filter(regclass2 == location,regressionclass == regclass) %>% select(timetrend,yvar) %>% distinct
  	colannotation = left_join(colannotation,taxdata,by=c('yvar'='query')) %>% column_to_rownames('yvar') 
  	
  	# mask unclassified
  	tomask = c('Phy_of_Anelloviridae','Phy_of_environmental samples','Phy_unclassified viruses','Phy_of_unclassified viruses','Phy_of_unclassified bacterial viruses','Phy_unclassified Riboviria')
  	
  	 colannotation = colannotation %>% mutate(Phylum2 = if_else(Phylum %in% tomask,'Other/Unclassified',Phylum)) %>% arrange(timetrend,Phylum2,Phylum)

  	# mask uncommon phyla
  	tonotmask = colannotation %>% filter(Phylum != 'Other/Unclassified') %>% select(Phylum) %>% table %>% data.frame %>% arrange(desc(Freq)) %>% head(10) %>% select(".") %>% unlist %>% unname %>% as.character
  	colannotation = colannotation %>% mutate(Phylum2 = if_else(Phylum2 %in% tonotmask,Phylum2,'Other/Unclassified')) %>% arrange(timetrend,Phylum2,Phylum)
  	
  	col_ha = columnAnnotation(show_legend = c(FALSE, FALSE),show_annotation_name = c(FALSE,FALSE),Trend = colannotation$timetrend,Phylum = colannotation$Phylum2,col = list(Trend = c('Decreased in flight'='#B2DF8A','Increased in flight'='#FB9A99','Increased in/after flight'='#E31A1C','Decreased in/after flight'='#33A02C'),Phylum = unlist(colormapping)))
  
  	#heatmapframe = heatmapframe[rownames(rowannotation),]
  	l2fc_hm = l2fc_hm %>% select(all_of(rownames(colannotation)))
  	
  	col_fun = colorRamp2(c(-10,-2,0,2,10), rev(c("#2ED8F0","#1768BD", "#000000","purple","#BE1E2D")))
  
	h1 = Heatmap(l2fc_hm,cluster_rows = F,cluster_columns=F,row_split = c(1,1,1,2,3,3,3),cluster_row_slices = F,show_row_names = F,show_column_names=F,column_split = colannotation$timetrend,cluster_column_slices = F,row_title=NULL,column_title_rot = 90,col=col_fun,top_annotation = col_ha,column_title = NULL,show_heatmap_legend = F)
    output[[1]] = h1
  }

    	temp_metag = regression_output %>% filter(BH_adjusted<0.05,dset == dataset,seqtype == 'METAGENOMICS',regressionclass == regclass)%>% filter(grepl('---',term)) %>% mutate(regclass2 = strsplit(as.character(term),' --- ') %>% map_chr(2))
  	  	temp_metag = inner_join(temp_metag,categories) %>% select(timetrend,regressionclass,yvar,dset,regclass2) %>% distinct
    	if(nrow(temp_metag)>0){
      	 	  print('Generating metagenomic heatmap')

  	temp2_metag_sub = metagdata %>% select(any_of(temp_metag$yvar)) %>% t %>% as.data.frame(check.names = F) %>% rownames_to_column('yvar') %>% melt
  	temp_metag = inner_join(temp_metag,temp2_metag_sub)
  	temp_metag = left_join(temp_metag,meta %>% rename(variable = SeqID)) %>% filter(!is.na(Timepoint))
  
  	# generate l2fc heatmap
  	l2fc_hm = temp_metag %>% filter(regclass2 == location)%>% group_by(yvar,Timepoint) %>% summarise(mean = mean(value)) %>% dcast(yvar ~ Timepoint,value.var ='mean') %>% mutate(mean_flight_abundance = (`Flight 1` + `Flight 2`)/2) %>% mutate(l2fc_PRE_june = log2((mean_flight_abundance+minval)) - log2(`21-Jun`+minval)) %>% mutate(l2fc_PRE_aug = log2((mean_flight_abundance+minval)) - log2(`21-Aug`+minval)) %>% mutate(l2fc_PRE_sept = log2((mean_flight_abundance+minval)) - log2(`Sept Pre-Launch`+minval)) %>% mutate(l2fcMID = 0) %>% mutate(l2fc_POST_sept = -1*(log2((mean_flight_abundance+minval)) - log2(`Sept Post-Return`+minval))) %>% mutate(l2fc_POST_nov = -1*(log2((mean_flight_abundance+minval)) - log2(`November`+minval))) %>% mutate(l2fc_POST_dec = -1*(log2((mean_flight_abundance+minval)) - log2(`21-Dec`+minval))) %>% select(-`Flight 1`,-`Flight 2`) %>% column_to_rownames("yvar") %>% t %>% as.data.frame(check.names=F)
  	l2fc_hm = l2fc_hm[grepl('l2fc',rownames(l2fc_hm)),,drop=F]
    l2fc_hm = l2fc_hm[nrow(l2fc_hm):1,,drop=F]
 	
	# sort tax dat
  	colannotation =  temp_metag%>% filter(regclass2 == location,regressionclass == regclass) %>% select(timetrend,yvar) %>% distinct
  	colannotation = left_join(colannotation,taxdata,by=c('yvar'='query')) %>% column_to_rownames('yvar') 
  	
  	# mask unclassified
  	tomask = c('Phy_of_Anelloviridae','Phy_of_environmental samples','Phy_unclassified viruses','Phy_of_unclassified viruses','Phy_of_unclassified bacterial viruses','Phy_unclassified Riboviria')
  	
  	 colannotation = colannotation %>% mutate(Phylum2 = if_else(Phylum %in% tomask,'Other/Unclassified',Phylum)) %>% arrange(timetrend,Phylum2,Phylum)

  	# mask uncommon phyla
  	tonotmask = colannotation %>% filter(Phylum != 'Other/Unclassified') %>% select(Phylum) %>% table %>% data.frame %>% arrange(desc(Freq)) %>% head(10) %>% select(".") %>% unlist %>% unname %>% as.character
  	colannotation = colannotation %>% mutate(Phylum2 = if_else(Phylum2 %in% tonotmask,Phylum2,'Other/Unclassified')) %>% arrange(timetrend,Phylum2,Phylum)
  	
  	col_ha = columnAnnotation(show_legend = c(FALSE, FALSE),show_annotation_name = c(FALSE,FALSE),Trend = colannotation$timetrend,Phylum = colannotation$Phylum2,col = list(Trend = c('Decreased in flight'='#B2DF8A','Increased in flight'='#FB9A99','Increased in/after flight'='#E31A1C','Decreased in/after flight'='#33A02C'),Phylum = unlist(colormapping)))
  
  	#heatmapframe = heatmapframe[rownames(rowannotation),]
  	l2fc_hm = l2fc_hm %>% select(all_of(rownames(colannotation)))
  	
	h2 = Heatmap(l2fc_hm,cluster_rows = F,cluster_columns=F,row_split = c(1,1,1,2,3,3,3),cluster_row_slices = F,show_row_names = F,show_column_names=F,column_split = colannotation$timetrend,cluster_column_slices = F,row_title=NULL,column_title_rot = 90,col=col_fun,top_annotation = col_ha,column_title = NULL,show_heatmap_legend = F)
	output[[2]] = h2
	output[[3]] =l2fc_hm1
	output[[4]] =l2fc_hm
  	}
  
	return(output)
}

#categories = fortrendline  %>% select(yvar,dset,seqtype,timetrend,regclass2)

### ORAL
gtdb_heatmap_oral = generate_fc_heatmap(regression_output,'GTDB',gtdb_metag,gtdb_metat,'Oral',categories %>% filter(!grepl('-- LP',timetrendrank)),gtdb_tax %>% distinct %>% select(query,Phylum,Species))
#kraken_heatmap_oral = generate_fc_heatmap(regression_output,'KRAKEN2',kraken_metag,kraken_metat,'Oral',categories,kraken_tax %>% select(kraken_column_name,phylum) %>% rename(query = kraken_column_name, Phylum = phylum))
genvir_heatmap_oral = generate_fc_heatmap(regression_output,'GENBANK-VIRUSES',viral_metag,viral_metat %>% t %>% as.data.frame(check.names=F),'Oral',categories %>% filter(!grepl('-- LP',timetrendrank)),genvir_tax %>% select(query,Phy) %>% distinct %>% rename(Phylum = Phy))

### NASAL
gtdb_heatmap_nasal = generate_fc_heatmap(regression_output,'GTDB',gtdb_metag,gtdb_metat,'Nasal',categories %>% filter(!grepl('-- LP',timetrendrank)),gtdb_tax %>% distinct %>% select(query,Phylum))
#kraken_heatmap_nasal = generate_fc_heatmap(regression_output,'KRAKEN2',kraken_metag,kraken_metat,'Nasal',categories,kraken_tax %>% select(kraken_column_name,phylum) %>% rename(query = kraken_column_name, Phylum = phylum))
genvir_heatmap_nasal = generate_fc_heatmap(regression_output,'GENBANK-VIRUSES',viral_metag,viral_metat %>% t %>% as.data.frame(check.names=F),'Nasal',categories %>% filter(!grepl('-- LP',timetrendrank)),genvir_tax %>% select(query,Phy) %>% distinct %>% rename(Phylum = Phy))

### SKIN
gtdb_heatmap_skin_components = generate_fc_heatmap(regression_output,'GTDB',gtdb_metag,gtdb_metat,'Skin_Components',categories %>% filter(!grepl('-- LP',timetrendrank)),gtdb_tax %>% distinct %>% select(query,Phylum))
#kraken_heatmap_skin_components = generate_fc_heatmap(regression_output,'KRAKEN2',kraken_metag,kraken_metat,'Skin_Components',categories,kraken_tax %>% select(kraken_column_name,phylum) %>% rename(query = kraken_column_name, Phylum = phylum))
genvir_heatmap_skin_components = generate_fc_heatmap(regression_output,'GENBANK-VIRUSES',viral_metag,viral_metat %>% t %>% as.data.frame(check.names=F),'Skin_Components',categories %>% filter(!grepl('-- LP',timetrendrank)),genvir_tax %>% select(query,Phy) %>% distinct %>% rename(Phylum = Phy))

### TO SHOW IN MAIN TEXT

# ORAL -- GTDB METAG/METAG, GENVIR METAT
pdf('~/Dropbox (Mason Lab)/i4/i4_data_packet/fc_heatmaps/oral_gtdb_metat.pdf',width=7,height=2)
gtdb_heatmap_oral[[1]]
dev.off()

pdf('~/Dropbox (Mason Lab)/i4/i4_data_packet/fc_heatmaps/oral_gtdb_metag.pdf',width=7,height=2)
gtdb_heatmap_oral[[2]]
dev.off()

pdf('~/Dropbox (Mason Lab)/i4/i4_data_packet/fc_heatmaps/oral_genvir_metat.pdf',width=7,height=1.5)
genvir_heatmap_oral[[1]]
dev.off()

# NASAL -- GENVIR METAT
pdf('~/Dropbox (Mason Lab)/i4/i4_data_packet/fc_heatmaps/nasal_genvir_metat.pdf',width=7,height=1.5)
genvir_heatmap_nasal[[1]]
dev.off()

# SKIN -- GTDB METAG/METAG, GENVIR METAG/METAT
pdf('~/Dropbox (Mason Lab)/i4/i4_data_packet/fc_heatmaps/skin_gtdb_metat.pdf',width=7,height=2)
gtdb_heatmap_skin_components[[1]]
dev.off()

pdf('~/Dropbox (Mason Lab)/i4/i4_data_packet/fc_heatmaps/skin_gtdb_metag.pdf',width=7,height=2)
gtdb_heatmap_skin_components[[2]]
dev.off()

pdf('~/Dropbox (Mason Lab)/i4/i4_data_packet/fc_heatmaps/skin_genvir_metat.pdf',width=7,height=1.5)
genvir_heatmap_skin_components[[1]]
dev.off()

pdf('~/Dropbox (Mason Lab)/i4/i4_data_packet/fc_heatmaps/skin_genvir_metag.pdf',width=7,height=1.5)
genvir_heatmap_skin_components[[2]]
dev.off()

```
### FOLD CHANGE DOTPLOTS
```{r}
# high coverage dotplot regressions

generate_fc_dotplot <- function(regression_output,dataset,metagdata,metatdata,regclass,categories,taxdata){
  output=list()
  minval = 0.0000000001
  	temp_metat = regression_output %>% filter(BH_adjusted<0.05,!grepl('cds',yvar),!grepl('partial ORF',yvar),dset == dataset,seqtype == 'METATRANSCRIPTOMICS',regressionclass == regclass)%>% filter(grepl('---',term)) %>% mutate(regclass2 = strsplit(as.character(term),' --- ') %>% map_chr(2))
  	temp_metat = inner_join(temp_metat,categories %>% filter(dset == dataset,seqtype == 'METATRANSCRIPTOMICS',regressionclass == regclass))  %>% group_by(timetrend)%>% slice_min(BH_adjusted,n=50) %>%slice_max(abs(estimate),n=15)  %>% select(timetrend,regressionclass,yvar,dset,regclass2,est_pre) %>% distinct
      	 	if(nrow(temp_metat)>0){
      	 	  print('Running metatranscriptomics')

    	temp2_metat_sub = metatdata %>% select(any_of(temp_metat$yvar)) %>% t %>% as.data.frame(check.names = F) %>% rownames_to_column('yvar') %>% melt
    	temp_metat = inner_join(temp_metat,temp2_metat_sub)
    	temp_metat = left_join(temp_metat,meta %>% rename(variable = SeqID)) %>% filter(!is.na(Timepoint))
    
    	# generate l2fc heatmap
    	l2fc_hm = temp_metat %>% filter(regclass2 == location)%>% group_by(yvar,Timepoint_Recode) %>% summarise(mean = mean(value)) %>% dcast(yvar ~ Timepoint_Recode,value.var ='mean') %>% mutate(l2fc_PRE = log2((`MID-FLIGHT`+minval)) - log2(`PRE-LAUNCH`+minval)) %>% mutate(l2fc_POST = -1*(log2((`MID-FLIGHT`+minval)) - log2(`POST-LAUNCH`+minval))) %>% select(yvar,l2fc_PRE,l2fc_POST) %>% mutate(diff = l2fc_POST - l2fc_PRE)%>% arrange(desc(abs((diff))))
    	l2fc_hm = left_join(l2fc_hm,taxdata,by=c('yvar'='query'))
    	l2fc_hm$Species = fct_reorder(as.factor(l2fc_hm$Species),l2fc_hm$diff)
    	#l2fc_hm$Genus = fct_reorder(as.factor(l2fc_hm$Genus),l2fc_hm$diff)
    	
   	temp =  left_join(l2fc_hm %>% select(-diff),temp_metat %>% select(yvar,timetrend) %>% distinct) %>% melt %>% mutate(seqtype = 'METATRANSCRIPTOMICS')  %>% mutate(ID  = paste(seqtype,yvar))
      temp2 = temp %>% dcast(yvar + timetrend ~ variable,value.var = 'value' ) %>% filter(!(abs(l2fc_PRE) <1 & abs(l2fc_POST)<1)) 
      temp2 = temp2 %>% filter(timetrend == 'Decreased in flight' & l2fc_PRE<0 & l2fc_POST>0 | timetrend == 'Increased in flight' & l2fc_PRE>0 & l2fc_POST<0 | timetrend == 'Increased in/after flight' & l2fc_POST>l2fc_PRE |timetrend == 'Decreased in/after flight' & l2fc_PRE>l2fc_POST)
    # TRY filtering based on log2fc (only show greater than abs 1 in one category, for the certain categories make sure log2fc is > 0 where it should be etc)
    	temp = temp %>% filter(yvar %in% temp2$yvar)
    	output[['metat']] = temp
    	
    	}
  	temp_metag = regression_output %>% filter(BH_adjusted<0.05,dset == dataset,seqtype == 'METAGENOMICS',regressionclass == regclass)%>% filter(grepl('---',term)) %>% mutate(regclass2 = strsplit(as.character(term),' --- ') %>% map_chr(2))
  	temp_metag = inner_join(temp_metag,categories %>% filter(dset == dataset,seqtype == 'METAGENOMICS',regressionclass == regclass))  %>% group_by(timetrend)  %>% slice_min(BH_adjusted,n=50) %>%slice_max(abs(estimate),n=15)  %>% select(timetrend,regressionclass,yvar,dset,regclass2,est_pre) %>% distinct
      	 	if(nrow(temp_metag)>0){
      	 	  print('Running metagenomics')

    	temp2_metag_sub = metagdata %>% select(any_of(temp_metag$yvar)) %>% t %>% as.data.frame(check.names = F) %>% rownames_to_column('yvar') %>% melt
    	temp_metag = inner_join(temp_metag,temp2_metag_sub)
    	temp_metag = left_join(temp_metag,meta %>% rename(variable = SeqID)) %>% filter(!is.na(Timepoint))
  
    	# generate l2fc heatmap
    	l2fc_hm = temp_metag %>% filter(regclass2 == location)%>% group_by(yvar,Timepoint_Recode) %>% summarise(mean = mean(value)) %>% dcast(yvar ~ Timepoint_Recode,value.var ='mean') %>% mutate(l2fc_PRE = log2((`MID-FLIGHT`+minval)) - log2(`PRE-LAUNCH`+minval)) %>% mutate(l2fc_POST = -1*(log2((`MID-FLIGHT`+minval)) - log2(`POST-LAUNCH`+minval))) %>% select(yvar,l2fc_PRE,l2fc_POST) %>% mutate(diff = l2fc_POST - l2fc_PRE)%>% arrange(desc(abs((diff))))
    	l2fc_hm = left_join(l2fc_hm,taxdata,by=c('yvar'='query'))
    	l2fc_hm$Species = fct_reorder(as.factor(l2fc_hm$Species),l2fc_hm$diff)
    	
    	temp =  left_join(l2fc_hm %>% select(-diff),temp_metag %>% select(yvar,timetrend) %>% distinct) %>% melt %>% mutate(seqtype = 'METAGENOMICS')  %>% mutate(ID  = paste(seqtype,yvar))
      temp2 = temp %>% dcast(yvar + timetrend ~ variable,value.var = 'value' ) %>% filter(!(abs(l2fc_PRE) <1 & abs(l2fc_POST)<1)) 
      temp2 = temp2 %>% filter(timetrend == 'Decreased in flight' & l2fc_PRE<0 & l2fc_POST>0 | timetrend == 'Increased in flight' & l2fc_PRE>0 & l2fc_POST<0 | timetrend == 'Increased in/after flight' & l2fc_POST>l2fc_PRE |timetrend == 'Decreased in/after flight' & l2fc_PRE>l2fc_POST)
    # TRY filtering based on log2fc (only show greater than abs 1 in one category, for the certain categories make sure log2fc is > 0 where it should be etc)
    	temp = temp %>% filter(yvar %in% temp2$yvar)
    	output[['metag']] = temp
      	 	}
  	return(output)
}

generate_fc_dotplot_vir <- function(regression_output,dataset,metagdata,metatdata,regclass,categories,taxdata){
  output=list()
  minval = 0.0000000001
  	colormapping = setNames(as.list(c(brewer.pal(n=11,'Set3'),brewer.pal(n=11,'Set3'),'gray',brewer.pal(n=7,'Set1'))), c('Actinobacteriota','Bacteroidota','Firmicutes','Firmicutes_A','Firmicutes_C','Fusobacteriota','Patescibacteria','Cyanobacteria','Deinococcota','Proteobacteria','Verrucomicrobiota','Cossaviricota','Cressdnaviricota','Duplornaviricota','Kitrinoviricota','Negarnaviricota','Phixviricota','Pisuviricota','Uroviricota','Lenarviricota','Nucleocytoviricota','Artverviricota','Other/Unclassified','Actinobacteria','Euryarchaeota','Actinobacteria','Crenarchaeota','Peploviricota','Chlamydiae','Candidatus Micrarchaeota'))
  	temp_metat = regression_output %>% filter(BH_adjusted<0.05,!grepl('cds',yvar),!grepl('RdRp gene',yvar),!grepl('partial',yvar),dset == dataset,seqtype == 'METATRANSCRIPTOMICS',regressionclass == regclass)%>% filter(grepl('---',term)) %>% mutate(regclass2 = strsplit(as.character(term),' --- ') %>% map_chr(2))
  	temp_metat = inner_join(temp_metat,categories %>% filter(dset == dataset,seqtype == 'METATRANSCRIPTOMICS',regressionclass == regclass))  %>% group_by(timetrend)%>% slice_min(BH_adjusted,n=50) %>%slice_max(abs(estimate),n=15)  %>% select(timetrend,regressionclass,yvar,dset,regclass2,est_pre) %>% distinct
      	 	if(nrow(temp_metat)>0){
      	 	  print('Running metatranscriptomics')

    	temp2_metat_sub = metatdata %>% select(any_of(temp_metat$yvar)) %>% t %>% as.data.frame(check.names = F) %>% rownames_to_column('yvar') %>% melt
    	temp_metat = inner_join(temp_metat,temp2_metat_sub)
    	temp_metat = left_join(temp_metat,meta %>% rename(variable = SeqID)) %>% filter(!is.na(Timepoint))
    
    	# generate l2fc heatmap
    	l2fc_hm = temp_metat %>% filter(regclass2 == location)%>% group_by(yvar,Timepoint_Recode) %>% summarise(mean = mean(value)) %>% dcast(yvar ~ Timepoint_Recode,value.var ='mean') %>% mutate(l2fc_PRE = log2((`MID-FLIGHT`+minval)) - log2(`PRE-LAUNCH`+minval)) %>% mutate(l2fc_POST = -1*(log2((`MID-FLIGHT`+minval)) - log2(`POST-LAUNCH`+minval))) %>% select(yvar,l2fc_PRE,l2fc_POST) %>% mutate(diff = l2fc_POST - l2fc_PRE)%>% arrange(desc(abs((diff))))
    	l2fc_hm = left_join(l2fc_hm,taxdata,by=c('yvar'='query'))
    	l2fc_hm$yvar = fct_reorder(as.factor(l2fc_hm$yvar),l2fc_hm$diff)
    #	l2fc_hm$Genus = fct_reorder(as.factor(l2fc_hm$Genus),l2fc_hm$diff)
    	
    	### NOW FILTER ON l2FC ALSO
    	temp =  left_join(l2fc_hm %>% select(-diff),temp_metat %>% select(yvar,timetrend) %>% distinct) %>% melt %>% mutate(seqtype = 'METATRANSCRIPTOMICS')  %>% mutate(ID  = paste(seqtype,yvar))
      temp2 = temp %>% dcast(yvar + timetrend ~ variable,value.var = 'value' ) %>% filter(!(abs(l2fc_PRE) <1 & abs(l2fc_POST)<1)) 
      temp2 = temp2 %>% filter(timetrend == 'Decreased in flight' & l2fc_PRE<0 & l2fc_POST>0 | timetrend == 'Increased in flight' & l2fc_PRE>0 & l2fc_POST<0 | timetrend == 'Increased in/after flight' & l2fc_POST>l2fc_PRE |timetrend == 'Decreased in/after flight' & l2fc_PRE>l2fc_POST)
    # TRY filtering based on log2fc (only show greater than abs 1 in one category, for the certain categories make sure log2fc is > 0 where it should be etc)
    	temp = temp %>% filter(yvar %in% temp2$yvar)
    	output[['metat']] = temp
      	 	}
  	temp_metag = regression_output %>% filter(BH_adjusted<0.05,!grepl('cds',yvar),!grepl('RdRp gene',yvar),!grepl('partial',yvar),dset == dataset,seqtype == 'METAGENOMICS',regressionclass == regclass)%>% filter(grepl('---',term)) %>% mutate(regclass2 = strsplit(as.character(term),' --- ') %>% map_chr(2))
  	temp_metag = inner_join(temp_metag,categories %>% filter(dset == dataset,seqtype == 'METAGENOMICS',regressionclass == regclass))  %>% group_by(timetrend)  %>% slice_min(BH_adjusted,n=50) %>%slice_max(abs(estimate),n=15)  %>% select(timetrend,regressionclass,yvar,dset,regclass2,est_pre) %>% distinct
      	 	if(nrow(temp_metag)>0){
      	 	  print('Running metagenomics')

    	temp2_metag_sub = metagdata %>% select(any_of(temp_metag$yvar)) %>% t %>% as.data.frame(check.names = F) %>% rownames_to_column('yvar') %>% melt
    	temp_metag = inner_join(temp_metag,temp2_metag_sub)
    	temp_metag = left_join(temp_metag,meta %>% rename(variable = SeqID)) %>% filter(!is.na(Timepoint))
  
    	# generate l2fc heatmap
    	l2fc_hm = temp_metag %>% filter(regclass2 == location)%>% group_by(yvar,Timepoint_Recode) %>% summarise(mean = mean(value)) %>% dcast(yvar ~ Timepoint_Recode,value.var ='mean') %>% mutate(l2fc_PRE = log2((`MID-FLIGHT`+minval)) - log2(`PRE-LAUNCH`+minval)) %>% mutate(l2fc_POST = -1*(log2((`MID-FLIGHT`+minval)) - log2(`POST-LAUNCH`+minval))) %>% select(yvar,l2fc_PRE,l2fc_POST) %>% mutate(diff = l2fc_POST - l2fc_PRE)%>% arrange(desc(abs((diff))))
    	l2fc_hm = left_join(l2fc_hm,taxdata,by=c('yvar'='query'))
    	l2fc_hm$Species = fct_reorder(as.factor(l2fc_hm$Species),l2fc_hm$diff)
    	
    	temp =  left_join(l2fc_hm %>% select(-diff),temp_metat %>% select(yvar,timetrend) %>% distinct) %>% melt %>% mutate(seqtype = 'METAGENOMICS')  %>% mutate(ID  = paste(seqtype,yvar))
      temp2 = temp %>% dcast(yvar + timetrend ~ variable,value.var = 'value' ) %>% filter(!(abs(l2fc_PRE) <1 & abs(l2fc_POST)<1)) 
      temp2 = temp2 %>% filter(timetrend == 'Decreased in flight' & l2fc_PRE<0 & l2fc_POST>0 | timetrend == 'Increased in flight' & l2fc_PRE>0 & l2fc_POST<0 | timetrend == 'Increased in/after flight' & l2fc_POST>l2fc_PRE |timetrend == 'Decreased in/after flight' & l2fc_PRE>l2fc_POST)
    # TRY filtering based on log2fc (only show greater than abs 1 in one category, for the certain categories make sure log2fc is > 0 where it should be etc)
    	temp = temp %>% filter(yvar %in% temp2$yvar)
    	output[['metag']] = temp
      	 	}
  	output = bind_rows(output)
  	output$yvar = gsub(", complete genome*","",output$yvar)
  	output$yvar = gsub(" complete genome*","",output$yvar)
  	output$yvar = gsub(", complete sequence*","",output$yvar)
  	output$yvar = gsub(" complete sequence*","",output$yvar)
  	output$yvar = gsub(" genome sequence, sequence","",output$yvar)
  	output$yvar = gsub(" genomic sequence, sequence","",output$yvar)
  	return(output)
}
### ORAL
gtdb_heatmap_oral = generate_fc_dotplot(regression_output,'GTDB',gtdb_metag,gtdb_metat,'Oral',categories,gtdb_tax %>% distinct %>% select(query,Phylum,Species)) %>% bind_rows %>% mutate(dset = 'ORAL')
#kraken_heatmap_oral = generate_fc_dotplot(regression_output,'KRAKEN2',kraken_metag,kraken_metat,'Oral',categories,kraken_tax %>% select(kraken_column_name,phylum) %>% rename(query = kraken_column_name, Phylum = phylum,Species = species))
genvir_heatmap_oral = generate_fc_dotplot_vir(regression_output,'GENBANK-VIRUSES',viral_metag,viral_metat %>% t %>% as.data.frame(check.names=F),'Oral',categories,genvir_tax %>% select(query,Phy,Gen,Spe) %>% distinct %>% rename(Phylum = Phy,Genus = Gen,Species = Spe))%>% mutate(dset = 'ORAL')

### NASAL
#gtdb_heatmap_nasal = generate_fc_dotplot(regression_output,'GTDB',gtdb_metag,gtdb_metat,'Nasal',categories,gtdb_tax %>% distinct %>% select(query,Phylum,Species))%>% mutate(dset = 'NASAL')
#kraken_heatmap_nasal = generate_fc_dotplot(regression_output,'KRAKEN2',kraken_metag,kraken_metat,'Nasal',categories,kraken_tax %>% select(kraken_column_name,phylum) %>% rename(query = kraken_column_name, Phylum = phylum,Species = species))
genvir_heatmap_nasal = generate_fc_dotplot_vir(regression_output,'GENBANK-VIRUSES',viral_metag,viral_metat %>% t %>% as.data.frame(check.names=F),'Nasal',categories,genvir_tax %>% select(query,Phy,Gen,Spe) %>% distinct %>% rename(Phylum = Phy,Genus = Gen,Species = Spe))%>% mutate(dset = 'NASAL')

### SKIN
gtdb_heatmap_skin_components = generate_fc_dotplot(regression_output,'GTDB',gtdb_metag,gtdb_metat,'Skin_Components',categories %>% filter(!grepl('-- LP',timetrendrank)),gtdb_tax %>% distinct %>% select(query,Phylum,Species))%>% bind_rows%>% mutate(dset = 'SKIN')
#kraken_heatmap_skin_components = generate_fc_dotplot(regression_output,'KRAKEN2',kraken_metag,kraken_metat,'Skin_Components',categories,kraken_tax %>% select(kraken_column_name,phylum) %>% rename(query = kraken_column_name, Phylum = phylum,Species = species))
genvir_heatmap_skin_components = generate_fc_dotplot_vir(regression_output,'GENBANK-VIRUSES',viral_metag,viral_metat %>% t %>% as.data.frame(check.names=F),'Skin_Components',categories,genvir_tax %>% select(query,Phy,Gen,Spe) %>% distinct %>% rename(Phylum = Phy,Genus = Gen,Species = Spe)) %>% mutate(dset = 'SKIN')


### BACTERIAL PLOT
bacplot = bind_rows(gtdb_heatmap_oral,gtdb_heatmap_skin_components) %>% mutate(dset = factor(dset,levels = c('ORAL','NASAL','SKIN'))) %>% arrange(dset,timetrend,seqtype) %>% mutate(seqv = seq(1,nrow(.)))

bacplot$Species = fct_reorder(bacplot$Species,bacplot$seqv)
bacplot$variable = gsub('l2fc_','',bacplot$variable)
bacplot$variable = factor(bacplot$variable,levels=c('PRE','POST'))

plot = ggplot(bacplot,aes(x = value, y = Species,shape = variable,fill = seqtype)) + geom_bar(stat='identity',position='dodge')  + theme_minimal() + xlab('Log2(Fold Change)') + geom_vline(xintercept = 0,linetype='dashed',color='gray') + theme(legend.position = 'None') + facet_grid(dset ~ timetrend + variable,scales= 'free')+scale_color_manual(values = c('darkblue','#BE1E2D')) +scale_fill_manual(values = c('METAGENOMICS'= 'darkblue','METATRANSCRIPTOMICS'='#BE1E2D'))  + ylab('')
ggsave(plot = plot,'~/Dropbox (Mason Lab)/i4/regression_dotplots/bacplot_fc.pdf',width=15,height=12)

### VIRAL PLOT
viralplot = bind_rows(genvir_heatmap_oral,genvir_heatmap_nasal,genvir_heatmap_skin_components) %>% mutate(dset = factor(dset,levels = c('ORAL','NASAL','SKIN'))) %>% arrange(dset,timetrend) %>% mutate(seqv = seq(1,nrow(.)))

viralplot$yvar = fct_reorder(viralplot$yvar,viralplot$seqv)
viralplot$variable = gsub('l2fc_','',viralplot$variable)
viralplot$variable = factor(viralplot$variable,levels=c('PRE','POST'))

plot = ggplot(viralplot,aes(x = value, y = yvar,shape = variable,fill = seqtype)) + geom_bar(stat='identity',position='dodge')  + theme_minimal() + xlab('Log2(Fold Change)') + geom_vline(xintercept = 0,linetype='dashed',color='gray') + theme(legend.position = 'None') + facet_grid(dset ~ timetrend + variable,scales= 'free',space = 'free_y')+scale_color_manual(values = c('darkblue','#BE1E2D')) +scale_fill_manual(values = c('METAGENOMICS'= 'darkblue','METATRANSCRIPTOMICS'='#BE1E2D'))  + ylab('')
ggsave(plot = plot,'~/Dropbox (Mason Lab)/i4/regression_dotplots/viral_fc.pdf',width=12,height=12)
#ggsave(plot = gtdb_heatmap_oral,'~/Dropbox (Mason Lab)/i4/regression_dotplots/gtdb_oral_dotplot_fc.pdf',width=7,height=15)

#ggsave(plot = gtdb_heatmap_nasal,'~/Dropbox (Mason Lab)/i4/regression_dotplots/gtdb_nasal_dotplot_fc.pdf',width=7,height=15)

#ggsave(plot = gtdb_heatmap_skin_components,'~/Dropbox (Mason Lab)/i4/regression_dotplots/gtdb_skin_dotplot_fc.pdf',width=7,height=15)

```
########FUNCTIONAL DOTPLOTS
```{r}
#regression_outputgene = regression_output_geneanno
#dataset = 'GENE-CATALOG'
#metagdata = gene_metag
#metatdata = gene_metat
#regclass = 'Skin_Components'
#taxdata =gene_annos %>% select(yvar,product) %>% distinct %>% filter(product != 'hypothetical protein') %>% rename(query=yvar,Species=product)

generate_fc_dotplot_gene <- function(regression_outputgene,dataset,metagdata,metatdata,regclass,categories,taxdata){
  output=list()
  minval = 0.0000000001
  	temp_metat = regression_outputgene %>% filter(BH_adjusted<0.05,!grepl('cds',yvar),!grepl('partial ORF',yvar),dset == dataset,seqtype == 'METATRANSCRIPTOMICS',regressionclass == regclass)
       if(regclass == 'Overall'){
        temp_metat = inner_join(temp_metat,categories %>% filter(dset == dataset,seqtype == 'METATRANSCRIPTOMICS',regressionclass == regclass))  %>% group_by(timetrend) %>% select(timetrend,regressionclass,yvar,dset,regclass2,est_pre) %>% distinct
       }
       if(regclass != 'Overall'){
      temp_metat = inner_join(temp_metat,categories %>% filter(dset == dataset,seqtype == 'METATRANSCRIPTOMICS',regressionclass == regclass))  %>% group_by(timetrend)%>% slice_min(BH_adjusted,n=50) %>%slice_max(abs(estimate),n=15)  %>% select(timetrend,regressionclass,yvar,dset,regclass2,est_pre) %>% distinct
      }      	 	
  	if(nrow(temp_metat)>0){
      	 	  print('Running metatranscriptomics')

    	temp2_metat_sub = metatdata %>% select(any_of(temp_metat$yvar)) %>% t %>% as.data.frame(check.names = F) %>% rownames_to_column('yvar') %>% melt
    	temp_metat = inner_join(temp_metat,temp2_metat_sub)
    	temp_metat = left_join(temp_metat,meta %>% rename(variable = SeqID)) %>% filter(!is.na(Timepoint))
    
    	# generate l2fc heatmap
    	l2fc_hm = temp_metat %>% group_by(yvar,Timepoint_Recode) %>% summarise(mean = mean(value)) %>% dcast(yvar ~ Timepoint_Recode,value.var ='mean') %>% mutate(l2fc_PRE = log2((`MID-FLIGHT`+minval)) - log2(`PRE-LAUNCH`+minval)) %>% mutate(l2fc_POST = -1*(log2((`MID-FLIGHT`+minval)) - log2(`POST-LAUNCH`+minval))) %>% select(yvar,l2fc_PRE,l2fc_POST) %>% mutate(diff = l2fc_POST - l2fc_PRE)%>% arrange(desc(abs((diff))))
    	l2fc_hm = left_join(l2fc_hm,taxdata,by=c('yvar'='query'))
    	l2fc_hm$Species = fct_reorder(as.factor(l2fc_hm$Species),l2fc_hm$diff)

   	temp =  left_join(l2fc_hm %>% select(-diff),temp_metat %>% select(yvar,timetrend) %>% distinct) %>% melt %>% mutate(seqtype = 'METATRANSCRIPTOMICS')  %>% mutate(ID  = paste(seqtype,yvar))
      temp2 = temp %>% dcast(yvar + timetrend ~ variable,value.var = 'value' ) %>% filter(!(abs(l2fc_PRE) <1 & abs(l2fc_POST)<1)) 
      temp2 = temp2 %>% filter(timetrend == 'Decreased in flight' & l2fc_PRE<0 & l2fc_POST>0 | timetrend == 'Increased in flight' & l2fc_PRE>0 & l2fc_POST<0 | timetrend == 'Increased in/after flight' & l2fc_POST>l2fc_PRE |timetrend == 'Decreased in/after flight' & l2fc_PRE>l2fc_POST)
    # TRY filtering based on log2fc (only show greater than abs 1 in one category, for the certain categories make sure log2fc is > 0 where it should be etc)
    	temp = temp %>% filter(yvar %in% temp2$yvar)
    	output[['metat']] = temp
    	}
       temp_metag = regression_outputgene %>% filter(BH_adjusted<0.05,!grepl('cds',yvar),!grepl('partial ORF',yvar),dset == dataset,seqtype == 'METAGENOMICS',regressionclass == regclass)#%>% filter(grepl('---',term)) %>% mutate(regclass2 = strsplit(as.character(term),' --- ') %>% map_chr(2))
       if(regclass == 'Overall'){
        temp_metag = inner_join(temp_metag,categories %>% filter(dset == dataset,seqtype == 'METAGENOMICS',regressionclass == regclass))  %>% group_by(timetrend) %>% select(timetrend,regressionclass,yvar,dset,regclass2,est_pre) %>% distinct
       }
       if(regclass != 'Overall'){
      temp_metag = inner_join(temp_metag,categories %>% filter(dset == dataset,seqtype == 'METAGENOMICS',regressionclass == regclass))  %>% group_by(timetrend)%>% slice_min(BH_adjusted,n=50) %>%slice_max(abs(estimate),n=15)  %>% select(timetrend,regressionclass,yvar,dset,regclass2,est_pre) %>% distinct
      }
       if(nrow(temp_metag)>0){
                    print('Running metagenomics')

      temp2_metag_sub = metagdata %>% select(any_of(temp_metag$yvar)) %>% t %>% as.data.frame(check.names = F) %>% rownames_to_column('yvar') %>% melt
      temp_metag = inner_join(temp_metag,temp2_metag_sub)
      temp_metag = left_join(temp_metag,meta %>% rename(variable = SeqID)) %>% filter(!is.na(Timepoint))
    
      # generate l2fc heatmap
      l2fc_hm = temp_metag %>% group_by(yvar,Timepoint_Recode) %>% summarise(mean = mean(value)) %>% dcast(yvar ~ Timepoint_Recode,value.var ='mean') %>% mutate(l2fc_PRE = log2((`MID-FLIGHT`+minval)) - log2(`PRE-LAUNCH`+minval)) %>% mutate(l2fc_POST = -1*(log2((`MID-FLIGHT`+minval)) - log2(`POST-LAUNCH`+minval))) %>% select(yvar,l2fc_PRE,l2fc_POST) %>% mutate(diff = l2fc_POST - l2fc_PRE)%>% arrange(desc(abs((diff))))
      l2fc_hm = left_join(l2fc_hm,taxdata,by=c('yvar'='query'))
      l2fc_hm$Species = fct_reorder(as.factor(l2fc_hm$Species),l2fc_hm$diff)
      #l2fc_hm$Genus = fct_reorder(as.factor(l2fc_hm$Genus),l2fc_hm$diff)
      
      temp =  left_join(l2fc_hm %>% select(-diff),temp_metag %>% select(yvar,timetrend) %>% distinct) %>% melt %>% mutate(seqtype = 'METAGENOMICS')  %>% mutate(ID  = paste(seqtype,yvar))
      temp2 = temp %>% dcast(yvar + timetrend ~ variable,value.var = 'value' ) %>% filter(!(abs(l2fc_PRE) <1 & abs(l2fc_POST)<1)) 
      temp2 = temp2 %>% filter(timetrend == 'Decreased in flight' & l2fc_PRE<0 & l2fc_POST>0 | timetrend == 'Increased in flight' & l2fc_PRE>0 & l2fc_POST<0 | timetrend == 'Increased in/after flight' & l2fc_POST>l2fc_PRE |timetrend == 'Decreased in/after flight' & l2fc_PRE>l2fc_POST)
    # TRY filtering based on log2fc (only show greater than abs 1 in one category, for the certain categories make sure log2fc is > 0 where it should be etc)
      temp = temp %>% filter(yvar %in% temp2$yvar)
    	output[['metag']] = temp
      	 	}
       return(output)
}

regression_output_geneanno = inner_join(regression_output %>% filter(dset == 'GENE-CATALOG'),gene_annos %>% filter(product!='hypothetical protein'))
categories = fortrendline %>% filter(regclass2 != 'OVERALL') %>% select(yvar,seqtype,dset,timetrend,timetrendrank,regclass2,adj_pre,est_pre,adj_post,est_post)%>% filter(regclass2 != 'Skin') %>% mutate(regressionclass = if_else(regclass2 == 'Nasal' | regclass2 == 'Oral' | regclass2 == 'Overall',regclass2,'Skin_Components'))

#### QUICK SIDESTEP TO FIND KEGG GROUPS
kegg = regression_output_geneanno %>% filter(grepl('KEGG',other_annotations)) %>% select(yvar,other_annotations) %>% mutate(kegg = strsplit(other_annotations,'KEGG:') %>% map_chr(2) %>% strsplit(',') %>% map_chr(1)) %>% distinct
keggegories = inner_join(kegg,categories)

for(val in unique(keggegories$timetrend)){
  sub =keggegories %>% filter(timetrend == val)
  for(val2 in unique(keggegories$regressionclass)){
    sub2 =sub %>% filter(regressionclass == val2)
    for(val3 in unique(keggegories$seqtype)){
      sub3 =sub2 %>% filter(seqtype == val3)
    write.table(file = paste(gsub('/','',gsub(' ','',val)),val2,val3,'.tsv',sep=''),sub3 %>% select(kegg),quote=F,sep='\t',col.names = FALSE,row.names=FALSE) 
    }
  }
}

gene_heatmap_overall = generate_fc_dotplot_gene(regression_output_geneanno,'GENE-CATALOG',gene_metag,gene_metat,'Overall',categories,gene_annos %>% select(yvar,product) %>% distinct %>% filter(product != 'hypothetical protein') %>% rename(query=yvar,Species=product)) %>% bind_rows %>% mutate(regclass = 'Overall')

# get prevalence for core genes
prevs = gene_metag %>% select(all_of(gene_heatmap_overall %>% filter(seqtype == 'METAGENOMICS') %>% select(yvar) %>% unlist %>% unname %>% unique)) %>%  rownames_to_column('SeqID')%>% melt
prevs = left_join(prevs,meta) %>% filter(!is.na(Crew.ID))
prevs = prevs %>% mutate(skinoralnasal = if_else(isoral == 1,'ORAL',if_else(isskin==1,'SKIN','NASAL'))) 
prevs = left_join(prevs %>% rename(yvar = variable), gene_heatmap_overall %>% select(yvar,Species))
prevs = prevs %>% select(skinoralnasal,value,Timepoint_Recode,Species)%>% filter(Timepoint_Recode == 'MID-FLIGHT') %>% mutate(value = if_else(value>0,1,0))
prevs1 = prevs %>% group_by(skinoralnasal,Species) %>% summarize(prevalence = mean(value > 0), .groups = "drop")

prevs = gene_metat %>% select(all_of(gene_heatmap_overall %>% filter(seqtype == 'METATRANSCRIPTOMICS') %>% select(yvar) %>% unlist %>% unname %>% unique)) %>%  rownames_to_column('SeqID')%>% melt
prevs = left_join(prevs,meta) %>% filter(!is.na(Crew.ID))
prevs = prevs %>% mutate(skinoralnasal = if_else(isoral == 1,'ORAL',if_else(isskin==1,'SKIN','NASAL'))) 
prevs = left_join(prevs %>% rename(yvar = variable), gene_heatmap_overall %>% select(yvar,Species))
prevs = prevs %>% select(skinoralnasal,value,Timepoint_Recode,Species)%>% filter(Timepoint_Recode == 'MID-FLIGHT') %>% mutate(value = if_else(value>0,1,0))
prevs2 = prevs %>% group_by(skinoralnasal,Species) %>% summarize(prevalence = mean(value > 0), .groups = "drop")

prevs = bind_rows(prevs1,prevs2)

gene_heatmap_oral = generate_fc_dotplot_gene(regression_output_geneanno,'GENE-CATALOG',gene_metag,gene_metat,'Oral',categories,gene_annos %>% select(yvar,product) %>% distinct %>% filter(product != 'hypothetical protein') %>% rename(query=yvar,Species=product)) %>% bind_rows %>% mutate(regclass = 'ORAL')
gene_heatmap_oral$variable = gsub('l2fc_','',gene_heatmap_oral$variable)
gene_heatmap_oral$variable = factor(gene_heatmap_oral$variable,levels=c('PRE','POST'))

gene_heatmap_nasal = generate_fc_dotplot_gene(regression_output_geneanno,'GENE-CATALOG',gene_metag,gene_metat,'Nasal',categories,gene_annos %>% select(yvar,product) %>% distinct %>% filter(product != 'hypothetical protein') %>% rename(query=yvar,Species=product)) %>% bind_rows %>% mutate(regclass = 'NASAL')
gene_heatmap_nasal$variable = gsub('l2fc_','',gene_heatmap_nasal$variable)
gene_heatmap_nasal$variable = factor(gene_heatmap_nasal$variable,levels=c('PRE','POST'))

gene_heatmap_skin = generate_fc_dotplot_gene(regression_output_geneanno,'GENE-CATALOG',gene_metag,gene_metat,'Skin_Components',categories,gene_annos %>% select(yvar,product) %>% distinct %>% filter(product != 'hypothetical protein') %>% rename(query=yvar,Species=product)) %>% bind_rows %>% mutate(regclass = 'SKIN')
gene_heatmap_skin$variable = gsub('l2fc_','',gene_heatmap_skin$variable)
gene_heatmap_skin$variable = factor(gene_heatmap_skin$variable,levels=c('PRE','POST'))


facord1 = prevs %>% group_by(Species) %>% summarise(mean = mean(prevalence)) %>% arrange(desc(mean))
facord2 = prevs%>% filter(prevalence>0) %>% select(skinoralnasal,Species) %>% distinct %>% group_by(Species) %>% count() %>% ungroup %>% select(Species,n)

facord = inner_join(facord1,facord2) %>% arrange((n),(mean)) %>% mutate(seqs=seq(1,nrow(.)))
facord$Species = fct_reorder(as.factor(facord$Species),facord$seqs)

gene_heatmap_overall$Species = factor(gene_heatmap_overall$Species,levels = facord$Species)
prevs$Species = factor(prevs$Species,levels = facord$Species)

plot0 = ggplot(gene_heatmap_overall %>% filter(timetrend!='Decreased in flight'),aes(x = value, y = Species,shape = variable,fill = seqtype)) +geom_bar(stat='identity',position='dodge') + theme_bw() + xlab('Log2(Fold Change)') + geom_vline(xintercept = 0,linetype='dashed',color='gray') + theme(legend.position = 'None') + facet_grid( timetrend ~ .,scales= 'free') +scale_fill_manual(values = c('METAGENOMICS'= 'darkblue','METATRANSCRIPTOMICS'='#BE1E2D')) + ylab('') +facet_wrap(variable~.)

plot00 =ggplot(prevs %>% mutate(variable= 'PRE')%>% filter(Species!='Ribosomal protein bL12, rplL'),aes(y = Species,x= prevalence,color=skinoralnasal)) + geom_point(alpha=.7,size=4) + theme_bw() +facet_wrap(variable~.) + scale_color_manual(values = c('#27ade3','#f5b183','#fce799'))
  
plot_grid(plot0,plot00+ theme(axis.text.y = element_blank(),axis.ticks.y = element_blank(), axis.title.y = element_blank(),legend.position = 'None'), ncol=2, align='h',rel_widths = c(4,1))
ggsave('~/Dropbox (Mason Lab)/i4/functional_plots/overall_functional.pdf',width=10,height=15)

#### SPEC PLOTS
plot1 = ggplot(gene_heatmap_oral,aes(x = value, y = Species,shape = variable,fill = seqtype)) +geom_bar(stat='identity',position='dodge') + theme_minimal() + xlab('Log2(Fold Change)') + geom_vline(xintercept = 0,linetype='dashed',color='gray') + theme(legend.position = 'None') + facet_grid( timetrend ~ variable,scales= 'free') +scale_fill_manual(values = c('METAGENOMICS'= 'darkblue','METATRANSCRIPTOMICS'='#BE1E2D')) + ylab('') 
ggsave('~/Dropbox (Mason Lab)/i4/functional_plots/oral_functional.pdf',width=7,height=9)

plot2 = ggplot(gene_heatmap_nasal,aes(x = value, y = Species,shape = variable,fill = seqtype)) +geom_bar(stat='identity',position='dodge') + theme_minimal() + xlab('Log2(Fold Change)') + geom_vline(xintercept = 0,linetype='dashed',color='gray') + theme(legend.position = 'None') + facet_grid( timetrend ~ variable,scales= 'free') +scale_fill_manual(values = c('METAGENOMICS'= 'darkblue','METATRANSCRIPTOMICS'='#BE1E2D')) + ylab('') 
ggsave('~/Dropbox (Mason Lab)/i4/functional_plots/nasal_functional.pdf',width=7,height=9)

plot3 = ggplot(gene_heatmap_skin,aes(x = value, y = Species,shape = variable,fill = seqtype)) +geom_bar(stat='identity',position='dodge') + theme_minimal() + xlab('Log2(Fold Change)') + geom_vline(xintercept = 0,linetype='dashed',color='gray') + theme(legend.position = 'None') + facet_grid( timetrend ~ variable,scales= 'free') +scale_fill_manual(values = c('METAGENOMICS'= 'darkblue','METATRANSCRIPTOMICS'='#BE1E2D')) + ylab('') 
ggsave('~/Dropbox (Mason Lab)/i4/functional_plots/skin_functional.pdf',width=7,height=9)

fullplot = bind_rows(gene_heatmap_oral,gene_heatmap_nasal,gene_heatmap_skin)
fullplot$regclass = factor(fullplot$regclass,levels=c('ORAL','NASAL','SKIN'))

bigplot = ggplot(fullplot,aes(x = value, y = Species,shape = variable,fill = seqtype)) +geom_bar(stat='identity',position='dodge') + theme_minimal() + xlab('Log2(Fold Change)') + geom_vline(xintercept = 0,linetype='dashed',color='gray') + theme(legend.position = 'None') + facet_grid( timetrend + regclass ~ variable,scales= 'free', space = "free_y") +scale_fill_manual(values = c('METAGENOMICS'= 'darkblue','METATRANSCRIPTOMICS'='#BE1E2D')) + ylab('')# + theme(axis.text.y=element_text(angle=30))
ggsave('~/Dropbox (Mason Lab)/i4/functional_plots/allbs.pdf',width=7,height=18)


forsinglecell_metat = bind_rows(gene_heatmap_overall,gene_heatmap_oral,gene_heatmap_nasal,gene_heatmap_skin) %>% filter(seqtype == 'METATRANSCRIPTOMICS') %>% select(yvar) %>% unlist %>% unname %>% unique

forsinglecell_metag = bind_rows(gene_heatmap_overall,gene_heatmap_oral,gene_heatmap_nasal,gene_heatmap_skin) %>% filter(seqtype == 'METAGENOMICS') %>% select(yvar) %>% unlist %>% unname %>% unique

gene_metag_sc = gene_metag %>% select(all_of(forsinglecell_metag))
gene_metag_sc = log(gene_metag_sc + min(gene_metag_sc[gene_metag_sc>0]))
gene_metag_sc = inner_join(gene_metag_sc %>% rownames_to_column('SeqID'),meta %>% select(SeqID,Crew.ID,Timepoint,Timepoint_Recode,location))
write.csv(gene_metag_sc,'gene_metagenomics_singlecell.csv')

gene_metat_sc = gene_metat %>% select(all_of(forsinglecell_metat))
gene_metat_sc = log(gene_metat_sc + min(gene_metat_sc[gene_metat_sc>0]))
gene_metat_sc = inner_join(gene_metat_sc %>% rownames_to_column('SeqID'),meta %>% select(SeqID,Crew.ID,Timepoint,Timepoint_Recode,location))
write.csv(gene_metat_sc,'gene_metatranscriptomics_singlecell.csv')

```
######### COGS
```{r}
d=read.csv('~/Dropbox (Mason Lab)/i4/i4_data_packet/gene_catalog/cog_annotations_parsed.csv') %>% rename(yvar = Locus.Tag)
d2 = d %>% select(yvar,COG_class)
d2 = inner_join(d2,fortrendline %>% select(yvar,seqtype,regclass2,timetrend)) %>% filter(!is.na(timetrend))
####1206 had some sort of COG annotation that were sig
d2 = d2 %>% mutate(splitcogs = strsplit(as.character(COG_class), "")) %>% unnest(splitcogs)
d2 =d2 %>% select(timetrend,seqtype,regclass2,splitcogs) %>% group_by(timetrend,seqtype,regclass2,splitcogs) %>% count %>% filter(regclass2 != 'Overall') %>% arrange(n)

cogmap=read.csv('~/Dropbox (Mason Lab)/i4/i4_data_packet/gene_catalog/cog_map.csv',header=F) 
colnames(cogmap) = c('GROUP','splitcogs','CATEGORY')

d3 = left_join(d2,cogmap)%>% ungroup%>% filter(!is.na(CATEGORY))
d3 = d3 %>% filter(!is.na(GROUP),!grepl('/',timetrend))

foo = d3 %>% ungroup%>% filter(!is.na(CATEGORY))%>%select(CATEGORY,n,timetrend) %>%filter(timetrend == 'Increased in flight')%>% group_by(CATEGORY) %>% summarise(n = sum(n)) %>% arrange(desc(n))
foo$CATEGORY = fct_reorder(as.factor(foo$CATEGORY),foo$n)
d3$CATEGORY = factor(d3$CATEGORY,levels = c("RNA processing and modification",levels(foo$CATEGORY)))

ggplot(data = d3,aes(x = n,y = CATEGORY,fill=regclass2)) + geom_bar(stat='identity',width=.5) + theme_minimal() + facet_grid(GROUP ~ timetrend ,scales='free_y',space = 'free') + scale_fill_manual(values = brewer.pal('Set3',n=11)) + xlab('') + theme(legend.position ='bottom')
ggsave("~/Dropbox (Mason Lab)/i4/functional_plots/summary_cog_cats.pdf",width=7,height=8)
```
######## FUNCTIONAL ANALYSIS
```{r}

c_gene = inner_join(a,b) %>% filter(dset == 'GENE-CATALOG')

### BOTH
c_gene = c_gene %>% mutate(timetrendrank = 'NONE')%>% mutate(timetrendrank = if_else(est_pre<0 & est_post>0 & adj_pre <0.05 |est_pre<0 & est_post>0 & adj_post <0.05,'Increased in/after flight',timetrendrank)) %>% mutate(timetrendrank = if_else(est_pre>0 & est_post<0 & adj_pre <0.05 | est_pre>0 & est_post<0 & adj_post <0.05,'Decreased in/after flight',timetrendrank)) %>% mutate(timetrendrank = if_else(est_pre<0 & est_post<0 & adj_pre<0.05 | est_pre<0 & est_post<0 & adj_post <0.05,'Increased in flight -- LP',timetrendrank)) %>% mutate(timetrendrank = if_else(est_pre>0 & est_post>0 & adj_pre<0.05 | est_pre>0 & est_post>0  & adj_post <0.05,'Decreased in flight -- LP',timetrendrank)) %>% mutate(timetrendrank = if_else(est_pre<0 & est_post<0 & adj_pre<0.05 & adj_post <0.05,'Increased in flight',timetrendrank)) %>% mutate(timetrendrank = if_else(est_pre>0 & est_post>0 & adj_pre<0.05 & adj_post <0.05,'Decreased in flight',timetrendrank)) 
c_gene = c_gene %>% mutate(timetrend = gsub(' -- LP','',timetrendrank))

fortrendline = c_gene %>% filter(regclass2 != 'OVERALL') %>% mutate(mergeid = paste(yvar,dset,seqtype,regclass2)) 

regression_output_geneanno = left_join(fortrendline %>% filter(dset == 'GENE-CATALOG'),gene_annos)
regression_output_geneanno = regression_output_geneanno %>% filter(regclass2 != 'Skin')# %>% filter(grepl('---',term) | regressionclass == 'Overall')
regression_output_geneanno = regression_output_geneanno %>% mutate(sig = if_else(timetrend == 'NONE','NO','YES'))
















# non hyps
nonhypcounts = regression_output_geneanno %>% filter(product != 'hypothetical protein') %>% filter(!grepl('LP',timetrendrank)) %>% group_by(timetrend,product,seqtype,sig) %>% count

# hyps
regression_output_geneanno %>% filter(product == 'hypothetical protein') %>% filter(BH_adjusted<0.05) %>% filter(!grepl('LP',timetrendrank)) %>% select(timetrend,product)



### FIGURE OUT AMRS

amrgenes = read.table('amr_gene_ids',header=F) %>% rename(refid = V1)

refmapper = regression_output_geneanno %>% filter(grepl('RefSeq',other_annotations)) %>% mutate(refid = other_annotations %>% strsplit('RefSeq:') %>% map_chr(2) %>% strsplit(',') %>% map_chr(1))
refmapper_amrgenes = inner_join(refmapper,amrgenes) %>% mutate(ISAMR = 'IS AMR') %>% select(yvar,refid,ISAMR)

regression_output_geneanno = left_join(regression_output_geneanno,refmapper_amrgenes)

regression_output_geneanno_sig = regression_output_geneanno %>% filter(BH_adjusted<0.05)

### COUNT THE COG CATEGORIES

regression_output_geneanno_sig %>% filter(grepl('COG',other_annotations)) %>% select(yvar,other_annotations) %>% mutate(cogclass = strsplit(other_annotations,'COG:COG') %>% map_chr(2) %>% strsplit('COG:') %>% map_chr(2) %>% strsplit)

### FIND COUNTS OF FUNCTIONS

```
#### LOAD IN REGRESSION OUTPUT RE ONLY -- NO GENES
```{r}

### LOAD IN REGRESSION OUTPUT AND MERGE

regdata = list()
### GTDB
regdata[[1]] = read.table("i4_data_packet/differential_abundance_or_expression_analysis/regression_output_oral_bacterial_dna_species_xtree_.05_.0025_GTDB_r207.tsv",sep='\t') %>% mutate(seqtype = 'METAGENOMICS') %>% mutate(regressionclass = 'Oral')   %>% mutate(dset = 'GTDB')
regdata[[2]] = read.table("i4_data_packet/differential_abundance_or_expression_analysis/regression_output_oral_bacterial_rna_species_xtree_.05_.0025_GTDB_r207.tsv",sep='\t') %>% mutate(seqtype = 'METATRANSCRIPTOMICS') %>% mutate(regressionclass = 'Oral') %>% mutate(dset = 'GTDB')
regdata[[3]] = read.table("i4_data_packet/differential_abundance_or_expression_analysis/regression_output_nasal_bacterial_dna_species_xtree_.05_.0025_GTDB_r207.tsv",sep='\t') %>% mutate(seqtype = 'METAGENOMICS') %>% mutate(regressionclass = 'Nasal')   %>% mutate(dset = 'GTDB')
regdata[[4]] = read.table("i4_data_packet/differential_abundance_or_expression_analysis/regression_output_nasal_bacterial_rna_species_xtree_.05_.0025_GTDB_r207.tsv",sep='\t') %>% mutate(seqtype = 'METATRANSCRIPTOMICS') %>% mutate(regressionclass = 'Nasal') %>% mutate(dset = 'GTDB')
regdata[[5]] = read.table("i4_data_packet/differential_abundance_or_expression_analysis/regression_output_skin_bacterial_dna_species_xtree_.05_.0025_GTDB_r207.tsv",sep='\t') %>% mutate(seqtype = 'METAGENOMICS') %>% mutate(regressionclass = 'Skin') %>% mutate(dset = 'GTDB')
regdata[[6]] = read.table("i4_data_packet/differential_abundance_or_expression_analysis/regression_output_skin_bacterial_rna_species_xtree_.05_.0025_GTDB_r207.tsv",sep='\t') %>% mutate(seqtype = 'METATRANSCRIPTOMICS') %>% mutate(regressionclass = 'Skin') %>% mutate(dset = 'GTDB')
regdata[[7]] = read.table("i4_data_packet/differential_abundance_or_expression_analysis/regression_output_skin_site_by_site_bacterial_dna_species_xtree_.05_.0025_GTDB_r207.tsv",sep='\t') %>% mutate(seqtype = 'METAGENOMICS') %>% mutate(regressionclass = 'Skin_Components') %>% mutate(dset = 'GTDB')
regdata[[8]] = read.table("i4_data_packet/differential_abundance_or_expression_analysis/regression_output_skin_site_by_site_bacterial_rna_species_xtree_.05_.0025_GTDB_r207.tsv",sep='\t') %>% mutate(seqtype = 'METATRANSCRIPTOMICS') %>% mutate(regressionclass = 'Skin_Components') %>% mutate(dset = 'GTDB')

### BACTERIAL ASSEMBLY
regdata[[9]] = read.table("i4_data_packet/differential_abundance_or_expression_analysis/regression_output_oral_bacterial_dna_species_xtree_.05_.0025_ASSEMBLIES.tsv",sep='\t') %>% mutate(seqtype = 'METAGENOMICS') %>% mutate(regressionclass = 'Oral') %>% mutate(dset = 'ASSEMBLED-BACTERIA')
regdata[[10]] = read.table("i4_data_packet/differential_abundance_or_expression_analysis/regression_output_oral_bacterial_rna_species_xtree_.05_.0025_ASSEMBLIES.tsv",sep='\t') %>% mutate(seqtype = 'METATRANSCRIPTOMICS') %>% mutate(regressionclass = 'Oral') %>% mutate(dset = 'ASSEMBLED-BACTERIA')
regdata[[11]] = read.table("i4_data_packet/differential_abundance_or_expression_analysis/regression_output_nasal_bacterial_dna_species_xtree_.05_.0025_ASSEMBLIES.tsv",sep='\t') %>% mutate(seqtype = 'METAGENOMICS') %>% mutate(regressionclass = 'Nasal')   %>% mutate(dset = 'ASSEMBLED-BACTERIA')
regdata[[12]] = read.table("i4_data_packet/differential_abundance_or_expression_analysis/regression_output_nasal_bacterial_rna_species_xtree_.05_.0025_ASSEMBLIES.tsv",sep='\t') %>% mutate(seqtype = 'METATRANSCRIPTOMICS') %>% mutate(regressionclass = 'Nasal') %>% mutate(dset = 'ASSEMBLED-BACTERIA')
regdata[[13]] = read.table("i4_data_packet/differential_abundance_or_expression_analysis/regression_output_skin_bacterial_dna_species_xtree_.05_.0025_ASSEMBLIES.tsv",sep='\t') %>% mutate(seqtype = 'METAGENOMICS') %>% mutate(regressionclass = 'Skin') %>% mutate(dset = 'ASSEMBLED-BACTERIA')
regdata[[14]] = read.table("i4_data_packet/differential_abundance_or_expression_analysis/regression_output_skin_bacterial_rna_species_xtree_.05_.0025_ASSEMBLIES.tsv",sep='\t') %>% mutate(seqtype = 'METATRANSCRIPTOMICS') %>% mutate(regressionclass = 'Skin')  %>% mutate(dset = 'ASSEMBLED-BACTERIA')
regdata[[15]] = read.table("i4_data_packet/differential_abundance_or_expression_analysis/regression_output_skin_site_by_site_bacterial_dna_species_xtree_.05_.0025_ASSEMBLIES.tsv",sep='\t') %>% mutate(seqtype = 'METAGENOMICS') %>% mutate(regressionclass = 'Skin_Components')  %>% mutate(dset = 'ASSEMBLED-BACTERIA')
regdata[[16]] = read.table("i4_data_packet/differential_abundance_or_expression_analysis/regression_output_skin_site_by_site_bacterial_rna_species_xtree_.05_.0025_ASSEMBLIES.tsv",sep='\t') %>% mutate(seqtype = 'METATRANSCRIPTOMICS') %>% mutate(regressionclass = 'Skin_Components') %>% mutate(dset = 'ASSEMBLED-BACTERIA')

### VIRAL GENBANK

regdata[[17]] = read.csv("i4_data_packet/differential_abundance_or_expression_analysis/regression_output_oral_viral_dna_species_xtree_.02_.0.015_VIRAL_GENBANK.tsv",sep='\t') %>% mutate(seqtype = 'METAGENOMICS') %>% mutate(regressionclass = 'Oral') %>% mutate(dset = 'GENBANK-VIRUSES')
regdata[[18]] = read.csv("i4_data_packet/differential_abundance_or_expression_analysis/regression_output_oral_viral_rna_species_xtree_.02_.0.015_VIRAL_GENBANK.tsv",sep='\t') %>% mutate(seqtype = 'METATRANSCRIPTOMICS') %>% mutate(regressionclass = 'Oral') %>% mutate(dset = 'GENBANK-VIRUSES')
regdata[[19]] = read.csv("i4_data_packet/differential_abundance_or_expression_analysis/regression_output_nasal_viral_dna_species_xtree_.02_.0.015_VIRAL_GENBANK.tsv",sep='\t') %>% mutate(seqtype = 'METAGENOMICS') %>% mutate(regressionclass = 'Nasal')   %>% mutate(dset = 'GENBANK-VIRUSES')
regdata[[20]] = read.csv("i4_data_packet/differential_abundance_or_expression_analysis/regression_output_nasal_viral_rna_species_xtree_.02_.0.015_VIRAL_GENBANK.tsv",sep='\t') %>% mutate(seqtype = 'METATRANSCRIPTOMICS') %>% mutate(regressionclass = 'Nasal') %>% mutate(dset = 'GENBANK-VIRUSES')
regdata[[21]] = read.csv("i4_data_packet/differential_abundance_or_expression_analysis/regression_output_skin_viral_dna_species_xtree_.02_.0.015_VIRAL_GENBANK.tsv",sep='\t') %>% mutate(seqtype = 'METAGENOMICS') %>% mutate(regressionclass = 'Skin') %>% mutate(dset = 'GENBANK-VIRUSES')
regdata[[22]] = read.csv("i4_data_packet/differential_abundance_or_expression_analysis/regression_output_skin_viral_rna_species_xtree_.02_.0.015_VIRAL_GENBANK.tsv",sep='\t') %>% mutate(seqtype = 'METATRANSCRIPTOMICS') %>% mutate(regressionclass = 'Skin') %>% mutate(dset = 'GENBANK-VIRUSES')
regdata[[23]] = read.csv("i4_data_packet/differential_abundance_or_expression_analysis/regression_output_skin_site_by_site_viral_dna_species_xtree_.02_.0.015_VIRAL_GENBANK.tsv",sep='\t') %>% mutate(seqtype = 'METAGENOMICS') %>% mutate(regressionclass = 'Skin_Components') %>% mutate(dset = 'GENBANK-VIRUSES')
regdata[[24]] = read.csv("i4_data_packet/differential_abundance_or_expression_analysis/regression_output_skin_site_by_site_viral_rna_species_xtree_.02_.0.015_VIRAL_GENBANK.tsv",sep='\t') %>% mutate(seqtype = 'METATRANSCRIPTOMICS') %>% mutate(regressionclass = 'Skin_Components') %>% mutate(dset = 'GENBANK-VIRUSES')

### VIRAL ASSEMBLED

regdata[[25]] = read.csv("i4_data_packet/differential_abundance_or_expression_analysis/regression_output_oral_viral_dna_species_xtree_.01_.0.005_ASSEMBLIES.tsv",sep='\t') %>% mutate(seqtype = 'METAGENOMICS') %>% mutate(regressionclass = 'Oral') %>% mutate(dset = 'ASSEMBLED-VIRUSES')
regdata[[26]] = read.csv("i4_data_packet/differential_abundance_or_expression_analysis/regression_output_oral_viral_rna_species_xtree_.01_.0.005_ASSEMBLIES.tsv",sep='\t') %>% mutate(seqtype = 'METATRANSCRIPTOMICS') %>% mutate(regressionclass = 'Oral') %>% mutate(dset = 'ASSEMBLED-VIRUSES')
regdata[[27]] = read.csv("i4_data_packet/differential_abundance_or_expression_analysis/regression_output_nasal_viral_dna_species_xtree_.01_.0.005_ASSEMBLIES.tsv",sep='\t') %>% mutate(seqtype = 'METAGENOMICS') %>% mutate(regressionclass = 'Nasal')   %>% mutate(dset = 'ASSEMBLED-VIRUSES')
regdata[[28]] = read.csv("i4_data_packet/differential_abundance_or_expression_analysis/regression_output_nasal_viral_rna_species_xtree_.01_.0.005_ASSEMBLIES.tsv",sep='\t') %>% mutate(seqtype = 'METATRANSCRIPTOMICS') %>% mutate(regressionclass = 'Nasal') %>% mutate(dset = 'ASSEMBLED-VIRUSES')
regdata[[29]] = read.csv("i4_data_packet/differential_abundance_or_expression_analysis/regression_output_skin_viral_dna_species_xtree_.01_.0.005_ASSEMBLIES.tsv",sep='\t') %>% mutate(seqtype = 'METAGENOMICS') %>% mutate(regressionclass = 'Skin') %>% mutate(dset = 'ASSEMBLED-VIRUSES')
regdata[[30]] = read.csv("i4_data_packet/differential_abundance_or_expression_analysis/regression_output_skin_viral_rna_species_xtree_.01_.0.005_ASSEMBLIES.tsv",sep='\t') %>% mutate(seqtype = 'METATRANSCRIPTOMICS') %>% mutate(regressionclass = 'Skin') %>% mutate(dset = 'ASSEMBLED-VIRUSES')
regdata[[31]] = read.csv("i4_data_packet/differential_abundance_or_expression_analysis/regression_output_skin_site_by_site_viral_dna_species_xtree_.01_.0.005_ASSEMBLIES.tsv",sep='\t') %>% mutate(seqtype = 'METAGENOMICS') %>% mutate(regressionclass = 'Skin_Components') %>% mutate(dset = 'ASSEMBLED-VIRUSES')
regdata[[32]] = read.csv("i4_data_packet/differential_abundance_or_expression_analysis/regression_output_skin_site_by_site_viral_rna_species_xtree_.01_.0.005_ASSEMBLIES.tsv",sep='\t') %>% mutate(seqtype = 'METATRANSCRIPTOMICS') %>% mutate(regressionclass = 'Skin_Components') %>% mutate(dset = 'ASSEMBLED-VIRUSES')

### KRAKEN2
regdata[[33]] = read.csv("i4_data_packet/differential_abundance_or_expression_analysis/regression_output_oral_bacterial-viral_dna_species_kraken2_none_KRAKEN2_REFSEQDB.tsv",sep='\t') %>% mutate(seqtype = 'METAGENOMICS') %>% mutate(regressionclass = 'Oral') %>% mutate(dset = 'KRAKEN2')
regdata[[34]] = read.csv("i4_data_packet/differential_abundance_or_expression_analysis/regression_output_oral_bacterial-viral_rna_species_kraken2_none_KRAKEN2_REFSEQDB.tsv",sep='\t') %>% mutate(seqtype = 'METATRANSCRIPTOMICS') %>% mutate(regressionclass = 'Oral') %>% mutate(dset = 'KRAKEN2')
regdata[[35]] = read.csv("i4_data_packet/differential_abundance_or_expression_analysis/regression_output_nasal_bacterial-viral_dna_species_kraken2_none_KRAKEN2_REFSEQDB.tsv",sep='\t') %>% mutate(seqtype = 'METAGENOMICS') %>% mutate(regressionclass = 'Nasal')   %>% mutate(dset = 'KRAKEN2')
regdata[[36]] = read.csv("i4_data_packet/differential_abundance_or_expression_analysis/regression_output_nasal_bacterial-viral_rna_species_kraken2_none_KRAKEN2_REFSEQDB.tsv",sep='\t') %>% mutate(seqtype = 'METATRANSCRIPTOMICS') %>% mutate(regressionclass = 'Nasal') %>% mutate(dset = 'KRAKEN2')
regdata[[37]] = read.csv("i4_data_packet/differential_abundance_or_expression_analysis/regression_output_skin_bacterial-viral_dna_species_kraken2_none_KRAKEN2_REFSEQDB.tsv",sep='\t') %>% mutate(seqtype = 'METAGENOMICS') %>% mutate(regressionclass = 'Skin') %>% mutate(dset = 'KRAKEN2')
regdata[[38]] = read.csv("i4_data_packet/differential_abundance_or_expression_analysis/regression_output_skin_bacterial-viral_rna_species_kraken2_none_KRAKEN2_REFSEQDB.tsv",sep='\t') %>% mutate(seqtype = 'METATRANSCRIPTOMICS') %>% mutate(regressionclass = 'Skin') %>% mutate(dset = 'KRAKEN2')
regdata[[39]] = read.csv("i4_data_packet/differential_abundance_or_expression_analysis/regression_output_skin_site_by_site_bacterial-viral_dna_species_kraken2_none_KRAKEN2_REFSEQDB.tsv",sep='\t') %>% mutate(seqtype = 'METAGENOMICS') %>% mutate(regressionclass = 'Skin_Components') %>% mutate(dset = 'KRAKEN2')
regdata[[40]] = read.csv("i4_data_packet/differential_abundance_or_expression_analysis/regression_output_skin_site_by_site_bacterial-viral_rna_species_kraken2_none_KRAKEN2_REFSEQDB.tsv",sep='\t') %>% mutate(seqtype = 'METATRANSCRIPTOMICS') %>% mutate(regressionclass = 'Skin_Components') %>% mutate(dset = 'KRAKEN2')

regression_output2 = bind_rows(regdata)
```
####### ICC ANALYSIS
```{r}

setwd('~/Dropbox (Mason Lab)/i4/')

# PARSE METADATA
metasub = meta %>% filter(Body.Location != "Swab Water",Body.Location != 'Open Air Control', Body.Location != "Deltoid - Pre-Biospy", !is.na(Body.Location)) %>% mutate(iscapsule = if_else(location == 'Capsule',1,0))  %>% mutate(skinoralnasal = if_else(isoral == 1,'ORAL',if_else(isskin==1,'SKIN','NASAL'))) %>% mutate(skinoralnasalcap = if_else(iscapsule == 1,'CAPSULE',skinoralnasal))
metasub$Timepoint_Recode = factor(metasub$Timepoint_Recode,levels = c('PRE-LAUNCH','MID-FLIGHT','POST-LAUNCH'))
metasub = metasub %>% mutate(Timepoint = if_else(Timepoint == 'Flight 1' | Timepoint == 'Flight 2','Mid-Flight',as.character(Timepoint)))
metasub$Timepoint = factor(metasub$Timepoint,levels=c('21-Jun','21-Aug','Sept Pre-Launch','Mid-Flight','Sept Post-Return','November','21-Dec',NA))
metasub$Timepoint_Recode = factor(metasub$Timepoint_Recode,levels=c('MID-FLIGHT','PRE-LAUNCH','POST-LAUNCH'))

build_icc_dfs <- function(regression_output2,c,regclass,st,dataset,abdata,metasub){
  sigs = c %>% filter(dset == dataset,seqtype == st,regclass2!='Overall',regclass2!='OVERALL') %>% filter(!grepl('-- LP',timetrend),!grepl('Decrease',timetrend)) %>% ungroup %>% select(yvar,regclass2) %>% distinct %>% mutate(regressionclass = if_else(regclass2 == 'Oral','Oral',if_else(regclass2=='Nasal','Nasal','Skin'))) 
  minval = min(abdata[abdata>0])
  abdata_meta = inner_join(abdata %>% rownames_to_column('SeqID'),metasub,by='SeqID') 
  relsigs = sigs %>% filter(regressionclass == regclass) %>% select(yvar) %>% unlist %>% unname%>% unique
  iccs = regression_output2 %>% filter(dset == dataset,seqtype == st,regressionclass == regclass) %>% filter(yvar %in% relsigs) %>% filter(effect == 'ran_pars') %>% select(group,estimate,yvar,regressionclass) %>% group_by(yvar) %>% summarise(group=group,total = sum(estimate),exp_crew = estimate/total) %>% ungroup%>% filter(group == 'Crew.ID') %>% mutate(loc2 = 'ICC')%>% select(loc2, yvar,exp_crew)%>% arrange(exp_crew) %>% rename(value = exp_crew) %>% mutate(regressionclass = regclass)
  iccs$yvar = fct_reorder(factor(iccs$yvar),iccs$value)
  prevs_melt = abdata_meta  %>% select(Crew.ID,Timepoint_Recode,location,any_of(relsigs)) %>% melt
  prevs_melt1 = prevs_melt %>% group_by(Crew.ID,Timepoint_Recode,location,variable) %>% summarise(average_ab = mean(value))
  totals =  prevs_melt %>% group_by(Crew.ID,location,Timepoint_Recode,variable) %>% summarise(totals = n())
  prevs_melt2 = prevs_melt %>% filter(value > 0) %>% group_by(Crew.ID,location,variable) %>% summarise(prevalence = n())
  prevs_melt = inner_join(prevs_melt1,prevs_melt2) 
  prevs_melt = inner_join(prevs_melt,totals) %>% mutate(prevalence = prevalence/totals) %>% select(-totals) %>% select(Crew.ID,variable,prevalence,average_ab,location,Timepoint_Recode) %>% dplyr::rename(yvar = variable) %>% melt(id.vars = c("Crew.ID","yvar","location","Timepoint_Recode")) %>% rename(loc2 = Crew.ID) %>% mutate(regressionclass = if_else(location != 'Oral' & location != 'Capsule' & location != 'Nasal','Skin',location)) %>% filter(regressionclass == regclass | regressionclass == 'Capsule')
  prevs_melt = prevs_melt %>% filter(yvar %in% relsigs)
  return(list(iccs,prevs_melt))
}

oraliccprevs = build_icc_dfs(regression_output2,c,'Oral','METAGENOMICS','GTDB',gtdb_metag,metasub)
#nasaliccprevs = build_icc_dfs(regression_output2,sigs,'Nasal','METAGENOMICS','GTDB',abdata_meta)
skiniccprevs = build_icc_dfs(regression_output2,c,'Skin','METAGENOMICS','GTDB',gtdb_metag,metasub)

iccs = bind_rows(oraliccprevs[[1]]%>% mutate(associationtype = 'Oral'),skiniccprevs[[1]]%>% mutate(associationtype = 'Skin') )
prevs = bind_rows(oraliccprevs[[2]]%>% mutate(associationtype = 'Oral'),skiniccprevs[[2]]%>% mutate(associationtype = 'Skin'))
##### ABOVE SHOULD BE ENOUGH TO GENERATE PLOT SANS PHYLUM ANNOTATION, WHICH WILL BE A BARPLOT
iccs$yvar = fct_reorder(factor(iccs$yvar),iccs$value)
prevs$yvar = factor(prevs$yvar,levels(iccs$yvar))

prevs$value [prevs$value ==0] =NA

colors = setNames(as.list(brewer.pal('Set3',n=11)),unique(metasub$location))

plot1 = ggplot(data = iccs,aes(x=yvar,y=value)) +theme_cowplot()+ geom_point(stat='identity',alpha=.7,size=2)  + theme(axis.text.x = element_blank())  + theme(legend.position = 'top',axis.title.x = element_blank())+ facet_grid(.~associationtype,scales='free_x',space='free_x') + ylab('ICC')


foo =  prevs %>% filter(variable =='average_ab',Timepoint_Recode == 'MID-FLIGHT') %>% filter(!is.na(value))

foo = foo %>% mutate(labelname = if_else(value>0 & variable =='average_ab' &location == 'Capsule',strsplit(as.character(yvar),'s__') %>% map_chr(2),''))


plot2 = ggplot(data =foo,aes(x=yvar,y=(value+minval),color=location))+theme_cowplot() + geom_point(stat='identity',alpha=.7,size=2) + facet_grid(loc2 ~ associationtype,scales='free_x',space='free_x') + theme(axis.text.x = element_blank(),axis.title.x = element_blank()) + scale_y_log10() + scale_color_manual(values = colors) + theme(legend.position = 'bottom') + ylab('log10(Relative Abundance)') 


plot1/plot2

ggsave('~/Dropbox (Mason Lab)/i4/iccanalysis/icc_bacterial_metag.pdf',width=10,height=10)
iccs_gtdb_metag  = iccs

oraliccprevs = build_icc_dfs(regression_output2,c,'Oral','METATRANSCRIPTOMICS','GENBANK-VIRUSES',viral_metat %>% t %>% as.data.frame(check.names=F) ,metasub)
nasaliccprevs = build_icc_dfs(regression_output2,c,'Nasal','METATRANSCRIPTOMICS','GENBANK-VIRUSES',viral_metat %>% t %>% as.data.frame(check.names=F) ,metasub)
skiniccprevs = build_icc_dfs(regression_output2,c,'Skin','METATRANSCRIPTOMICS','GENBANK-VIRUSES',viral_metat %>% t %>% as.data.frame(check.names=F) ,metasub)

iccs = bind_rows(oraliccprevs[[1]]%>% mutate(associationtype = 'Oral'),skiniccprevs[[1]]%>% mutate(associationtype = 'Skin'),nasaliccprevs[[1]]%>% mutate(associationtype = 'Nasal') )
prevs = bind_rows(oraliccprevs[[2]]%>% mutate(associationtype = 'Oral'),skiniccprevs[[2]]%>% mutate(associationtype = 'Skin'),nasaliccprevs[[2]]%>% mutate(associationtype = 'Nasal'))
##### ABOVE SHOULD BE ENOUGH TO GENERATE PLOT SANS PHYLUM ANNOTATION, WHICH WILL BE A BARPLOT
iccs$yvar = fct_reorder(factor(iccs$yvar),iccs$value)
prevs$yvar = factor(prevs$yvar,levels(iccs$yvar))

prevs$value [prevs$value ==0] =NA

plot3 = ggplot(data = iccs,aes(x=yvar,y=value)) +theme_cowplot()+ geom_point(stat='identity',alpha=.7,size=2)  + theme(axis.text.x = element_blank())  + theme(legend.position = 'top',axis.title.x = element_blank())+ facet_grid(.~associationtype,scales='free_x',space='free_x') + ylab('ICC')

plot4 = ggplot(data = prevs %>% filter(variable =='average_ab',Timepoint_Recode == 'MID-FLIGHT'),aes(x=yvar,y=(value+minval),color=location)) +theme_cowplot()+ geom_point(stat='identity',alpha=.7,size=2) + facet_grid(loc2 ~ associationtype,scales='free_x',space='free_x') + theme(axis.text.x = element_blank(),axis.title.x = element_blank()) + scale_y_log10() + scale_color_manual(values = colors) + theme(legend.position = 'bottom') + ylab('log10(Relative Abundance)')

plot3/plot4
ggsave('~/Dropbox (Mason Lab)/i4/iccanalysis/icc_viral_metat.pdf',width=10,height=10)

iccs_genvir_metag  = iccs

```
####### ENV VS CREW ACQUISITION -- BACTERIAL
```{r}
abdata_meta = inner_join(gtdb_metag %>% rownames_to_column('SeqID'),metasub %>% filter(!(location == 'Capsule' & Timepoint_Recode == 'PRE-FLIGHT'))%>% select(SeqID,Timepoint,Timepoint_Recode,Crew.ID,location),by='SeqID')  %>% melt %>% select(-SeqID) %>% mutate(skinoralnasalcapsule = if_else(location == 'Oral' | location == "Capsule" | location == 'Nasal',location,'Skin')) 
prevalence_across_time = abdata_meta %>% group_by(variable,skinoralnasalcapsule,location,Crew.ID,Timepoint) %>% summarize(n = mean(value > 0)) %>% dcast(Crew.ID + variable + skinoralnasalcapsule + location~ Timepoint, value.var = "n")
cap = prevalence_across_time %>% filter(location == 'Capsule') %>% select(variable,"Mid-Flight") %>% rename(capsule_prev_midflight = `Mid-Flight`)

prevalence_across_time = prevalence_across_time %>% group_by(variable,skinoralnasalcapsule,location) %>% summarise(pre_launch = sum(`Sept Pre-Launch`>0),post_launch = sum(`Sept Post-Return`>0))

prevalence_across_time = left_join(prevalence_across_time %>% filter(location != 'Capsule'),cap)
prevalence_across_time[is.na(prevalence_across_time)] = 0 

prevalence_across_time$capsule_prev_midflight[prevalence_across_time$capsule_prev_midflight>0]=1
prevalence_across_time$names = paste(prevalence_across_time$variable,prevalence_across_time$location,prevalence_across_time$skinoralnasalcapsule)

prevalence_across_time = prevalence_across_time %>% group_by(location) %>% arrange(location,desc(capsule_prev_midflight),pre_launch,post_launch) %>% column_to_rownames('names') %>% select(pre_launch,capsule_prev_midflight,post_launch,location) %>% filter(post_launch > pre_launch,post_launch>1)

temp = prevalence_across_time %>% select(location,capsule_prev_midflight) %>% group_by(location) %>% summarise(mean = mean(capsule_prev_midflight)) %>% arrange(desc(mean)) %>% select(location) %>% unlist %>% unname

prevalence_across_time$location=factor(prevalence_across_time$location,levels=temp)

prevalence_across_time = prevalence_across_time  %>% arrange(location,desc(capsule_prev_midflight),pre_launch,post_launch) 

col1 = c('1'='red','2'='blue','3'='cyan','4'='green','0' = 'black')
col2 = c('0' = 'black','1'='grey')

h1 = Heatmap(prevalence_across_time %>% ungroup %>% select(pre_launch) %>% t %>% as.data.frame,column_split = prevalence_across_time$location,show_row_names = T,show_column_names = F,cluster_columns =F,cluster_rows=F,col=col1,column_title_rot = 90)

h2 = Heatmap(prevalence_across_time %>% ungroup %>% select(capsule_prev_midflight) %>% t %>% as.data.frame,column_split = prevalence_across_time$location,show_row_names = T,show_column_names = F,cluster_columns =F,cluster_rows=F,col=col2)

h3 = Heatmap(prevalence_across_time %>% ungroup %>% select(post_launch) %>% t %>% as.data.frame,column_split = prevalence_across_time$location,show_row_names = T,show_column_names = F,cluster_columns =F,cluster_rows=F,col=col1,column_title_rot = 90)

pdf('~/Dropbox (Mason Lab)/i4/iccanalysis/shared_env_signal_bacmetag.pdf',width=8,height=3)
h1 %v% h2 %v% h3
dev.off()

prop_shared_bacmetag = prevalence_across_time %>% select(location,capsule_prev_midflight) %>% group_by(location) %>% summarise(shared_with_environment = mean(capsule_prev_midflight)) %>% mutate(shared_between_crew = 1-shared_with_environment,dset = 'Bacterial/Archaeal') %>% melt
```
####### ENV VS CREW ACQUISITION -- VIRAL
```{r}
abdata_meta = inner_join(viral_metat %>% t %>% as.data.frame(check.names=F) %>% rownames_to_column('SeqID'),metasub %>% filter(!(location == 'Capsule' & Timepoint_Recode == 'PRE-FLIGHT'))%>% select(SeqID,Timepoint,Timepoint_Recode,Crew.ID,location),by='SeqID')  %>% melt %>% select(-SeqID) %>% mutate(skinoralnasalcapsule = if_else(location == 'Oral' | location == "Capsule" | location == 'Nasal',location,'Skin')) 
prevalence_across_time = abdata_meta %>% group_by(variable,skinoralnasalcapsule,location,Crew.ID,Timepoint) %>% summarize(n = mean(value > 0)) %>% dcast(Crew.ID + variable + skinoralnasalcapsule + location~ Timepoint, value.var = "n")
cap = prevalence_across_time %>% filter(location == 'Capsule') %>% select(variable,"Mid-Flight") %>% rename(capsule_prev_midflight = `Mid-Flight`)

prevalence_across_time = prevalence_across_time %>% group_by(variable,skinoralnasalcapsule,location) %>% summarise(pre_launch = sum(`Sept Pre-Launch`>0),post_launch = sum(`Sept Post-Return`>0))

prevalence_across_time = left_join(prevalence_across_time %>% filter(location != 'Capsule'),cap)
prevalence_across_time[is.na(prevalence_across_time)] = 0 

prevalence_across_time$capsule_prev_midflight[prevalence_across_time$capsule_prev_midflight>0]=1
prevalence_across_time$names = paste(prevalence_across_time$variable,prevalence_across_time$location,prevalence_across_time$skinoralnasalcapsule)

prevalence_across_time = prevalence_across_time %>% group_by(location) %>% arrange(location,desc(capsule_prev_midflight),pre_launch,post_launch) %>% column_to_rownames('names') %>% select(pre_launch,capsule_prev_midflight,post_launch,location)%>% filter(post_launch > pre_launch,post_launch>1)

temp = prevalence_across_time %>% select(location,capsule_prev_midflight) %>% group_by(location) %>% summarise(mean = mean(capsule_prev_midflight)) %>% arrange(desc(mean)) %>% select(location) %>% unlist %>% unname

prevalence_across_time$location=factor(prevalence_across_time$location,levels=temp)

prevalence_across_time = prevalence_across_time  %>% arrange(location,desc(capsule_prev_midflight),pre_launch,post_launch) 

col1 = c('1'='red','2'='blue','3'='cyan','4'='green','0' = 'black')
col2 = c('0' = 'black','1'='grey')

h1 = Heatmap(prevalence_across_time %>% ungroup %>% select(pre_launch) %>% t %>% as.data.frame,column_split = prevalence_across_time$location,show_row_names = T,show_column_names = F,cluster_columns =F,cluster_rows=F,col=col1,column_title_rot = 90)

h2 = Heatmap(prevalence_across_time %>% ungroup %>% select(capsule_prev_midflight) %>% t %>% as.data.frame,column_split = prevalence_across_time$location,show_row_names = T,show_column_names = F,cluster_columns =F,cluster_rows=F,col=col2)

h3 = Heatmap(prevalence_across_time %>% ungroup %>% select(post_launch) %>% t %>% as.data.frame,column_split = prevalence_across_time$location,show_row_names = T,show_column_names = F,cluster_columns =F,cluster_rows=F,col=col1,column_title_rot = 90)

pdf('~/Dropbox (Mason Lab)/i4/iccanalysis/shared_env_signal_virmetat.pdf',width=8,height=3)
h1 %v% h2 %v% h3
dev.off()

prop_shared_virmetat = prevalence_across_time %>% select(location,capsule_prev_midflight) %>% group_by(location) %>% summarise(shared_with_environment = mean(capsule_prev_midflight)) %>% mutate(shared_between_crew = 1-shared_with_environment,dset = 'Viral') %>% melt
```
####### ENV VS CREW ACQUISITION -- SUPP
```{r}
# env shared summary

envsharedplot = bind_rows(prop_shared_bacmetag,prop_shared_virmetat) %>% mutate(skinoralnasal = if_else(location != 'Oral' & location != 'Nasal','Skin',as.character(location)))
ggplot(data = envsharedplot,aes(x = location,y = value*100,fill = variable)) + geom_bar(stat='identity') +facet_grid(dset ~ skinoralnasal,scales='free',space='free') + theme_bw() + theme(axis.text.x = element_text(angle=60,hjust=1)) + xlab('') + ylab('%') + scale_fill_manual(values=c('gray','black'))
ggsave('~/Dropbox (Mason Lab)/i4/iccanalysis/shared_env_signal_virmetat_overall.pdf',width=6,height=5)

envsharedplot$skinoralnasal = factor(envsharedplot$skinoralnasal,levels = c('Oral','Nasal','Skin'))

ggplot(envsharedplot %>% filter(variable == 'shared_with_environment') %>% group_by(skinoralnasal,dset)%>% summarise(mean = mean(value)*100,sd = sd(value)*100),aes(x = skinoralnasal,y=mean,fill=dset)) + geom_bar(stat='identity',position='dodge') + scale_fill_manual(values=c('lightgray','black')) + xlab('') + ggtitle('Taxa potentially propagated through capsule') + ylab('%') + geom_errorbar(aes(ymin = mean -sd,ymax = mean+sd), width=.2, position=position_dodge(.9),color='darkgrey') + theme_bw() + geom_text(aes(label=round(mean,1)), position=position_dodge(width=0.9), vjust=-0.25)
ggsave('~/Dropbox (Mason Lab)/i4/iccanalysis/propagation_rates.pdf',width=6,height=5)

```
####### ICC PLOTS FOR SUPP
```{r}

#### FOR SUPP

oraliccprevs = build_icc_dfs(regression_output2,c,'Oral','METAGENOMICS','ASSEMBLED-BACTERIA',bass_metag,metasub)
#nasaliccprevs = build_icc_dfs(regression_output2,sigs,'Nasal','METAGENOMICS','GTDB',abdata_meta)
skiniccprevs = build_icc_dfs(regression_output2,c,'Skin','METAGENOMICS','ASSEMBLED-BACTERIA',bass_metag,metasub)

iccs = bind_rows(oraliccprevs[[1]]%>% mutate(associationtype = 'Oral'),skiniccprevs[[1]]%>% mutate(associationtype = 'Skin') )
prevs = bind_rows(oraliccprevs[[2]]%>% mutate(associationtype = 'Oral'),skiniccprevs[[2]]%>% mutate(associationtype = 'Skin'))
##### ABOVE SHOULD BE ENOUGH TO GENERATE PLOT SANS PHYLUM ANNOTATION, WHICH WILL BE A BARPLOT
iccs$yvar = fct_reorder(factor(iccs$yvar),iccs$value)
prevs$yvar = factor(prevs$yvar,levels(iccs$yvar))

plot1 = ggplot(data = iccs,aes(x=yvar,y=value)) + geom_point(stat='identity',alpha=.7,size=2)  + theme(axis.text.x = element_blank())  + theme(legend.position = 'top',axis.title.x = element_blank())+ facet_grid(.~associationtype,scales='free_x',space='free_x') + ylab('ICC')

prevs$value [prevs$value ==0] =NA

plot2 = ggplot(data = prevs %>% filter(variable =='average_ab',Timepoint_Recode == 'MID-FLIGHT',regressionclass!='NASAL') ,aes(x=yvar,y=(value+minval),color=location)) + geom_point(stat='identity',alpha=.7,size=2) + facet_grid(loc2 ~ associationtype,scales='free_x',space='free_x') + theme(axis.text.x = element_blank(),axis.title.x = element_blank()) + scale_y_log10() + scale_color_brewer(palette = 'Set3') + theme(legend.position = 'Bottom') + ylab('log10(Relative Abundance)')

plot1/plot2

ggsave('~/Dropbox (Mason Lab)/i4/iccanalysis/icc_assembledbacterial_metag.pdf',width=8,height=10)

oraliccprevs = build_icc_dfs(regression_output2,c,'Oral','METATRANSCRIPTOMICS','ASSEMBLED-VIRUSES',viral_ass_metat %>% t %>% as.data.frame(check.names=F) ,metasub)
#nasaliccprevs = build_icc_dfs(regression_output2,c,'Nasal','METATRANSCRIPTOMICS','ASSEMBLED-VIRUSES',viral_ass_metat %>% t %>% as.data.frame(check.names=F) ,metasub)
skiniccprevs = build_icc_dfs(regression_output2,c,'Skin','METATRANSCRIPTOMICS','ASSEMBLED-VIRUSES',viral_ass_metat %>% t %>% as.data.frame(check.names=F) ,metasub)

iccs = bind_rows(oraliccprevs[[1]]%>% mutate(associationtype = 'Oral'),skiniccprevs[[1]]%>% mutate(associationtype = 'Skin'),nasaliccprevs[[1]]%>% mutate(associationtype = 'Nasal') )
prevs = bind_rows(oraliccprevs[[2]]%>% mutate(associationtype = 'Oral'),skiniccprevs[[2]]%>% mutate(associationtype = 'Skin'),nasaliccprevs[[2]]%>% mutate(associationtype = 'Nasal'))
##### ABOVE SHOULD BE ENOUGH TO GENERATE PLOT SANS PHYLUM ANNOTATION, WHICH WILL BE A BARPLOT
iccs$yvar = fct_reorder(factor(iccs$yvar),iccs$value)
prevs$yvar = factor(prevs$yvar,levels(iccs$yvar))

plot3 = ggplot(data = iccs,aes(x=yvar,y=value)) + geom_point(stat='identity',alpha=.7,size=2)  + theme(axis.text.x = element_blank())  + theme(legend.position = 'top',axis.title.x = element_blank())+ facet_grid(.~associationtype,scales='free_x',space='free_x') + ylab('ICC')

plot4 = ggplot(data = prevs %>% filter(variable =='average_ab',Timepoint_Recode == 'MID-FLIGHT',regressionclass!='NASAL'),aes(x=yvar,y=(value+minval),color=location)) + geom_point(stat='identity',alpha=.7,size=2) + facet_grid(loc2 ~ associationtype,scales='free_x',space='free_x') + theme(axis.text.x = element_blank(),axis.title.x = element_blank()) + scale_y_log10() + scale_color_brewer(palette = 'Set3') + theme(legend.position = 'bottom') + ylab('log10(Relative Abundance)')

plot3/plot4
ggsave('~/Dropbox (Mason Lab)/i4/iccanalysis/icc_viral_metat_ass.pdf',width=8,height=10)

```
####### IMMUNE STUFF
```{r} 
# immune cell interaction
setwd('~/Dropbox (Mason Lab)/i4/immune_modeling/')

parse_output <- function(dfile,seqtyped){
  immunecell = gsub('.gene.expression.orig.ident.csv','',dfile)
  immunecell = gsub('DECREASED','',immunecell)
  immunecell = gsub('LASSO_OUT','',immunecell)
  immunecell = gsub('VIRUSES','',immunecell)
  immunecell = gsub('VIRAL','',immunecell)
  immunecell = gsub('METATRANSCRIPTOMIC','',immunecell)
  immunecell = gsub('METAGENOMIC','',immunecell)
  immunecell = gsub('average','',immunecell)
  immunecell = gsub('_','',immunecell)
  immunecell = gsub('\\.','',immunecell)
  # heatmap
  type = 'BACTERIAL'
  if(grepl('VIRUSES',dfile)){
    type='VIRAL'
  }
  if(grepl('VIRAL',dfile)){
    type='VIRAL'
  }
  dir = 'INCREASED'
  if(grepl('DECREASED',dfile)){
    dir='DECREASED'
  }
  data = read.csv(dfile) %>% mutate(seqtype = seqtyped)%>% filter(features!='(Intercept)') %>% mutate(dset = type,directiontoflight = dir) %>% mutate(dset = if_else(grepl('d__Bacteria',microbialfeaturename) | dset == 'VIRAL',dset,'GENE-CATALOG')) %>% mutate(celltype =immunecell)
  if(nrow(data)==0){
    return('none')
  }
  x = data
  # count number of activated host genes 
  
  counts = data %>% select(regressionclass) %>% table %>% data.frame %>% mutate(seqtype = seqtyped)%>% mutate(celltype =immunecell)
  colnames(counts)[1] = 'Site Type'
  colnames(counts)[2] = 'Number of associated genes'
  
  # count features with most genes
  counts2 =data %>% select(regressionclass,microbialfeaturename) %>% table %>% data.frame %>% filter(Freq>1) %>% group_by(regressionclass) %>% slice_max(n=3,Freq) %>% mutate(seqtype = seqtyped)%>% mutate(celltype =immunecell)
  colnames(counts2)[3] = 'Number of associated genes'
  
  # report most strongly activated host genes
  maxes = data %>% group_by(regressionclass) %>% slice_max(n=3,coefficient) %>% mutate(seqtype = seqtyped)%>% mutate(celltype =immunecell)
  mins = data %>% group_by(regressionclass)  %>% slice_min(n=3,coefficient) %>% mutate(seqtype = seqtyped)%>% mutate(celltype =immunecell)
  
  data = data  %>% dcast(microbialfeaturename + regressionclass + seqtype~ features, value.var = 'coefficient')
  data[is.na(data)] = 0
  return(list(counts,counts2,bind_rows(maxes,mins),x))
}

metagparse = list()
files = list.files()
files = files[grepl('LASSO',files)]
files = files[grepl('METAGENOMIC',files)]
for(f in files){
  out = parse_output(f,'METAGENOMICS')
  if(typeof(out)!='character'){
      metagparse[[f]] = out
  }
}
 
metatparse = list()
files = list.files()
files = files[grepl('LASSO',files)]
files = files[grepl('METATRANSCRIPTOMIC',files)]
for(f in files){
  out = parse_output(f,'METATRANSCRIPTOMICS')
  if(typeof(out)!='character'){
      metatparse[[f]] = out
  }
}

metagout = bind_rows(map(metagparse, function(x) x[[4]])) %>% select(-any_of(c("X","X.1")))
metatout = bind_rows(map(metatparse, function(x) x[[4]]))%>% select(-any_of(c("X","X.1")))

alldata = bind_rows(metagout,metatout)

foo = expand.grid(unique(alldata$seqtype),unique(alldata$regressionclass),unique(alldata$dset),unique(alldata$directiontoflight),unique(alldata$celltype))
colnames(foo) =c('seqtype','regressionclass','dset','directiontoflight','celltype')

alldata = left_join(foo,alldata)
alldata[is.na(alldata)]=0

alldata$dset = factor(alldata$dset,levels = c('BACTERIAL','VIRAL','GENE-CATALOG'))
alldata$regressionclass = factor(alldata$regressionclass,levels = c('Oral','Nasal','Skin'))
alldata$celltype = gsub('T',' T',alldata$celltype)
alldata$celltype = gsub('Mono',' Mono',alldata$celltype)
alldata$celltype = factor(alldata$celltype,levels = c("CD4 T", "CD8 T", "other T", "B", "NK", "CD14 Mono", "CD16 Mono", "DC", "other"))

#### INCREASED FEATURES
celltypehitscount = alldata %>% filter(directiontoflight == 'INCREASED') %>% group_by(seqtype,dset,regressionclass,celltype) %>% count() %>% ungroup %>% dcast(seqtype +dset +regressionclass ~celltype,value.var = "n")
celltypehitscount[is.na(celltypehitscount)] = 0
celltypehitscount = melt(celltypehitscount,id.vars = c("seqtype","dset","regressionclass")) %>% rename(celltype = variable)

ggplot(celltypehitscount,aes(y=value,x=regressionclass,fill=seqtype)) + theme_minimal() + geom_bar(stat='identity',position = position_dodge(preserve = "single")) + facet_grid(dset ~ celltype,scales='free_y',space='free_x') + theme(legend.position = 'None',axis.text.x = element_text(angle = 60,hjust = 1))+scale_fill_manual(values = c('darkblue','#BE1E2D')) + ggtitle('Immune cell gene expression and microbial features increased after flight')
ggsave('~/Dropbox (Mason Lab)/i4/immune_modeling/top_counts_increase.pdf',width=5,height=3)
### REPORT COUNTS OF 

# get top ones and plot barplots
#hm = Heatmap(data %>% select(-regressionclass,-seqtype,-microbialfeaturename),row_split = data$regressionclass,show_row_names=T,column_labels = data %>% select(-regressionclass,-seqtype,-microbialfeaturename) %>% colnames,show_column_names = T)

#### DECREASED FEATURES
celltypehitscount = alldata %>% filter(directiontoflight == 'DECREASED') %>% group_by(seqtype,dset,regressionclass,celltype) %>% count() %>% ungroup %>% dcast(seqtype +dset +regressionclass ~celltype,value.var = "n")
celltypehitscount[is.na(celltypehitscount)] = 0
celltypehitscount = melt(celltypehitscount,id.vars = c("seqtype","dset","regressionclass")) %>% rename(celltype = variable)

ggplot(celltypehitscount,aes(y=value,x=regressionclass,fill=seqtype)) + theme_minimal() + geom_bar(stat='identity',position = position_dodge(preserve = "single")) + facet_grid(dset ~ celltype,scales='free_y',space='free_x') + theme(legend.position = 'None',axis.text.x = element_text(angle = 60,hjust = 1))+scale_fill_manual(values = c('darkblue','#BE1E2D')) + ggtitle('Immune cell gene expression and microbial features decreased after flight')
ggsave('~/Dropbox (Mason Lab)/i4/immune_modeling/top_counts_decrease.pdf',width=5,height=3)

### GET MOST IMPACTFUL MICROBIAL EXPRESSION CHANGES
foo = left_join(alldata,gene_annos,c('microbialfeaturename' ='yvar'))

impactfulbugs = foo %>% filter(features!="0")%>% filter(directiontoflight == 'INCREASED') %>% filter(dset == 'VIRAL' | dset == 'BACTERIAL' | product != 'hypothetical protein')%>% group_by(seqtype,dset,regressionclass,microbialfeaturename) %>% count() %>% group_by(seqtype,dset,regressionclass)%>% slice_max(n,n = 10) %>% arrange(n)%>% ungroup %>% select(-n)

temp = impactfulbugs %>% filter(dset == 'GENE-CATALOG')  %>% select(microbialfeaturename)%>% unlist %>% unname
temp2 = impactfulbugs %>% filter(dset == 'VIRAL') %>% mutate(microbialfeaturename = strsplit(microbialfeaturename,',') %>% map_chr(1)) %>% select(microbialfeaturename)%>% unlist %>% unname
temp3 = impactfulbugs %>% filter(dset == 'BACTERIAL') %>% mutate(microbialfeaturename = strsplit(microbialfeaturename,'s__') %>% map_chr(2)) %>% select(microbialfeaturename)%>% unlist %>% unname

ordering = c(temp,temp2,temp3)

impactfulbugplot = left_join(impactfulbugs,alldata %>% select(seqtype,dset,regressionclass,microbialfeaturename,celltype,features,directiontoflight)%>% filter(directiontoflight == 'INCREASED')  %>% filter(!is.na(features))) %>% distinct
impactfulbugplot = impactfulbugplot %>% group_by(seqtype,dset,regressionclass,microbialfeaturename,celltype) %>% count()

temp = impactfulbugplot %>% filter(dset == 'GENE-CATALOG') 
temp2 = impactfulbugplot %>% filter(dset == 'VIRAL') %>% mutate(microbialfeaturename = strsplit(microbialfeaturename,',') %>% map_chr(1))
temp3 = impactfulbugplot %>% filter(dset == 'BACTERIAL') %>% mutate(microbialfeaturename = strsplit(microbialfeaturename,'s__') %>% map_chr(2))

impactfulbugplot = bind_rows(temp,temp2,temp3) 

impactfulbugplot$microbialfeaturename = factor(impactfulbugplot$microbialfeaturename,levels = ordering)
impactfulbugplot$dset = factor(impactfulbugplot$dset,levels = c('BACTERIAL','VIRAL','GENE-CATALOG'))
impactfulbugplot$regressionclass = factor(impactfulbugplot$regressionclass,levels = c('Oral','Nasal','Skin'))
#impactfulbugplot$celltype = gsub('T',' T',impactfulbugplot$celltype)
#impactfulbugplot$celltype = gsub('Mono',' Mono',impactfulbugplot$celltype)
#impactfulbugplot$celltype = factor(impactfulbugplot$celltype,levels = c("CD4 T", "CD8 T", "other T", "B", "NK", "CD14 Mono", "CD16 Mono", "DC", "other"))

impactfulbugplot = left_join(impactfulbugplot,gene_annos,c('microbialfeaturename' ='yvar'))
impactfulbugplot = impactfulbugplot %>% mutate(microbialfeaturename = if_else(!is.na(product),product,as.character(microbialfeaturename)))

#get plot factor order, again I guess
foobar = impactfulbugplot %>% group_by(seqtype,dset,regressionclass,microbialfeaturename) %>% summarise(n=sum(n))
foobar$microbialfeaturename = fct_reorder(as.factor(foobar$microbialfeaturename),foobar$n)

impactfulbugplot$microbialfeaturename = factor(impactfulbugplot$microbialfeaturename ,levels = levels(foobar$microbialfeaturename))

ggplot(impactfulbugplot %>% filter(seqtype == 'METATRANSCRIPTOMICS'),aes(x = n,y =microbialfeaturename,fill = celltype))+ geom_bar(stat='identity') + facet_grid(dset +regressionclass~.  ,scales = 'free_y',space = 'free_y') + theme_minimal() + scale_fill_brewer(palette = 'Set3') + xlab('Count') + ylab('')+theme(legend.position = 'None')
ggsave('~/Dropbox (Mason Lab)/i4/immune_modeling/impactfulbugplot.pdf',width=6,height=10)

#### GET MOST IMPACTED HUMAN GENES BY CELL TYPE

tophits = alldata %>% filter(dset != 'GENE-CATALOG')%>% ungroup %>% filter(features!="0",seqtype == 'METATRANSCRIPTOMICS',directiontoflight =='INCREASED')%>% select(celltype,coefficient,features,dset) %>% group_by(celltype,dset) %>% filter(coefficient>0) %>% slice_max(coefficient,n=5) 
bottomhits = alldata %>% filter(dset != 'GENE-CATALOG')%>% ungroup%>% filter(features!="0",seqtype == 'METATRANSCRIPTOMICS',directiontoflight =='INCREASED') %>% select(celltype,coefficient,features,dset) %>% group_by(celltype,dset) %>% filter(coefficient<0) %>% slice_min(coefficient,n=5)

allhits = bind_rows(tophits,bottomhits) %>% arrange(coefficient) %>% group_by(celltype) %>% arrange(desc(coefficient)) 
library(annotables)
annos = melt(grch38 %>% select(-start,-end,-strand),c('description')) 

allhits = left_join(allhits,annos,by=c('features'='value')) %>% mutate(label = if_else(is.na(description),features,description))

ggplot(allhits %>% filter(dset == 'BACTERIAL'),aes(x = coefficient, y = reorder(label,coefficient))) + geom_bar(stat='identity') + facet_grid(celltype ~.  ,scales = 'free_y',space = 'free_y') + theme_bw() + xlab('Coefficient') + ylab('') + ggtitle('Top bacteria-associated human genes')
ggsave('~/Dropbox (Mason Lab)/i4/immune_modeling/tophumangenes_annotations_BAC.pdf',width=7,height=8)
2
ggplot(allhits %>% filter(dset == 'VIRAL'),aes(x = coefficient, y = reorder(label,coefficient))) + geom_bar(stat='identity') + facet_grid(celltype ~.  ,scales = 'free_y',space = 'free_y') + theme_bw() + xlab('Coefficient') + ylab('') + ggtitle('Top virus-associated human genes')
ggsave('~/Dropbox (Mason Lab)/i4/immune_modeling/tophumangenes_annotations_VIR.pdf',width=7,height=5)



ggplot(allhits,aes(x = coefficient, y = reorder(label,coefficient))) + geom_bar(stat='identity') + facet_grid(dset + celltype ~.  ,scales = 'free_y',space = 'free_y') + theme_bw() + xlab('Coefficient') + ylab('') + ggtitle('Top associated human genes')
ggsave('~/Dropbox (Mason Lab)/i4/immune_modeling/tophumangenes_annotation.pdf',width=6,height=15)



```



